#3 MAGS_Catalouge

##3.1 Genom Phylogeny

### Load data

```{r load_data}
load("data/data.Rdata")
```

### Generate the phylum color heatmap

```{r load_phylum_heatmap}
# Generate the phylum color heatmap
phylum_heatmap <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(genome,phylum) %>%
    mutate(phylum = factor(phylum, levels = unique(phylum))) %>%
    column_to_rownames(var = "genome")

# Generate  basal tree
circular_tree <- force.ultrametric(genome_tree, method="extend") %>% # extend to ultrametric for the sake of visualisation
    ggtree(., layout="fan", open.angle=10, size=0.5)
```

### Build circular tree

###Build basal tree

```{r load_phylum_basal_tree}
circular_tree <- force.ultrametric(genome_tree, method="extend") %>% # extend to ultrametric for the sake of visualisation
  ggtree(., layout="fan", open.angle=10, size=0.5)
```

### Add phylum ring

```{r load_circular_tree}
circular_tree <- gheatmap(circular_tree, phylum_heatmap, offset=0.55, width=0.1, colnames=FALSE) +
  scale_fill_manual(values=phylum_colors) +
  geom_tiplab2(size=1, hjust=-0.1) +
  theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))
```

### Flush color scale to enable a new color scheme in the next ring

```{r load_circular_tree_colour}
circular_tree <- circular_tree + new_scale_fill()
```

### Add completeness ring

```{r load_circular_tree_ring}
circular_tree <- circular_tree +
  new_scale_fill() +
  scale_fill_gradient(low = "#d1f4ba", high = "#f4baba") +
  geom_fruit(
    data=genome_metadata,
    geom=geom_bar,
    mapping = aes(x=completeness, y=genome, fill=contamination),
    offset = 0.55,
    orientation="y",
    stat="identity")
```

### Add genome-size ring

```{r load_circular_tree_sizering}
circular_tree <-  circular_tree +
  new_scale_fill() +
  scale_fill_manual(values = "#cccccc") +
  geom_fruit(
    data=genome_metadata,
    geom=geom_bar,
    mapping = aes(x=length, y=genome),
    offset = 0.05,
    orientation="y",
    stat="identity")
```

### Generate tree (without labels)

```{r data_estimations_plot, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggsave("clean_circular_tree_large.png", plot=circular_tree, dpi=300, width=70, height=70, limitsize = FALSE)

#Plot circular tree
circular_tree %>% open_tree(30) %>% rotate_tree(90)
```

##3.2 Genom Quality

```{r genome_quality}
genome_metadata$completeness %>% mean()

genome_metadata$completeness %>% sd()

genome_metadata$contamination %>% mean()

genome_metadata$contamination %>% sd()
```

### Genom biplot
```{r genome_biplot}
genome_biplot <- genome_metadata %>%
  select(c(genome,domain,phylum,completeness,contamination,length)) %>%
  arrange(match(genome, rev(genome_tree$tip.label))) %>% #sort MAGs according to phylogenetic tree
  ggplot(aes(x=completeness,y=contamination,size=length,color=phylum)) +
  geom_point(alpha=0.7) +
  ylim(c(10,0)) +
  scale_color_manual(values=phylum_colors) +
  labs(y= "Contamination", x = "Completeness") +
  theme_classic() +
  theme(legend.position = "none")
```

### Genome contamination
```{r genome_contamination}
genome_contamination <- genome_metadata %>%
  ggplot(aes(y=contamination)) +
  ylim(c(10,0)) +
  geom_boxplot(colour = "#999999", fill="#cccccc") +
  theme_void() +
  theme(legend.position = "none",
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        plot.margin = unit(c(0, 0, 0.40, 0),"inches")) #add bottom-margin (top, right, bottom, left)
```
### Genome completness
```{r genome_completness}
genome_completeness <- genome_metadata %>%
  ggplot(aes(x=completeness)) +
  xlim(c(50,100)) +
  geom_boxplot(colour = "#999999", fill="#cccccc") +
  theme_void() +
  theme(legend.position = "none",
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        plot.margin = unit(c(0, 0, 0, 0.50),"inches")) #add left-margin (top, right, bottom, left)
```

###Render composite figure
```{r compisote_figure}
grid.arrange(grobs = list(genome_completeness,genome_biplot,genome_contamination),
             layout_matrix = rbind(c(1,1,1,1,1,1,1,1,1,1,1,4),
                                   c(2,2,2,2,2,2,2,2,2,2,2,3),
                                   c(2,2,2,2,2,2,2,2,2,2,2,3),
                                   c(2,2,2,2,2,2,2,2,2,2,2,3),
                                   c(2,2,2,2,2,2,2,2,2,2,2,3),
                                   c(2,2,2,2,2,2,2,2,2,2,2,3),
                                   c(2,2,2,2,2,2,2,2,2,2,2,3),
                                   c(2,2,2,2,2,2,2,2,2,2,2,3),
                                   c(2,2,2,2,2,2,2,2,2,2,2,3),
                                   c(2,2,2,2,2,2,2,2,2,2,2,3),
                                   c(2,2,2,2,2,2,2,2,2,2,2,3)))
```

##3.3 Functional Overview

### Aggregate basal GIFT into elements

```{r GIFT}
function_table <- genome_gifts %>%
  to.elements(., GIFT_db)
```

###Generate basal tree
```{r function}
function_tree <- force.ultrametric(genome_tree, method="extend") %>%
  ggtree(., size = 0.3)
```

###Add phylum colors next to the tree tips
```{r phylum_colors}
function_tree <- gheatmap(function_tree, phylum_heatmap, offset=0, width=0.1, colnames=FALSE) +
  scale_fill_manual(values=phylum_colors) +
  labs(fill="Phylum")
```
###Reset fill scale to use a different colour profile in the heatmap
```{r reset_scale}
function_tree <- function_tree + new_scale_fill()
```
###Add functions heatmap
```{r heatmap}
function_tree <- gheatmap(function_tree, function_table, offset=0.5, width=3.5, colnames=FALSE) +
  vexpand(.08) +
  coord_cartesian(clip = "off") +
  scale_fill_gradient(low = "#f4f4f4", high = "steelblue", na.value="white") +
  labs(fill="GIFT")
```
###Add completeness barplots
```{r completness_barplot}
function_tree <- function_tree +
  geom_fruit(data=genome_metadata,
             geom=geom_bar,
             grid.params=list(axis="x", text.size=2, nbreak = 1),
             axis.params=list(vline=TRUE),
             mapping = aes(x=length, y=genome, fill=completeness),
             offset = 3.8,
             orientation="y",
             stat="identity") +
  scale_fill_gradient(low = "#cf8888", high = "#a2cc87") +
  labs(fill="Genome\ncompleteness")

function_tree
```

##3.4 Functional Ordination

### Generate the tSNE ordination
```{r tSNE}
tSNE_function <- Rtsne(X=function_table, dims = 2, check_duplicates = FALSE)
```
### Plot the ordination
```{r functional_ordination}
function_ordination <- tSNE_function$Y %>%
  as.data.frame() %>%
  mutate(genome=rownames(function_table)) %>%
  inner_join(genome_metadata, by="genome") %>%
  rename(tSNE1="V1", tSNE2="V2") %>%
  select(genome,phylum,tSNE1,tSNE2, length) %>%
  ggplot(aes(x = tSNE1, y = tSNE2, color = phylum, size=length))+
  geom_point(shape=16, alpha=0.7) +
  scale_color_manual(values=phylum_colors) +
  theme_minimal() +
  labs(color="Phylum", size="Genome size") +
  guides(color = guide_legend(override.aes = list(size = 5))) # enlarge Phylum dots in legend

function_ordination
```