<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>AlberdiLab | Fix et al 2025</title>
  <meta name="description" content="<p>Data analysis code for the study on the gut microbiomes of brushtail possums in tasmania.</p>" />
  <meta name="generator" content="bookdown #bookdown:version# and GitBook 2.6.7" />

  <meta property="og:title" content="AlberdiLab | Fix et al 2025" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>Data analysis code for the study on the gut microbiomes of brushtail possums in tasmania.</p>" />
  <meta name="github-repo" content="<a href="https://github.com/alberdilab/btp_metagenomics.git" class="uri">https://github.com/alberdilab/btp_metagenomics.git</a>" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="AlberdiLab | Fix et al 2025" />
  
  <meta name="twitter:description" content="<p>Data analysis code for the study on the gut microbiomes of brushtail possums in tasmania.</p>" />
  

<meta name="author" content="Lukas Fix, Antton Alberdi, Raphael Eisenhofer" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<!--bookdown:link_prev-->
<!--bookdown:link_next-->
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



<!--bookdown:title:start-->
<div id="header">
<h1 class="title">AlberdiLab | Fix et al 2025</h1>
<h2 class="subtitle"><em>Brush-tail possum metagenomics</em></h2>
<p class="author"><em>Lukas Fix, Antton Alberdi, Raphael Eisenhofer</em></p>
<p class="date"><em>Last update: 2025-05-20</em></p>
</div>
<!--bookdown:title:end-->

<!--bookdown:toc:start-->
  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">
<!--bookdown:toc2:start-->
<ul>
<li><a href="#libaries" id="toc-libaries"><span class="toc-section-number">0.1</span> Libaries</a></li>
<li><a href="#load-data" id="toc-load-data"><span class="toc-section-number">0.2</span> Load data</a>
<ul>
<li><a href="#sample-metadata" id="toc-sample-metadata"><span class="toc-section-number">0.2.1</span> Sample metadata</a></li>
</ul></li>
<li><a href="#load_read_counts" id="toc-load_read_counts"><span class="toc-section-number">0.3</span> load_read_counts</a></li>
<li><a href="#load_genome_metadata" id="toc-load_genome_metadata"><span class="toc-section-number">0.4</span> load_genome_metadata</a></li>
<li><a href="#genome-base-hits" id="toc-genome-base-hits"><span class="toc-section-number">0.5</span> Genome base hits</a></li>
<li><a href="#genome-tree" id="toc-genome-tree"><span class="toc-section-number">0.6</span> Genome tree</a></li>
<li><a href="#genome-annotations" id="toc-genome-annotations"><span class="toc-section-number">0.7</span> Genome annotations</a></li>
<li><a href="#filter-reads-by-coverage" id="toc-filter-reads-by-coverage"><span class="toc-section-number">0.8</span> Filter reads by coverage</a></li>
<li><a href="#transform-reads-into-genome-counts" id="toc-transform-reads-into-genome-counts"><span class="toc-section-number">0.9</span> Transform reads into genome counts</a>
<ul>
<li><a href="#distill-annotations-into-gifts" id="toc-distill-annotations-into-gifts"><span class="toc-section-number">0.9.1</span> Distill annotations into GIFTs</a></li>
</ul></li>
<li><a href="#prepare-color-scheme" id="toc-prepare-color-scheme"><span class="toc-section-number">0.10</span> Prepare color scheme</a></li>
<li><a href="#wrap-working-objects" id="toc-wrap-working-objects"><span class="toc-section-number">0.11</span> Wrap working objects</a></li>
<li><a href="#data-statistics" id="toc-data-statistics"><span class="toc-section-number">1</span> Data statistics</a>
<ul>
<li><a href="#filter-data" id="toc-filter-data"><span class="toc-section-number">1.1</span> Filter Data</a></li>
<li><a href="#load-map" id="toc-load-map"><span class="toc-section-number">1.2</span> Load Map</a></li>
<li><a href="#plot-possum-locations" id="toc-plot-possum-locations"><span class="toc-section-number">1.3</span> Plot Possum Locations</a></li>
<li><a href="#load-map-1" id="toc-load-map-1"><span class="toc-section-number">1.4</span> Load Map</a></li>
<li><a href="#plot-possum-locations-1" id="toc-plot-possum-locations-1"><span class="toc-section-number">1.5</span> Plot Possum Locations</a></li>
<li><a href="#sequencing-reads-statistics" id="toc-sequencing-reads-statistics"><span class="toc-section-number">1.6</span> Sequencing reads statistics</a></li>
<li><a href="#dna-fractions" id="toc-dna-fractions"><span class="toc-section-number">1.7</span> DNA fractions</a></li>
<li><a href="#recovered-microbial-fraction" id="toc-recovered-microbial-fraction"><span class="toc-section-number">1.8</span> Recovered microbial fraction</a></li>
</ul></li>
<li><a href="#mag-catalogue" id="toc-mag-catalogue"><span class="toc-section-number">2</span> MAG catalogue</a>
<ul>
<li><a href="#gut-microbiota" id="toc-gut-microbiota"><span class="toc-section-number">2.1</span> Gut microbiota</a>
<ul>
<li><a href="#genome-phylogeny" id="toc-genome-phylogeny"><span class="toc-section-number">2.1.1</span> Genome phylogeny</a></li>
<li><a href="#taxonomy-overview" id="toc-taxonomy-overview"><span class="toc-section-number">2.1.2</span> Taxonomy overview</a></li>
<li><a href="#mag-size-mb" id="toc-mag-size-mb"><span class="toc-section-number">2.1.3</span> Mag size (MB)</a></li>
<li><a href="#genome-quality" id="toc-genome-quality"><span class="toc-section-number">2.1.4</span> Genome quality</a></li>
<li><a href="#functional-overview" id="toc-functional-overview"><span class="toc-section-number">2.1.5</span> Functional overview</a></li>
<li><a href="#functional-ordination" id="toc-functional-ordination"><span class="toc-section-number">2.1.6</span> Functional ordination</a></li>
</ul></li>
</ul></li>
<li><a href="#community-composition" id="toc-community-composition"><span class="toc-section-number">3</span> Community composition</a>
<ul>
<li><a href="#gut-microbiota-1" id="toc-gut-microbiota-1"><span class="toc-section-number">3.1</span> Gut microbiota</a>
<ul>
<li><a href="#taxonomy-overview-1" id="toc-taxonomy-overview-1"><span class="toc-section-number">3.1.1</span> Taxonomy overview</a></li>
</ul></li>
</ul></li>
<li><a href="#alpha-diversity" id="toc-alpha-diversity"><span class="toc-section-number">4</span> Alpha diversity</a>
<ul>
<li><a href="#gut-microbiota-2" id="toc-gut-microbiota-2"><span class="toc-section-number">4.1</span> Gut microbiota</a>
<ul>
<li><a href="#mixed-models" id="toc-mixed-models"><span class="toc-section-number">4.1.1</span> Mixed models</a></li>
</ul></li>
</ul></li>
<li><a href="#beta-diversity" id="toc-beta-diversity"><span class="toc-section-number">5</span> Beta diversity</a>
<ul>
<li><a href="#gut-microbiota-3" id="toc-gut-microbiota-3"><span class="toc-section-number">5.1</span> Gut microbiota</a>
<ul>
<li><a href="#permanova" id="toc-permanova"><span class="toc-section-number">5.1.1</span> Permanova</a></li>
</ul></li>
</ul></li>
<li><a href="#richness" id="toc-richness"><span class="toc-section-number">6</span> Richness</a></li>
<li><a href="#phylogenetic-diversity-beta_q1p" id="toc-phylogenetic-diversity-beta_q1p"><span class="toc-section-number">7</span> Phylogenetic diversity (beta_q1p)</a></li>
<li><a href="#functional-diversity-beta_q1f" id="toc-functional-diversity-beta_q1f"><span class="toc-section-number">8</span> Functional diversity (beta_q1f)</a>
<ul>
<li><a href="#nmds" id="toc-nmds"><span class="toc-section-number">8.0.1</span> NMDS</a></li>
</ul></li>
<li><a href="#gut-microbiota-differential-abundance-analysis" id="toc-gut-microbiota-differential-abundance-analysis"><span class="toc-section-number">9</span> Gut microbiota: Differential abundance analysis</a>
<ul>
<li><a href="#structural-zeros" id="toc-structural-zeros"><span class="toc-section-number">9.1</span> Structural zeros</a></li>
<li><a href="#mags-in-different-locations-and-shared-among-locations" id="toc-mags-in-different-locations-and-shared-among-locations"><span class="toc-section-number">9.2</span> MAGs in different locations and shared among locations</a></li>
<li><a href="#enrichment-analysis-between-male-and-female-ancombc2" id="toc-enrichment-analysis-between-male-and-female-ancombc2"><span class="toc-section-number">9.3</span> Enrichment analysis between male and female: Ancombc2</a>
<ul>
<li><a href="#with-random-effect" id="toc-with-random-effect"><span class="toc-section-number">9.3.1</span> With random effect</a></li>
<li><a href="#without-random-effect" id="toc-without-random-effect"><span class="toc-section-number">9.3.2</span> Without random effect</a></li>
<li><a href="#phylum-level-2" id="toc-phylum-level-2"><span class="toc-section-number">9.3.3</span> Phylum level</a></li>
<li><a href="#family-level-2" id="toc-family-level-2"><span class="toc-section-number">9.3.4</span> Family level</a></li>
<li><a href="#genera-1" id="toc-genera-1"><span class="toc-section-number">9.3.5</span> Genera</a></li>
</ul></li>
</ul></li>
<li><a href="#gut-microbiota-functional-analysis" id="toc-gut-microbiota-functional-analysis"><span class="toc-section-number">10</span> Gut microbiota: functional analysis</a>
<ul>
<li><a href="#mci" id="toc-mci"><span class="toc-section-number">10.1</span> MCI</a></li>
<li><a href="#wilcoxon" id="toc-wilcoxon"><span class="toc-section-number">10.2</span> Wilcoxon</a>
<ul>
<li><a href="#community-elements-differences" id="toc-community-elements-differences"><span class="toc-section-number">10.2.1</span> Community elements differences:</a></li>
<li><a href="#community-functions-differences" id="toc-community-functions-differences"><span class="toc-section-number">10.2.2</span> Community functions differences</a></li>
<li><a href="#community-domains-differences" id="toc-community-domains-differences"><span class="toc-section-number">10.2.3</span> Community domains differences</a></li>
</ul></li>
</ul></li>
</ul>
<!--bookdown:toc2:end-->
      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">AlberdiLab | Fix et al 2025</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<!--bookdown:toc:end-->
<!--bookdown:body:start-->
<div id="libaries" class="section level2 hasAnchor" number="0.1">
<h2 class="hasAnchor"><span class="header-section-number">0.1</span> Libaries<a href="#libaries" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Base</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(R.utils)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">library</span>(tinytable)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">library</span>(rairtable)</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co"># For tree handling</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="fu">library</span>(ape)</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="fu">library</span>(phytools)</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="co"># For plotting</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="fu">library</span>(ggplot2)  <span class="co"># Begin plotting using ggplot2</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="fu">library</span>(ggnewscale)</span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a><span class="fu">library</span>(ggtreeExtra)</span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a><span class="fu">library</span>(ggh4x)</span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a><span class="co"># For statistics</span></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a><span class="fu">library</span>(spaa)</span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a><span class="fu">library</span>(Rtsne)</span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a><span class="fu">library</span>(geiger)</span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a><span class="fu">library</span>(hilldiv2)</span>
<span id="cb1-32"><a href="#cb1-32" tabindex="-1"></a><span class="fu">library</span>(distillR)</span>
<span id="cb1-33"><a href="#cb1-33" tabindex="-1"></a><span class="fu">library</span>(broom.mixed)</span>
<span id="cb1-34"><a href="#cb1-34" tabindex="-1"></a><span class="fu">library</span>(emmeans)</span>
<span id="cb1-35"><a href="#cb1-35" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb1-36"><a href="#cb1-36" tabindex="-1"></a><span class="co">#library(lmerTest)</span></span>
<span id="cb1-37"><a href="#cb1-37" tabindex="-1"></a><span class="fu">library</span>(Hmsc)</span>
<span id="cb1-38"><a href="#cb1-38" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb1-39"><a href="#cb1-39" tabindex="-1"></a><span class="fu">library</span>(lme4)</span>
<span id="cb1-40"><a href="#cb1-40" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb1-41"><a href="#cb1-41" tabindex="-1"></a><span class="fu">library</span>(ANCOMBC)</span>
<span id="cb1-42"><a href="#cb1-42" tabindex="-1"></a></span>
<span id="cb1-43"><a href="#cb1-43" tabindex="-1"></a><span class="co">#map visualtisation</span></span>
<span id="cb1-44"><a href="#cb1-44" tabindex="-1"></a><span class="fu">library</span>(rnaturalearth)</span>
<span id="cb1-45"><a href="#cb1-45" tabindex="-1"></a><span class="fu">library</span>(rnaturalearthdata)</span>
<span id="cb1-46"><a href="#cb1-46" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span></code></pre></div>
<!--chapter:end:index.Rmd-->
</div>
<div id="load-data" class="section level2 hasAnchor" number="0.2">
<h2 class="hasAnchor"><span class="header-section-number">0.2</span> Load data<a href="#load-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Load the original data files outputted by the bioinformatic pipeline.</p>
<div id="sample-metadata" class="section level3 hasAnchor" number="0.2.1">
<h3 class="hasAnchor"><span class="header-section-number">0.2.1</span> Sample metadata<a href="#sample-metadata" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>sample_metadata <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/merged_metadata.tsv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">sample=</span><span class="dv">1</span>)</span></code></pre></div>
</div>
</div>
<div id="load_read_counts" class="section level2 hasAnchor" number="0.3">
<h2 class="hasAnchor"><span class="header-section-number">0.3</span> load_read_counts<a href="#load_read_counts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/DMB0167_counts.tsv.gz&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">genome=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">one_of</span>(<span class="fu">c</span>(<span class="st">&quot;genome&quot;</span>,sample_metadata<span class="sc">$</span>sample)))  <span class="co"># Select specific columns from the data</span></span></code></pre></div>
</div>
<div id="load_genome_metadata" class="section level2 hasAnchor" number="0.4">
<h2 class="hasAnchor"><span class="header-section-number">0.4</span> load_genome_metadata<a href="#load_genome_metadata" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>genome_metadata <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/DMB0167_mag_info.tsv.gz&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">length=</span>mag_size)<span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="fu">semi_join</span>(., read_counts, <span class="at">by =</span> <span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">match</span>(genome,read_counts<span class="sc">$</span>genome))</span></code></pre></div>
</div>
<div id="genome-base-hits" class="section level2 hasAnchor" number="0.5">
<h2 class="hasAnchor"><span class="header-section-number">0.5</span> Genome base hits<a href="#genome-base-hits" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>genome_coverage <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/DMB0167_coverage.tsv.gz&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">genome=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">one_of</span>(<span class="fu">c</span>(<span class="st">&quot;genome&quot;</span>,sample_metadata<span class="sc">$</span>sample)))<span class="sc">%&gt;%</span>   <span class="co"># Select specific columns from the data</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="fu">semi_join</span>(., genome_metadata, <span class="at">by =</span> <span class="st">&quot;genome&quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">match</span>(genome, read_counts<span class="sc">$</span>genome))</span></code></pre></div>
</div>
<div id="genome-tree" class="section level2 hasAnchor" number="0.6">
<h2 class="hasAnchor"><span class="header-section-number">0.6</span> Genome tree<a href="#genome-tree" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>genome_tree <span class="ot">&lt;-</span> <span class="fu">read_tree</span>(<span class="st">&quot;data/DMB0167.tree.gz&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>genome_tree<span class="sc">$</span>tip.label <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(genome_tree<span class="sc">$</span>tip.label,<span class="st">&quot;&#39;&quot;</span>, <span class="st">&quot;&quot;</span>) <span class="co">#remove single quotes in MAG names</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>genome_tree <span class="ot">&lt;-</span> <span class="fu">keep.tip</span>(genome_tree, <span class="at">tip=</span>genome_metadata<span class="sc">$</span>genome) <span class="co"># keep only MAG tips</span></span></code></pre></div>
</div>
<div id="genome-annotations" class="section level2 hasAnchor" number="0.7">
<h2 class="hasAnchor"><span class="header-section-number">0.7</span> Genome annotations<a href="#genome-annotations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Downloading individual annotation files from ERDA using information in Airtable and writing them to a single compressed table takes a while. The following chunk only needs to be run once, to generate the <strong><em>genome_annotations</em></strong> table that is saved in the data directory. Note that the airtable connection requires a personal access token.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">airtable</span>(<span class="st">&quot;MAGs&quot;</span>, <span class="st">&quot;appWbHBNLE6iAsMRV&quot;</span>) <span class="sc">%&gt;%</span> <span class="co">#get base ID from Airtable browser URL</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="fu">read_airtable</span>(., <span class="at">fields =</span> <span class="fu">c</span>(<span class="st">&quot;ID&quot;</span>,<span class="st">&quot;mag_name&quot;</span>,<span class="st">&quot;number_genes&quot;</span>,<span class="st">&quot;anno_url&quot;</span>), <span class="at">id_to_col =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="co">#get 3 columns from MAGs table</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  <span class="fu">filter</span>(mag_name <span class="sc">%in%</span> <span class="fu">paste0</span>(genome_metadata<span class="sc">$</span>genome,<span class="st">&quot;.fa&quot;</span>)) <span class="sc">%&gt;%</span> <span class="co">#filter by MAG name</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  <span class="fu">filter</span>(number_genes <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="co">#genes need to exist</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>  <span class="fu">select</span>(anno_url) <span class="sc">%&gt;%</span> <span class="co">#list MAG annotation urls</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>  <span class="fu">pull</span>() <span class="sc">%&gt;%</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>  <span class="fu">read_tsv</span>() <span class="sc">%&gt;%</span> <span class="co">#load all tables</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gene=</span><span class="dv">1</span>, <span class="at">genome=</span><span class="dv">2</span>, <span class="at">contig=</span><span class="dv">3</span>) <span class="sc">%&gt;%</span> <span class="co">#rename first 3 columns</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>  <span class="fu">write_tsv</span>(<span class="at">file=</span><span class="st">&quot;data/genome_annotations.tsv.xz&quot;</span>) <span class="co">#write to overall compressed file</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>genome_annotations <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/genome_annotations.tsv.xz&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">gene=</span><span class="dv">1</span>, <span class="at">genome=</span><span class="dv">2</span>, <span class="at">contig=</span><span class="dv">3</span>)</span></code></pre></div>
</div>
<div id="filter-reads-by-coverage" class="section level2 hasAnchor" number="0.8">
<h2 class="hasAnchor"><span class="header-section-number">0.8</span> Filter reads by coverage<a href="#filter-reads-by-coverage" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>min_coverage<span class="ot">=</span><span class="fl">0.3</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>read_counts_filt <span class="ot">&lt;-</span> genome_coverage <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">ifelse</span>(. <span class="sc">&gt;</span> min_coverage, <span class="dv">1</span>, <span class="dv">0</span>))) <span class="sc">%&gt;%</span>  <span class="co"># Add or modify columns in the dataset</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="sc">~</span> . <span class="sc">*</span> read_counts[[<span class="fu">cur_column</span>()]]))  <span class="co"># Add or modify columns in the dataset</span></span></code></pre></div>
</div>
<div id="transform-reads-into-genome-counts" class="section level2 hasAnchor" number="0.9">
<h2 class="hasAnchor"><span class="header-section-number">0.9</span> Transform reads into genome counts<a href="#transform-reads-into-genome-counts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>readlength<span class="ot">=</span><span class="dv">150</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>genome_counts <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> . <span class="sc">/</span> (genome_metadata<span class="sc">$</span>length <span class="sc">/</span> readlength) ))  <span class="co"># Add or modify columns in the dataset</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>readlength<span class="ot">=</span><span class="dv">150</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>genome_counts_filt <span class="ot">&lt;-</span> read_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> . <span class="sc">/</span> (genome_metadata<span class="sc">$</span>length <span class="sc">/</span> readlength) ))  <span class="co"># Add or modify columns in the dataset</span></span></code></pre></div>
<div id="distill-annotations-into-gifts" class="section level3 hasAnchor" number="0.9.1">
<h3 class="hasAnchor"><span class="header-section-number">0.9.1</span> Distill annotations into GIFTs<a href="#distill-annotations-into-gifts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a> genome_gifts <span class="ot">&lt;-</span> <span class="fu">distill</span>(genome_annotations,GIFT_db,<span class="at">genomecol=</span><span class="dv">2</span>,<span class="at">annotcol=</span><span class="fu">c</span>(<span class="dv">9</span>,<span class="dv">10</span>,<span class="dv">19</span>), <span class="at">verbosity=</span>F)</span></code></pre></div>
</div>
</div>
<div id="prepare-color-scheme" class="section level2 hasAnchor" number="0.10">
<h2 class="hasAnchor"><span class="header-section-number">0.10</span> Prepare color scheme<a href="#prepare-color-scheme" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>phylum_colors <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>    <span class="fu">right_join</span>(genome_metadata, <span class="at">by=</span><span class="fu">join_by</span>(phylum <span class="sc">==</span> phylum)) <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">match</span>(genome, genome_tree<span class="sc">$</span>tip.label)) <span class="sc">%&gt;%</span></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>    <span class="fu">select</span>(phylum, colors) <span class="sc">%&gt;%</span></span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>    <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>    <span class="fu">arrange</span>(phylum) <span class="sc">%&gt;%</span></span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>    <span class="fu">pull</span>(colors, <span class="at">name=</span>phylum)</span></code></pre></div>
</div>
<div id="wrap-working-objects" class="section level2 hasAnchor" number="0.11">
<h2 class="hasAnchor"><span class="header-section-number">0.11</span> Wrap working objects<a href="#wrap-working-objects" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>All working objects are wrapped into a single Rdata object to facilitate downstream usage.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">save</span>(sample_metadata,</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>     genome_metadata,</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>     read_counts,</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>     genome_counts,</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>     genome_counts_filt,</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>     genome_tree,</span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>     genome_gifts,</span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a>     phylum_colors,</span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a>     <span class="at">file =</span> <span class="st">&quot;data/data.Rdata&quot;</span>)</span></code></pre></div>
<!--chapter:end:01_prepare_data.Rmd-->
</div>
<div id="data-statistics" class="section level1 hasAnchor" number="1">
<h1 class="hasAnchor"><span class="header-section-number">1</span> Data statistics<a href="#data-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/data.Rdata&quot;</span>)</span></code></pre></div>
<p>#Visualisation of geographical location of samples and their respectiv envirotypes</p>
<div id="filter-data" class="section level2 hasAnchor" number="1.1">
<h2 class="hasAnchor"><span class="header-section-number">1.1</span> Filter Data<a href="#filter-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co"># Define extended boundary to include all possum data points</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>lat_min <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fl">45.0</span>  <span class="co"># Slightly smaller than -42.93</span></span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>lat_max <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fl">28.0</span>  <span class="co"># Slightly larger than -30.65</span></span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>lon_min <span class="ot">&lt;-</span> <span class="fl">110.0</span>  <span class="co"># Slightly smaller than 116.47</span></span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>lon_max <span class="ot">&lt;-</span> <span class="fl">150.0</span>  <span class="co"># Slightly larger than 148.28</span></span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a><span class="co"># Full dataset with all points (Tasmania &amp; Australia border)</span></span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>full_possums <span class="ot">&lt;-</span> sample_metadata <span class="sc">%&gt;%</span></span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>  <span class="fu">filter</span>(latitude <span class="sc">&gt;=</span> lat_min <span class="sc">&amp;</span> latitude <span class="sc">&lt;=</span> lat_max, </span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a>         longitude <span class="sc">&gt;=</span> lon_min <span class="sc">&amp;</span> longitude <span class="sc">&lt;=</span> lon_max)</span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a></span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a><span class="co"># Filter for Tasmania only</span></span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>tasmania_possums <span class="ot">&lt;-</span> full_possums <span class="sc">%&gt;%</span></span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a>  <span class="fu">filter</span>(latitude <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">38</span> <span class="sc">&amp;</span> latitude <span class="sc">&gt;</span> <span class="sc">-</span><span class="dv">45</span>,  <span class="co"># Tasmania lat range</span></span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a>         longitude <span class="sc">&gt;</span> <span class="dv">140</span> <span class="sc">&amp;</span> longitude <span class="sc">&lt;</span> <span class="dv">150</span>)  <span class="co"># Tasmania lon range</span></span>
<span id="cb16-16"><a href="#cb16-16" tabindex="-1"></a></span>
<span id="cb16-17"><a href="#cb16-17" tabindex="-1"></a><span class="co"># Aggregate possum count per location</span></span>
<span id="cb16-18"><a href="#cb16-18" tabindex="-1"></a>full_possums_count <span class="ot">&lt;-</span> full_possums <span class="sc">%&gt;%</span></span>
<span id="cb16-19"><a href="#cb16-19" tabindex="-1"></a>  <span class="fu">group_by</span>(latitude, longitude, broad_environment) <span class="sc">%&gt;%</span></span>
<span id="cb16-20"><a href="#cb16-20" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>)</span>
<span id="cb16-21"><a href="#cb16-21" tabindex="-1"></a></span>
<span id="cb16-22"><a href="#cb16-22" tabindex="-1"></a>tasmania_possums_count <span class="ot">&lt;-</span> tasmania_possums <span class="sc">%&gt;%</span></span>
<span id="cb16-23"><a href="#cb16-23" tabindex="-1"></a>  <span class="fu">group_by</span>(latitude, longitude, broad_environment) <span class="sc">%&gt;%</span></span>
<span id="cb16-24"><a href="#cb16-24" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>)</span></code></pre></div>
</div>
<div id="load-map" class="section level2 hasAnchor" number="1.2">
<h2 class="hasAnchor"><span class="header-section-number">1.2</span> Load Map<a href="#load-map" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># Load world map</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>australia_map <span class="ot">&lt;-</span> <span class="fu">ne_countries</span>(<span class="at">scale =</span> <span class="st">&quot;medium&quot;</span>, <span class="at">returnclass =</span> <span class="st">&quot;sf&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>  <span class="fu">filter</span>(admin <span class="sc">==</span> <span class="st">&quot;Australia&quot;</span>)</span></code></pre></div>
</div>
<div id="plot-possum-locations" class="section level2 hasAnchor" number="1.3">
<h2 class="hasAnchor"><span class="header-section-number">1.3</span> Plot Possum Locations<a href="#plot-possum-locations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="co"># Get Tasmania data range to set zoom</span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>lat_range <span class="ot">&lt;-</span> <span class="fu">range</span>(tasmania_possums<span class="sc">$</span>latitude, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>lon_range <span class="ot">&lt;-</span> <span class="fu">range</span>(tasmania_possums<span class="sc">$</span>longitude, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a><span class="co"># Count total possums</span></span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>total_full <span class="ot">&lt;-</span> <span class="fu">sum</span>(full_possums_count<span class="sc">$</span>count)</span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a>total_tasmania <span class="ot">&lt;-</span> <span class="fu">sum</span>(tasmania_possums_count<span class="sc">$</span>count)</span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a><span class="co"># Plot 1: Full Dataset (Including Border Data)</span></span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a>map_plot_full <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> australia_map, <span class="at">fill =</span> <span class="st">&quot;lightgray&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span>  <span class="co"># Base map</span></span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> full_possums_count, <span class="fu">aes</span>(<span class="at">x =</span> longitude, <span class="at">y =</span> latitude, </span>
<span id="cb18-13"><a href="#cb18-13" tabindex="-1"></a>                                            <span class="at">color =</span> broad_environment, <span class="at">size =</span> count), </span>
<span id="cb18-14"><a href="#cb18-14" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb18-15"><a href="#cb18-15" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb18-16"><a href="#cb18-16" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">&quot;PD - Full Dataset (&quot;</span>, total_full, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb18-17"><a href="#cb18-17" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Longitude&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Latitude&quot;</span>, <span class="at">color =</span> <span class="st">&quot;Environment Type&quot;</span>, <span class="at">size =</span> <span class="st">&quot;Count&quot;</span>, <span class="at">shape =</span> <span class="st">&quot;species&quot;</span>) <span class="sc">+</span></span>
<span id="cb18-18"><a href="#cb18-18" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb18-19"><a href="#cb18-19" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>))</span>
<span id="cb18-20"><a href="#cb18-20" tabindex="-1"></a></span>
<span id="cb18-21"><a href="#cb18-21" tabindex="-1"></a><span class="co"># Plot 2: Zoomed-in Tasmania (with entire landmass and cleaned longitude labels)</span></span>
<span id="cb18-22"><a href="#cb18-22" tabindex="-1"></a>map_plot_zoomed <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb18-23"><a href="#cb18-23" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> australia_map, <span class="at">fill =</span> <span class="st">&quot;lightgray&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span>  <span class="co"># Base map</span></span>
<span id="cb18-24"><a href="#cb18-24" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> tasmania_possums_count, <span class="fu">aes</span>(<span class="at">x =</span> longitude, <span class="at">y =</span> latitude, </span>
<span id="cb18-25"><a href="#cb18-25" tabindex="-1"></a>                                                <span class="at">color =</span> broad_environment, <span class="at">size =</span> count), </span>
<span id="cb18-26"><a href="#cb18-26" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span> </span>
<span id="cb18-27"><a href="#cb18-27" tabindex="-1"></a>  <span class="fu">coord_sf</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">143</span>, <span class="dv">149</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">44</span>, <span class="sc">-</span><span class="dv">39</span>)) <span class="sc">+</span>  <span class="co"># Full Tasmania view</span></span>
<span id="cb18-28"><a href="#cb18-28" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb18-29"><a href="#cb18-29" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">&quot;PD - Tasmania (&quot;</span>, total_tasmania, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb18-30"><a href="#cb18-30" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Longitude&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Latitude&quot;</span>,</span>
<span id="cb18-31"><a href="#cb18-31" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">&quot;Environment Type&quot;</span>, <span class="at">size =</span> <span class="st">&quot;Count&quot;</span>, <span class="at">shape =</span> <span class="st">&quot;species&quot;</span> ) <span class="sc">+</span></span>
<span id="cb18-32"><a href="#cb18-32" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>, </span>
<span id="cb18-33"><a href="#cb18-33" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb18-34"><a href="#cb18-34" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>))</span>
<span id="cb18-35"><a href="#cb18-35" tabindex="-1"></a></span>
<span id="cb18-36"><a href="#cb18-36" tabindex="-1"></a><span class="co"># Combine the plots side by side with shared legend and increased size</span></span>
<span id="cb18-37"><a href="#cb18-37" tabindex="-1"></a>(map_plot_full <span class="sc">+</span> map_plot_zoomed) <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">&quot;collect&quot;</span>, <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.2</span>))</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/plot_maps-1.png" width="672" /></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="co"># Define extended boundary to include all possum data points</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>lat_min <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fl">45.0</span>  <span class="co"># Slightly smaller than -42.93</span></span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>lat_max <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fl">28.0</span>  <span class="co"># Slightly larger than -30.65</span></span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>lon_min <span class="ot">&lt;-</span> <span class="fl">110.0</span>  <span class="co"># Slightly smaller than 116.47</span></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>lon_max <span class="ot">&lt;-</span> <span class="fl">150.0</span>  <span class="co"># Slightly larger than 148.28</span></span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a><span class="co"># Full dataset with all points (Tasmania &amp; Australia border)</span></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a>full_possums <span class="ot">&lt;-</span> sample_metadata <span class="sc">%&gt;%</span></span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>  <span class="fu">filter</span>(latitude <span class="sc">&gt;=</span> lat_min <span class="sc">&amp;</span> latitude <span class="sc">&lt;=</span> lat_max, </span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a>         longitude <span class="sc">&gt;=</span> lon_min <span class="sc">&amp;</span> longitude <span class="sc">&lt;=</span> lon_max)</span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a></span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a><span class="co"># Filter for Tasmania only</span></span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a>tasmania_possums <span class="ot">&lt;-</span> full_possums <span class="sc">%&gt;%</span></span>
<span id="cb19-14"><a href="#cb19-14" tabindex="-1"></a>  <span class="fu">filter</span>(latitude <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">38</span> <span class="sc">&amp;</span> latitude <span class="sc">&gt;</span> <span class="sc">-</span><span class="dv">45</span>,  <span class="co"># Tasmania lat range</span></span>
<span id="cb19-15"><a href="#cb19-15" tabindex="-1"></a>         longitude <span class="sc">&gt;</span> <span class="dv">140</span> <span class="sc">&amp;</span> longitude <span class="sc">&lt;</span> <span class="dv">150</span>)  <span class="co"># Tasmania lon range</span></span>
<span id="cb19-16"><a href="#cb19-16" tabindex="-1"></a></span>
<span id="cb19-17"><a href="#cb19-17" tabindex="-1"></a><span class="co"># Aggregate possum count per location</span></span>
<span id="cb19-18"><a href="#cb19-18" tabindex="-1"></a>full_possums_count <span class="ot">&lt;-</span> full_possums <span class="sc">%&gt;%</span></span>
<span id="cb19-19"><a href="#cb19-19" tabindex="-1"></a>  <span class="fu">group_by</span>(latitude, longitude, local_environment) <span class="sc">%&gt;%</span></span>
<span id="cb19-20"><a href="#cb19-20" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>)</span>
<span id="cb19-21"><a href="#cb19-21" tabindex="-1"></a></span>
<span id="cb19-22"><a href="#cb19-22" tabindex="-1"></a>tasmania_possums_count <span class="ot">&lt;-</span> tasmania_possums <span class="sc">%&gt;%</span></span>
<span id="cb19-23"><a href="#cb19-23" tabindex="-1"></a>  <span class="fu">group_by</span>(latitude, longitude, local_environment) <span class="sc">%&gt;%</span></span>
<span id="cb19-24"><a href="#cb19-24" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>)</span></code></pre></div>
</div>
<div id="load-map-1" class="section level2 hasAnchor" number="1.4">
<h2 class="hasAnchor"><span class="header-section-number">1.4</span> Load Map<a href="#load-map-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="co"># Load world map</span></span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>australia_map <span class="ot">&lt;-</span> <span class="fu">ne_countries</span>(<span class="at">scale =</span> <span class="st">&quot;medium&quot;</span>, <span class="at">returnclass =</span> <span class="st">&quot;sf&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>  <span class="fu">filter</span>(admin <span class="sc">==</span> <span class="st">&quot;Australia&quot;</span>)</span></code></pre></div>
</div>
<div id="plot-possum-locations-1" class="section level2 hasAnchor" number="1.5">
<h2 class="hasAnchor"><span class="header-section-number">1.5</span> Plot Possum Locations<a href="#plot-possum-locations-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a><span class="co"># Get Tasmania data range to set zoom</span></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>lat_range <span class="ot">&lt;-</span> <span class="fu">range</span>(tasmania_possums<span class="sc">$</span>latitude, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a>lon_range <span class="ot">&lt;-</span> <span class="fu">range</span>(tasmania_possums<span class="sc">$</span>longitude, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a><span class="co"># Count total possums</span></span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a>total_full <span class="ot">&lt;-</span> <span class="fu">sum</span>(full_possums_count<span class="sc">$</span>count)</span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>total_tasmania <span class="ot">&lt;-</span> <span class="fu">sum</span>(tasmania_possums_count<span class="sc">$</span>count)</span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a></span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a><span class="co"># Plot 1: Full Dataset (Including Border Data)</span></span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a>map_plot_full <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb21-13"><a href="#cb21-13" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> australia_map, <span class="at">fill =</span> <span class="st">&quot;lightgray&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb21-14"><a href="#cb21-14" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> full_possums_count, <span class="fu">aes</span>(<span class="at">x =</span> longitude, <span class="at">y =</span> latitude,</span>
<span id="cb21-15"><a href="#cb21-15" tabindex="-1"></a>                                            <span class="at">color =</span> local_environment, <span class="at">size =</span> count),</span>
<span id="cb21-16"><a href="#cb21-16" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb21-17"><a href="#cb21-17" tabindex="-1"></a>  <span class="fu">scale_color_viridis_d</span>(<span class="at">name =</span> <span class="st">&quot;Environment Type&quot;</span>, <span class="at">option =</span> <span class="st">&quot;D&quot;</span>) <span class="sc">+</span></span>
<span id="cb21-18"><a href="#cb21-18" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb21-19"><a href="#cb21-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">&quot;PD - Full Dataset (&quot;</span>, total_full, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb21-20"><a href="#cb21-20" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Longitude&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Latitude&quot;</span>) <span class="sc">+</span></span>
<span id="cb21-21"><a href="#cb21-21" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb21-22"><a href="#cb21-22" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb21-23"><a href="#cb21-23" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb21-24"><a href="#cb21-24" tabindex="-1"></a>  )</span>
<span id="cb21-25"><a href="#cb21-25" tabindex="-1"></a></span>
<span id="cb21-26"><a href="#cb21-26" tabindex="-1"></a><span class="co"># Plot 2: Zoomed-in Tasmania</span></span>
<span id="cb21-27"><a href="#cb21-27" tabindex="-1"></a>map_plot_zoomed <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb21-28"><a href="#cb21-28" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> australia_map, <span class="at">fill =</span> <span class="st">&quot;lightgray&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb21-29"><a href="#cb21-29" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> tasmania_possums_count, <span class="fu">aes</span>(<span class="at">x =</span> longitude, <span class="at">y =</span> latitude,</span>
<span id="cb21-30"><a href="#cb21-30" tabindex="-1"></a>                                                <span class="at">color =</span> local_environment, <span class="at">size =</span> count),</span>
<span id="cb21-31"><a href="#cb21-31" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb21-32"><a href="#cb21-32" tabindex="-1"></a>  <span class="fu">coord_sf</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">143</span>, <span class="dv">149</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">44</span>, <span class="sc">-</span><span class="dv">39</span>)) <span class="sc">+</span></span>
<span id="cb21-33"><a href="#cb21-33" tabindex="-1"></a>  <span class="fu">scale_color_viridis_d</span>(<span class="at">name =</span> <span class="st">&quot;Environment Type&quot;</span>, <span class="at">option =</span> <span class="st">&quot;D&quot;</span>) <span class="sc">+</span></span>
<span id="cb21-34"><a href="#cb21-34" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb21-35"><a href="#cb21-35" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">&quot;PD - Tasmania (&quot;</span>, total_tasmania, <span class="st">&quot;)&quot;</span>),</span>
<span id="cb21-36"><a href="#cb21-36" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Longitude&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Latitude&quot;</span>) <span class="sc">+</span></span>
<span id="cb21-37"><a href="#cb21-37" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb21-38"><a href="#cb21-38" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb21-39"><a href="#cb21-39" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb21-40"><a href="#cb21-40" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb21-41"><a href="#cb21-41" tabindex="-1"></a>  )</span>
<span id="cb21-42"><a href="#cb21-42" tabindex="-1"></a></span>
<span id="cb21-43"><a href="#cb21-43" tabindex="-1"></a><span class="co"># Combine the plots side by side</span></span>
<span id="cb21-44"><a href="#cb21-44" tabindex="-1"></a>(map_plot_full <span class="sc">+</span> map_plot_zoomed) <span class="sc">+</span></span>
<span id="cb21-45"><a href="#cb21-45" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.2</span>))</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/plot_maps2-1.png" width="672" /></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="co"># Step 1: Create simplified/aggregated legend data</span></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>legend_data <span class="ot">&lt;-</span> tasmania_possums_count <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>  <span class="fu">group_by</span>(local_environment) <span class="sc">%&gt;%</span></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean_count =</span> <span class="fu">mean</span>(count, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(local_environment)) <span class="sc">%&gt;%</span></span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(mean_count)) <span class="sc">%&gt;%</span></span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">env_label =</span> <span class="fu">paste0</span>(local_environment))</span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a><span class="co"># Step 2: Plot custom &quot;legend&quot; with clean layout</span></span>
<span id="cb22-10"><a href="#cb22-10" tabindex="-1"></a>custom_legend_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(legend_data, <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span>, <span class="at">y =</span> <span class="fu">reorder</span>(env_label, mean_count),</span>
<span id="cb22-11"><a href="#cb22-11" tabindex="-1"></a>                                              <span class="at">color =</span> env_label, <span class="at">size =</span> mean_count)) <span class="sc">+</span></span>
<span id="cb22-12"><a href="#cb22-12" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb22-13"><a href="#cb22-13" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">6</span>)) <span class="sc">+</span>  <span class="co"># tweak dot size scaling</span></span>
<span id="cb22-14"><a href="#cb22-14" tabindex="-1"></a>  <span class="fu">scale_color_viridis_d</span>(<span class="at">name =</span> <span class="st">&quot;Environment Type&quot;</span>, <span class="at">option =</span> <span class="st">&quot;D&quot;</span>) <span class="sc">+</span></span>
<span id="cb22-15"><a href="#cb22-15" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb22-16"><a href="#cb22-16" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb22-17"><a href="#cb22-17" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-18"><a href="#cb22-18" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-19"><a href="#cb22-19" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-20"><a href="#cb22-20" tabindex="-1"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-21"><a href="#cb22-21" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-22"><a href="#cb22-22" tabindex="-1"></a>    <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">10</span>, <span class="dv">30</span>, <span class="dv">10</span>, <span class="dv">10</span>),</span>
<span id="cb22-23"><a href="#cb22-23" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>, <span class="at">hjust =</span> <span class="dv">0</span>)</span>
<span id="cb22-24"><a href="#cb22-24" tabindex="-1"></a>  )</span>
<span id="cb22-25"><a href="#cb22-25" tabindex="-1"></a></span>
<span id="cb22-26"><a href="#cb22-26" tabindex="-1"></a><span class="co"># Show it</span></span>
<span id="cb22-27"><a href="#cb22-27" tabindex="-1"></a>custom_legend_plot</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/combined_legend-1.png" width="672" /></p>
</div>
<div id="sequencing-reads-statistics" class="section level2 hasAnchor" number="1.6">
<h2 class="hasAnchor"><span class="header-section-number">1.6</span> Sequencing reads statistics<a href="#sequencing-reads-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>sample_metadata <span class="sc">%&gt;%</span> </span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">Total=</span><span class="fu">sum</span>(reads_post_fastp <span class="sc">*</span> <span class="dv">150</span> <span class="sc">/</span> <span class="dv">1000000000</span>) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>), </span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>              <span class="at">mean=</span><span class="fu">mean</span>(reads_post_fastp <span class="sc">*</span> <span class="dv">150</span> <span class="sc">/</span> <span class="dv">1000000000</span>) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>),</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>              <span class="at">sd=</span><span class="fu">sd</span>(reads_post_fastp <span class="sc">*</span> <span class="dv">150</span> <span class="sc">/</span> <span class="dv">1000000000</span>) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a>    <span class="fu">unite</span>(<span class="st">&quot;Average&quot;</span>,mean, sd, <span class="at">sep =</span> <span class="st">&quot; ± &quot;</span>, <span class="at">remove =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>    <span class="fu">tt</span>()</span></code></pre></div>
<!-- preamble start -->

    <script>

      function styleCell_v4g84v85l40zzfd9l2n2(i, j, css_id) {
          var table = document.getElementById("tinytable_v4g84v85l40zzfd9l2n2");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_v4g84v85l40zzfd9l2n2');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_v4g84v85l40zzfd9l2n2(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_v4g84v85l40zzfd9l2n2");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 },  ], css_id: 'tinytable_css_k7b40zlodzxv7azagnhz',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 },  ], css_id: 'tinytable_css_fv6cwoec6rbjtcv3dhu8',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_v4g84v85l40zzfd9l2n2(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_k7b40zlodzxv7azagnhz, .table th.tinytable_css_k7b40zlodzxv7azagnhz { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_fv6cwoec6rbjtcv3dhu8, .table th.tinytable_css_fv6cwoec6rbjtcv3dhu8 { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_v4g84v85l40zzfd9l2n2" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing='true'>
        <thead>
        
              <tr>
                <th scope="col">Total</th>
                <th scope="col">Average</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>354.78</td>
                  <td>4.86 ± 1.43</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
<div id="dna-fractions" class="section level2 hasAnchor" number="1.7">
<h2 class="hasAnchor"><span class="header-section-number">1.7</span> DNA fractions<a href="#dna-fractions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>sequence_fractions <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>genome, <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mags =</span> <span class="fu">sum</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a>    <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a>    <span class="fu">select</span>(sample,mags,metagenomic_bases,host_bases,bases_lost_fastp_percent) <span class="sc">%&gt;%</span></span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mags_bases =</span> mags<span class="sc">*</span><span class="dv">146</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">lowqual_bases =</span> ((metagenomic_bases<span class="sc">+</span>host_bases)<span class="sc">/</span>(<span class="dv">1</span><span class="sc">-</span>bases_lost_fastp_percent))<span class="sc">-</span>(metagenomic_bases<span class="sc">+</span>host_bases)) <span class="sc">%&gt;%</span></span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">unmapped_bases =</span> metagenomic_bases <span class="sc">-</span> mags_bases) <span class="sc">%&gt;%</span></span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">unmapped_bases =</span> <span class="fu">ifelse</span>(unmapped_bases <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="dv">0</span>, unmapped_bases)) <span class="sc">%&gt;%</span></span>
<span id="cb24-11"><a href="#cb24-11" tabindex="-1"></a>    <span class="fu">select</span>(sample, lowqual_bases, host_bases, unmapped_bases, mags_bases)</span>
<span id="cb24-12"><a href="#cb24-12" tabindex="-1"></a></span>
<span id="cb24-13"><a href="#cb24-13" tabindex="-1"></a>sequence_fractions <span class="sc">%&gt;%</span></span>
<span id="cb24-14"><a href="#cb24-14" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>sample), <span class="sc">~</span>.<span class="sc">/</span><span class="dv">1000000000</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-15"><a href="#cb24-15" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;Sample&quot;</span><span class="ot">=</span><span class="dv">1</span>, <span class="st">&quot;Low quality&quot;</span><span class="ot">=</span><span class="dv">2</span>, <span class="st">&quot;Mapped to host&quot;</span><span class="ot">=</span><span class="dv">3</span>, <span class="st">&quot;Unmapped&quot;</span><span class="ot">=</span><span class="dv">4</span>, <span class="st">&quot;Mapped to MAGs&quot;</span><span class="ot">=</span><span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-16"><a href="#cb24-16" tabindex="-1"></a>  <span class="fu">tt</span>()</span></code></pre></div>
<!-- preamble start -->

    <script>

      function styleCell_pa4euwxsx778yyi08hig(i, j, css_id) {
          var table = document.getElementById("tinytable_pa4euwxsx778yyi08hig");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_pa4euwxsx778yyi08hig');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_pa4euwxsx778yyi08hig(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_pa4euwxsx778yyi08hig");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 73, j: 0 }, { i: 73, j: 1 }, { i: 73, j: 2 }, { i: 73, j: 3 }, { i: 73, j: 4 },  ], css_id: 'tinytable_css_zr1v2viipj24vhsr1fm3',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 }, { i: 0, j: 4 },  ], css_id: 'tinytable_css_5xzl0z0ipzoz03moqgss',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_pa4euwxsx778yyi08hig(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_zr1v2viipj24vhsr1fm3, .table th.tinytable_css_zr1v2viipj24vhsr1fm3 { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_5xzl0z0ipzoz03moqgss, .table th.tinytable_css_5xzl0z0ipzoz03moqgss { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_pa4euwxsx778yyi08hig" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing='true'>
        <thead>
        
              <tr>
                <th scope="col">Sample</th>
                <th scope="col">Low quality</th>
                <th scope="col">Mapped to host</th>
                <th scope="col">Unmapped</th>
                <th scope="col">Mapped to MAGs</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>EHI01248</td>
                  <td>0.35511316</td>
                  <td>0.004968189</td>
                  <td>2.3488847</td>
                  <td>8.70559471</td>
                </tr>
                <tr>
                  <td>EHI01253</td>
                  <td>0.15065275</td>
                  <td>0.007306842</td>
                  <td>0.7958509</td>
                  <td>3.15690325</td>
                </tr>
                <tr>
                  <td>EHI01254</td>
                  <td>0.17713763</td>
                  <td>0.008641986</td>
                  <td>0.7698065</td>
                  <td>3.46426201</td>
                </tr>
                <tr>
                  <td>EHI01256</td>
                  <td>0.25601713</td>
                  <td>0.011186636</td>
                  <td>1.6153318</td>
                  <td>4.62283158</td>
                </tr>
                <tr>
                  <td>EHI01257</td>
                  <td>0.44278829</td>
                  <td>0.028219625</td>
                  <td>1.6424801</td>
                  <td>6.37891199</td>
                </tr>
                <tr>
                  <td>EHI01258</td>
                  <td>0.18025206</td>
                  <td>0.021893733</td>
                  <td>0.9061351</td>
                  <td>3.11043101</td>
                </tr>
                <tr>
                  <td>EHI01259</td>
                  <td>0.15785294</td>
                  <td>0.002551853</td>
                  <td>1.1230441</td>
                  <td>3.63491878</td>
                </tr>
                <tr>
                  <td>EHI01264</td>
                  <td>0.10980388</td>
                  <td>0.194713827</td>
                  <td>0.8459937</td>
                  <td>1.94399131</td>
                </tr>
                <tr>
                  <td>EHI01265</td>
                  <td>0.20096759</td>
                  <td>0.027838593</td>
                  <td>0.8781951</td>
                  <td>4.42282487</td>
                </tr>
                <tr>
                  <td>EHI01266</td>
                  <td>0.16495605</td>
                  <td>0.004145198</td>
                  <td>1.0104139</td>
                  <td>3.02528367</td>
                </tr>
                <tr>
                  <td>EHI01268</td>
                  <td>0.11654687</td>
                  <td>0.043538943</td>
                  <td>0.4199424</td>
                  <td>2.18818215</td>
                </tr>
                <tr>
                  <td>EHI01270</td>
                  <td>0.15490780</td>
                  <td>0.006255568</td>
                  <td>0.7491831</td>
                  <td>1.85948097</td>
                </tr>
                <tr>
                  <td>EHI01271</td>
                  <td>0.08546883</td>
                  <td>0.008773597</td>
                  <td>0.5176363</td>
                  <td>1.98955149</td>
                </tr>
                <tr>
                  <td>EHI01272</td>
                  <td>0.12521297</td>
                  <td>0.004006385</td>
                  <td>0.6499615</td>
                  <td>2.73090343</td>
                </tr>
                <tr>
                  <td>EHI01273</td>
                  <td>0.18374616</td>
                  <td>0.099650655</td>
                  <td>1.1683174</td>
                  <td>3.70517179</td>
                </tr>
                <tr>
                  <td>EHI01339</td>
                  <td>0.21133261</td>
                  <td>0.082072269</td>
                  <td>1.0958456</td>
                  <td>4.01574329</td>
                </tr>
                <tr>
                  <td>EHI01340</td>
                  <td>0.32934907</td>
                  <td>0.078517930</td>
                  <td>2.2365738</td>
                  <td>0.01091408</td>
                </tr>
                <tr>
                  <td>EHI01341</td>
                  <td>0.26571726</td>
                  <td>0.023522716</td>
                  <td>1.3209604</td>
                  <td>3.21859599</td>
                </tr>
                <tr>
                  <td>EHI01342</td>
                  <td>0.21296031</td>
                  <td>0.114741396</td>
                  <td>1.6557153</td>
                  <td>3.12134276</td>
                </tr>
                <tr>
                  <td>EHI01343</td>
                  <td>0.18906343</td>
                  <td>0.112255644</td>
                  <td>1.1418032</td>
                  <td>3.97856774</td>
                </tr>
                <tr>
                  <td>EHI01345</td>
                  <td>0.20874962</td>
                  <td>0.177845762</td>
                  <td>1.5032861</td>
                  <td>3.47998475</td>
                </tr>
                <tr>
                  <td>EHI01387</td>
                  <td>0.23576644</td>
                  <td>4.644537522</td>
                  <td>0.4353141</td>
                  <td>1.57377576</td>
                </tr>
                <tr>
                  <td>EHI01389</td>
                  <td>0.22833259</td>
                  <td>0.150555697</td>
                  <td>1.5157673</td>
                  <td>4.06490864</td>
                </tr>
                <tr>
                  <td>EHI02064</td>
                  <td>0.36180367</td>
                  <td>0.073958684</td>
                  <td>1.7560470</td>
                  <td>3.13915797</td>
                </tr>
                <tr>
                  <td>EHI02074</td>
                  <td>0.17452029</td>
                  <td>0.007193557</td>
                  <td>0.5946409</td>
                  <td>1.57287231</td>
                </tr>
                <tr>
                  <td>EHI02082</td>
                  <td>0.39485375</td>
                  <td>0.094697133</td>
                  <td>1.6620324</td>
                  <td>4.43909934</td>
                </tr>
                <tr>
                  <td>EHI02098</td>
                  <td>0.52162812</td>
                  <td>0.139932802</td>
                  <td>1.6241102</td>
                  <td>3.79457081</td>
                </tr>
                <tr>
                  <td>EHI02760</td>
                  <td>0.27278081</td>
                  <td>0.117175084</td>
                  <td>0.5856983</td>
                  <td>2.76896198</td>
                </tr>
                <tr>
                  <td>EHI02766</td>
                  <td>0.34206992</td>
                  <td>0.026262224</td>
                  <td>1.0131013</td>
                  <td>3.38893828</td>
                </tr>
                <tr>
                  <td>EHI02767</td>
                  <td>0.39776392</td>
                  <td>0.027589923</td>
                  <td>1.4920663</td>
                  <td>4.50953237</td>
                </tr>
                <tr>
                  <td>EHI02768</td>
                  <td>0.37285123</td>
                  <td>0.057921344</td>
                  <td>0.6534089</td>
                  <td>2.85662184</td>
                </tr>
                <tr>
                  <td>EHI02769</td>
                  <td>0.23901114</td>
                  <td>0.015625837</td>
                  <td>0.5237186</td>
                  <td>1.80651100</td>
                </tr>
                <tr>
                  <td>EHI02770</td>
                  <td>0.26006960</td>
                  <td>0.015636651</td>
                  <td>0.8099749</td>
                  <td>2.58298046</td>
                </tr>
                <tr>
                  <td>EHI02771</td>
                  <td>0.43362442</td>
                  <td>0.078674560</td>
                  <td>0.8927819</td>
                  <td>3.75367475</td>
                </tr>
                <tr>
                  <td>EHI02772</td>
                  <td>0.39350819</td>
                  <td>0.133333189</td>
                  <td>0.9955418</td>
                  <td>3.97147448</td>
                </tr>
                <tr>
                  <td>EHI02773</td>
                  <td>0.47175703</td>
                  <td>0.004630464</td>
                  <td>1.2818831</td>
                  <td>4.30891829</td>
                </tr>
                <tr>
                  <td>EHI02774</td>
                  <td>0.35219413</td>
                  <td>0.031638370</td>
                  <td>0.9683903</td>
                  <td>3.44730645</td>
                </tr>
                <tr>
                  <td>EHI02775</td>
                  <td>0.31535990</td>
                  <td>0.006253892</td>
                  <td>1.2347646</td>
                  <td>3.06181082</td>
                </tr>
                <tr>
                  <td>EHI02776</td>
                  <td>0.28106302</td>
                  <td>0.008270565</td>
                  <td>0.8075659</td>
                  <td>3.52331025</td>
                </tr>
                <tr>
                  <td>EHI02779</td>
                  <td>0.34735466</td>
                  <td>0.029550201</td>
                  <td>0.7625330</td>
                  <td>2.52847910</td>
                </tr>
                <tr>
                  <td>EHI02780</td>
                  <td>0.33353488</td>
                  <td>0.041239947</td>
                  <td>1.1610012</td>
                  <td>3.29078029</td>
                </tr>
                <tr>
                  <td>EHI02784</td>
                  <td>0.31121206</td>
                  <td>0.019685481</td>
                  <td>0.8413476</td>
                  <td>3.58321420</td>
                </tr>
                <tr>
                  <td>EHI02785</td>
                  <td>0.32741021</td>
                  <td>0.014638539</td>
                  <td>0.9515597</td>
                  <td>3.36888357</td>
                </tr>
                <tr>
                  <td>EHI02789</td>
                  <td>0.42120190</td>
                  <td>0.170396074</td>
                  <td>1.0281381</td>
                  <td>4.40650732</td>
                </tr>
                <tr>
                  <td>EHI02792</td>
                  <td>0.36123108</td>
                  <td>0.015462248</td>
                  <td>1.3463286</td>
                  <td>3.76356741</td>
                </tr>
                <tr>
                  <td>EHI02796</td>
                  <td>1.07935523</td>
                  <td>0.004050961</td>
                  <td>1.0416445</td>
                  <td>4.15780523</td>
                </tr>
                <tr>
                  <td>EHI02797</td>
                  <td>0.41305976</td>
                  <td>0.086506413</td>
                  <td>0.8415216</td>
                  <td>3.53097613</td>
                </tr>
                <tr>
                  <td>EHI02801</td>
                  <td>0.52781243</td>
                  <td>0.036344570</td>
                  <td>0.8388848</td>
                  <td>2.91079981</td>
                </tr>
                <tr>
                  <td>EHI02804</td>
                  <td>0.42394759</td>
                  <td>0.159371043</td>
                  <td>1.0649697</td>
                  <td>4.43245152</td>
                </tr>
                <tr>
                  <td>EHI02806</td>
                  <td>0.44246621</td>
                  <td>0.089581175</td>
                  <td>1.7887921</td>
                  <td>4.03903963</td>
                </tr>
                <tr>
                  <td>EHI02811</td>
                  <td>0.29358681</td>
                  <td>0.013750538</td>
                  <td>1.1729986</td>
                  <td>3.28550574</td>
                </tr>
                <tr>
                  <td>EHI02812</td>
                  <td>0.36744061</td>
                  <td>0.057112125</td>
                  <td>0.8678336</td>
                  <td>4.12914616</td>
                </tr>
                <tr>
                  <td>EHI02817</td>
                  <td>0.34058911</td>
                  <td>0.023035237</td>
                  <td>0.8059566</td>
                  <td>2.90095167</td>
                </tr>
                <tr>
                  <td>EHI02819</td>
                  <td>0.23782849</td>
                  <td>0.002853452</td>
                  <td>2.0747359</td>
                  <td>1.56371986</td>
                </tr>
                <tr>
                  <td>EHI02822</td>
                  <td>0.32932578</td>
                  <td>0.266269016</td>
                  <td>0.8566608</td>
                  <td>2.91142265</td>
                </tr>
                <tr>
                  <td>EHI02823</td>
                  <td>0.46831299</td>
                  <td>0.319368041</td>
                  <td>0.9890243</td>
                  <td>4.23615481</td>
                </tr>
                <tr>
                  <td>EHI02824</td>
                  <td>0.92746482</td>
                  <td>0.011070271</td>
                  <td>0.9754646</td>
                  <td>2.92498444</td>
                </tr>
                <tr>
                  <td>EHI02826</td>
                  <td>0.44208850</td>
                  <td>0.043681080</td>
                  <td>0.9461530</td>
                  <td>4.12595664</td>
                </tr>
                <tr>
                  <td>EHI02829</td>
                  <td>0.61719061</td>
                  <td>0.020333202</td>
                  <td>1.0838773</td>
                  <td>5.08522117</td>
                </tr>
                <tr>
                  <td>EHI02830</td>
                  <td>0.63356691</td>
                  <td>1.608315016</td>
                  <td>0.8548023</td>
                  <td>3.32875927</td>
                </tr>
                <tr>
                  <td>EHI02831</td>
                  <td>0.48579678</td>
                  <td>0.249927066</td>
                  <td>1.0747457</td>
                  <td>3.71679223</td>
                </tr>
                <tr>
                  <td>EHI02832</td>
                  <td>0.55980508</td>
                  <td>0.002363983</td>
                  <td>1.1179965</td>
                  <td>3.46662093</td>
                </tr>
                <tr>
                  <td>EHI02835</td>
                  <td>0.36642675</td>
                  <td>0.046904442</td>
                  <td>1.3031776</td>
                  <td>4.02345778</td>
                </tr>
                <tr>
                  <td>EHI02836</td>
                  <td>0.73282342</td>
                  <td>0.011061548</td>
                  <td>0.7145331</td>
                  <td>2.22777765</td>
                </tr>
                <tr>
                  <td>EHI02837</td>
                  <td>0.63165003</td>
                  <td>0.233233891</td>
                  <td>1.2696173</td>
                  <td>4.08635940</td>
                </tr>
                <tr>
                  <td>EHI02838</td>
                  <td>0.71576136</td>
                  <td>0.008130997</td>
                  <td>1.6067697</td>
                  <td>4.58593636</td>
                </tr>
                <tr>
                  <td>EHI02839</td>
                  <td>0.53704346</td>
                  <td>0.005429642</td>
                  <td>1.1846421</td>
                  <td>5.22051908</td>
                </tr>
                <tr>
                  <td>EHI02840</td>
                  <td>0.40520960</td>
                  <td>0.170732154</td>
                  <td>0.9714674</td>
                  <td>4.15870634</td>
                </tr>
                <tr>
                  <td>EHI02841</td>
                  <td>0.54078237</td>
                  <td>0.066720106</td>
                  <td>1.3402000</td>
                  <td>4.38657015</td>
                </tr>
                <tr>
                  <td>EHI02843</td>
                  <td>0.51124373</td>
                  <td>0.025768852</td>
                  <td>1.0209520</td>
                  <td>3.79170293</td>
                </tr>
                <tr>
                  <td>EHI02844</td>
                  <td>0.33724044</td>
                  <td>0.257831962</td>
                  <td>1.1673368</td>
                  <td>2.84963063</td>
                </tr>
                <tr>
                  <td>EHI02845</td>
                  <td>0.41473841</td>
                  <td>0.000195490</td>
                  <td>0.5934703</td>
                  <td>2.25767567</td>
                </tr>
                <tr>
                  <td>EHI02846</td>
                  <td>0.43528710</td>
                  <td>0.009803089</td>
                  <td>0.4380401</td>
                  <td>2.07829540</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>sequence_fractions <span class="sc">%&gt;%</span></span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>sample), <span class="sc">~</span>.<span class="sc">/</span><span class="dv">1000000000</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;Sample&quot;</span><span class="ot">=</span><span class="dv">1</span>, <span class="st">&quot;Low quality&quot;</span><span class="ot">=</span><span class="dv">2</span>, <span class="st">&quot;Mapped to host&quot;</span><span class="ot">=</span><span class="dv">3</span>, <span class="st">&quot;Unmapped&quot;</span><span class="ot">=</span><span class="dv">4</span>, <span class="st">&quot;Mapped to MAGs&quot;</span><span class="ot">=</span><span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), mean, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>  <span class="fu">tt</span>()</span></code></pre></div>
<!-- preamble start -->

    <script>

      function styleCell_cgmg6o4u95wm7c52qg6i(i, j, css_id) {
          var table = document.getElementById("tinytable_cgmg6o4u95wm7c52qg6i");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_cgmg6o4u95wm7c52qg6i');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_cgmg6o4u95wm7c52qg6i(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_cgmg6o4u95wm7c52qg6i");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_jx0ii8740xpzclkokwvv',}, 
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 }, { i: 1, j: 2 }, { i: 1, j: 3 },  ], css_id: 'tinytable_css_2ddk7p6gzodogxvlfcdi',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_cgmg6o4u95wm7c52qg6i(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_jx0ii8740xpzclkokwvv, .table th.tinytable_css_jx0ii8740xpzclkokwvv { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
      .table td.tinytable_css_2ddk7p6gzodogxvlfcdi, .table th.tinytable_css_2ddk7p6gzodogxvlfcdi { border-bottom: solid #d3d8dc 0.1em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_cgmg6o4u95wm7c52qg6i" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing='true'>
        <thead>
        
              <tr>
                <th scope="col">Low quality</th>
                <th scope="col">Mapped to host</th>
                <th scope="col">Unmapped</th>
                <th scope="col">Mapped to MAGs</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>0.3650321</td>
                  <td>0.1481815</td>
                  <td>1.08684</td>
                  <td>3.461414</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>sequence_fractions <span class="sc">%&gt;%</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span>sample, <span class="at">names_to =</span> <span class="st">&quot;fraction&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">value =</span> value <span class="sc">/</span> <span class="dv">1000000000</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">fraction =</span> <span class="fu">factor</span>(fraction, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;lowqual_bases&quot;</span>,<span class="st">&quot;host_bases&quot;</span>,<span class="st">&quot;unmapped_bases&quot;</span>,<span class="st">&quot;mags_bases&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>  </span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> sample, <span class="at">y =</span> value, <span class="at">fill=</span>fraction)) <span class="sc">+</span></span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>        <span class="fu">geom_bar</span>(<span class="at">position=</span><span class="st">&quot;stack&quot;</span>, <span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a>      <span class="fu">scale_fill_manual</span>(<span class="at">name=</span><span class="st">&quot;Sequence type&quot;</span>,</span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a>                    <span class="at">breaks=</span><span class="fu">c</span>(<span class="st">&quot;lowqual_bases&quot;</span>,<span class="st">&quot;host_bases&quot;</span>,<span class="st">&quot;unmapped_bases&quot;</span>,<span class="st">&quot;mags_bases&quot;</span>),</span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a>                    <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;Low quality&quot;</span>,<span class="st">&quot;Mapped to host&quot;</span>,<span class="st">&quot;Unmapped&quot;</span>,<span class="st">&quot;Mapped to MAGs&quot;</span>),</span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a>                    <span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;#CCCCCC&quot;</span>, <span class="st">&quot;#bcdee1&quot;</span>, <span class="st">&quot;#d8b8a3&quot;</span>,<span class="st">&quot;#93655c&quot;</span>))<span class="sc">+</span></span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Samples&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Amount of data (GB)&quot;</span>) <span class="sc">+</span></span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a>        <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size=</span><span class="dv">6</span>),<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/dna_fractions_plot-1.png" width="960" /></p>
</div>
<div id="recovered-microbial-fraction" class="section level2 hasAnchor" number="1.8">
<h2 class="hasAnchor"><span class="header-section-number">1.8</span> Recovered microbial fraction<a href="#recovered-microbial-fraction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>singlem_table <span class="ot">&lt;-</span> sequence_fractions <span class="sc">%&gt;%</span></span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mags_proportion =</span> <span class="fu">round</span>((mags_bases <span class="sc">/</span> (mags_bases <span class="sc">+</span> unmapped_bases))<span class="sc">*</span><span class="dv">100</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>    <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample))  <span class="sc">%&gt;%</span></span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">singlem_proportion =</span> <span class="fu">round</span>(singlem_fraction,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a>    <span class="fu">select</span>(sample,mags_proportion,singlem_proportion) <span class="sc">%&gt;%</span></span>
<span id="cb27-6"><a href="#cb27-6" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mags_proportion =</span> <span class="fu">ifelse</span>(singlem_proportion <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, mags_proportion)) <span class="sc">%&gt;%</span> <span class="co">#convert zeros to NA</span></span>
<span id="cb27-7"><a href="#cb27-7" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">singlem_proportion =</span> <span class="fu">ifelse</span>(singlem_proportion <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">NA</span>, singlem_proportion)) <span class="sc">%&gt;%</span> <span class="co">#convert zeros to NA</span></span>
<span id="cb27-8"><a href="#cb27-8" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">singlem_proportion =</span> <span class="fu">ifelse</span>(singlem_proportion <span class="sc">&lt;</span> mags_proportion, <span class="cn">NA</span>, singlem_proportion)) <span class="sc">%&gt;%</span> <span class="co">#if singlem is smaller, then NA, to simplify plot</span></span>
<span id="cb27-9"><a href="#cb27-9" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">singlem_proportion =</span> <span class="fu">ifelse</span>(singlem_proportion <span class="sc">&gt;</span> <span class="dv">100</span>, <span class="dv">100</span>, singlem_proportion)) <span class="co">#simplify</span></span>
<span id="cb27-10"><a href="#cb27-10" tabindex="-1"></a></span>
<span id="cb27-11"><a href="#cb27-11" tabindex="-1"></a>singlem_table <span class="sc">%&gt;%</span></span>
<span id="cb27-12"><a href="#cb27-12" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span>sample, <span class="at">names_to =</span> <span class="st">&quot;proportion&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-13"><a href="#cb27-13" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">proportion =</span> <span class="fu">factor</span>(proportion, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;mags_proportion&quot;</span>,<span class="st">&quot;singlem_proportion&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb27-14"><a href="#cb27-14" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">y =</span> sample, <span class="at">color=</span>proportion)) <span class="sc">+</span></span>
<span id="cb27-15"><a href="#cb27-15" tabindex="-1"></a>            <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> sample), <span class="at">color =</span> <span class="st">&quot;#f8a538&quot;</span>) <span class="sc">+</span></span>
<span id="cb27-16"><a href="#cb27-16" tabindex="-1"></a>            <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb27-17"><a href="#cb27-17" tabindex="-1"></a>      <span class="fu">scale_color_manual</span>(<span class="at">name=</span><span class="st">&quot;Proportion&quot;</span>,</span>
<span id="cb27-18"><a href="#cb27-18" tabindex="-1"></a>                    <span class="at">breaks=</span><span class="fu">c</span>(<span class="st">&quot;mags_proportion&quot;</span>,<span class="st">&quot;singlem_proportion&quot;</span>),</span>
<span id="cb27-19"><a href="#cb27-19" tabindex="-1"></a>                    <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;Recovered&quot;</span>,<span class="st">&quot;Estimated&quot;</span>),</span>
<span id="cb27-20"><a href="#cb27-20" tabindex="-1"></a>                    <span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;#52e1e8&quot;</span>, <span class="st">&quot;#876b53&quot;</span>))<span class="sc">+</span></span>
<span id="cb27-21"><a href="#cb27-21" tabindex="-1"></a>            <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb27-22"><a href="#cb27-22" tabindex="-1"></a>            <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Samples&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Prokaryotic fraction (%)&quot;</span>) <span class="sc">+</span></span>
<span id="cb27-23"><a href="#cb27-23" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size=</span><span class="dv">6</span>),<span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/data_estimations_plot1-1.png" width="960" /></p>
<!--chapter:end:02_data_statistics.Rmd-->
</div>
</div>
<div id="mag-catalogue" class="section level1 hasAnchor" number="2">
<h1 class="hasAnchor"><span class="header-section-number">2</span> MAG catalogue<a href="#mag-catalogue" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="gut-microbiota" class="section level2 hasAnchor" number="2.1">
<h2 class="hasAnchor"><span class="header-section-number">2.1</span> Gut microbiota<a href="#gut-microbiota" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/data.Rdata&quot;</span>)</span></code></pre></div>
<div id="genome-phylogeny" class="section level3 hasAnchor" number="2.1.1">
<h3 class="hasAnchor"><span class="header-section-number">2.1.1</span> Genome phylogeny<a href="#genome-phylogeny" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="co"># Generate the phylum color heatmap</span></span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a>phylum_heatmap <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a>    <span class="fu">right_join</span>(genome_metadata, <span class="at">by=</span><span class="fu">join_by</span>(phylum <span class="sc">==</span> phylum)) <span class="sc">%&gt;%</span></span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">match</span>(genome, genome_tree<span class="sc">$</span>tip.label)) <span class="sc">%&gt;%</span></span>
<span id="cb29-5"><a href="#cb29-5" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" tabindex="-1"></a>    <span class="fu">select</span>(genome,phylum) <span class="sc">%&gt;%</span></span>
<span id="cb29-7"><a href="#cb29-7" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">phylum =</span> <span class="fu">factor</span>(phylum, <span class="at">levels =</span> <span class="fu">unique</span>(phylum))) <span class="sc">%&gt;%</span></span>
<span id="cb29-8"><a href="#cb29-8" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;genome&quot;</span>)</span>
<span id="cb29-9"><a href="#cb29-9" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" tabindex="-1"></a><span class="co"># Generate new species table</span></span>
<span id="cb29-11"><a href="#cb29-11" tabindex="-1"></a>newspecies_table <span class="ot">&lt;-</span> genome_metadata <span class="sc">%&gt;%</span></span>
<span id="cb29-12"><a href="#cb29-12" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">newspecies=</span><span class="fu">ifelse</span>(species<span class="sc">==</span><span class="st">&quot;s__&quot;</span>,<span class="st">&quot;Y&quot;</span>,<span class="st">&quot;N&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb29-13"><a href="#cb29-13" tabindex="-1"></a>  <span class="fu">select</span>(genome,newspecies) <span class="sc">%&gt;%</span></span>
<span id="cb29-14"><a href="#cb29-14" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;genome&quot;</span>)</span>
<span id="cb29-15"><a href="#cb29-15" tabindex="-1"></a></span>
<span id="cb29-16"><a href="#cb29-16" tabindex="-1"></a><span class="co"># Generate  basal tree</span></span>
<span id="cb29-17"><a href="#cb29-17" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span> <span class="fu">force.ultrametric</span>(genome_tree, <span class="at">method=</span><span class="st">&quot;extend&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># extend to ultrametric for the sake of visualisation</span></span>
<span id="cb29-18"><a href="#cb29-18" tabindex="-1"></a>    <span class="fu">ggtree</span>(., <span class="at">layout=</span><span class="st">&quot;fan&quot;</span>, <span class="at">open.angle=</span><span class="dv">10</span>, <span class="at">size=</span><span class="fl">0.5</span>)</span></code></pre></div>
<pre><code>## ***************************************************************
## *                          Note:                              *
## *    force.ultrametric does not include a formal method to    *
## *    ultrametricize a tree &amp; should only be used to coerce    *
## *   a phylogeny that fails is.ultrametric due to rounding --  *
## *    not as a substitute for formal rate-smoothing methods.   *
## ***************************************************************</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="co"># Add phylum ring</span></span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a><span class="co"># circular_tree &lt;- gheatmap(circular_tree, phylum_heatmap, offset=0.55, width=0.1, colnames=FALSE) +</span></span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a><span class="co">#         scale_fill_manual(values=phylum_colors) +</span></span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a><span class="co">#         geom_tiplab2(size=1, hjust=-0.1) +</span></span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a><span class="co">#         theme(legend.position = &quot;none&quot;, plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))</span></span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span> <span class="fu">gheatmap</span>(circular_tree, phylum_heatmap, <span class="at">offset=</span><span class="fl">0.05</span>, <span class="at">width=</span><span class="fl">0.05</span>, <span class="at">colnames=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a>        <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>phylum_colors) <span class="sc">+</span></span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill=</span><span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a>        <span class="co">#theme(legend.position = &quot;none&quot;, plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))</span></span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a></span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a><span class="co"># Flush color scale to enable a new color scheme in the next ring</span></span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span> circular_tree <span class="sc">+</span> <span class="fu">new_scale_fill</span>()</span>
<span id="cb31-13"><a href="#cb31-13" tabindex="-1"></a></span>
<span id="cb31-14"><a href="#cb31-14" tabindex="-1"></a><span class="co"># Add completeness ring</span></span>
<span id="cb31-15"><a href="#cb31-15" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span> circular_tree <span class="sc">+</span></span>
<span id="cb31-16"><a href="#cb31-16" tabindex="-1"></a>        <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb31-17"><a href="#cb31-17" tabindex="-1"></a>        <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#d1f4ba&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#f4baba&quot;</span>) <span class="sc">+</span></span>
<span id="cb31-18"><a href="#cb31-18" tabindex="-1"></a>        <span class="fu">geom_fruit</span>(</span>
<span id="cb31-19"><a href="#cb31-19" tabindex="-1"></a>                <span class="at">data=</span>genome_metadata,</span>
<span id="cb31-20"><a href="#cb31-20" tabindex="-1"></a>                <span class="at">geom=</span>geom_bar,</span>
<span id="cb31-21"><a href="#cb31-21" tabindex="-1"></a>                <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x=</span>completeness, <span class="at">y=</span>genome, <span class="at">fill=</span>contamination),</span>
<span id="cb31-22"><a href="#cb31-22" tabindex="-1"></a>                <span class="at">offset =</span> <span class="fl">0.24</span>,</span>
<span id="cb31-23"><a href="#cb31-23" tabindex="-1"></a>                <span class="at">pwidth =</span> <span class="fl">0.1</span>,</span>
<span id="cb31-24"><a href="#cb31-24" tabindex="-1"></a>                <span class="at">orientation=</span><span class="st">&quot;y&quot;</span>,</span>
<span id="cb31-25"><a href="#cb31-25" tabindex="-1"></a>              <span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb31-26"><a href="#cb31-26" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill=</span><span class="st">&quot;Contamination&quot;</span>)</span>
<span id="cb31-27"><a href="#cb31-27" tabindex="-1"></a></span>
<span id="cb31-28"><a href="#cb31-28" tabindex="-1"></a><span class="co"># Add genome-size ring</span></span>
<span id="cb31-29"><a href="#cb31-29" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span>  circular_tree <span class="sc">+</span> <span class="fu">new_scale_fill</span>()</span>
<span id="cb31-30"><a href="#cb31-30" tabindex="-1"></a></span>
<span id="cb31-31"><a href="#cb31-31" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span> <span class="fu">gheatmap</span>(circular_tree, newspecies_table, <span class="at">offset=</span><span class="fl">0.3</span>, <span class="at">width=</span><span class="fl">0.05</span>, <span class="at">colnames=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb31-32"><a href="#cb31-32" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;#f4f4f4&quot;</span>,<span class="st">&quot;#74C8AE&quot;</span>))<span class="sc">+</span></span>
<span id="cb31-33"><a href="#cb31-33" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill=</span><span class="st">&quot;New species&quot;</span>)</span>
<span id="cb31-34"><a href="#cb31-34" tabindex="-1"></a>        <span class="co">#theme(legend.position = &quot;none&quot;, plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))</span></span>
<span id="cb31-35"><a href="#cb31-35" tabindex="-1"></a></span>
<span id="cb31-36"><a href="#cb31-36" tabindex="-1"></a></span>
<span id="cb31-37"><a href="#cb31-37" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span>  circular_tree <span class="sc">+</span></span>
<span id="cb31-38"><a href="#cb31-38" tabindex="-1"></a>  <span class="fu">geom_fruit</span>(</span>
<span id="cb31-39"><a href="#cb31-39" tabindex="-1"></a>    <span class="at">data=</span>genome_metadata,</span>
<span id="cb31-40"><a href="#cb31-40" tabindex="-1"></a>    <span class="at">geom=</span>geom_bar,</span>
<span id="cb31-41"><a href="#cb31-41" tabindex="-1"></a>    <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x=</span>length, <span class="at">y=</span>genome),</span>
<span id="cb31-42"><a href="#cb31-42" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">&quot;#1e6e55&quot;</span>,</span>
<span id="cb31-43"><a href="#cb31-43" tabindex="-1"></a>    <span class="at">offset =</span> <span class="fl">0.05</span>,</span>
<span id="cb31-44"><a href="#cb31-44" tabindex="-1"></a>    <span class="at">orientation=</span><span class="st">&quot;y&quot;</span>,</span>
<span id="cb31-45"><a href="#cb31-45" tabindex="-1"></a>    <span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)</span>
<span id="cb31-46"><a href="#cb31-46" tabindex="-1"></a></span>
<span id="cb31-47"><a href="#cb31-47" tabindex="-1"></a></span>
<span id="cb31-48"><a href="#cb31-48" tabindex="-1"></a><span class="co"># Add text</span></span>
<span id="cb31-49"><a href="#cb31-49" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span>  circular_tree <span class="sc">+</span></span>
<span id="cb31-50"><a href="#cb31-50" tabindex="-1"></a>        <span class="fu">annotate</span>(<span class="st">&#39;text&#39;</span>, <span class="at">x=</span><span class="fl">3.1</span>, <span class="at">y=</span><span class="dv">0</span>, <span class="at">label=</span><span class="st">&#39;            Phylum&#39;</span>, <span class="at">family=</span><span class="st">&#39;arial&#39;</span>, <span class="at">size=</span><span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb31-51"><a href="#cb31-51" tabindex="-1"></a>        <span class="fu">annotate</span>(<span class="st">&#39;text&#39;</span>, <span class="at">x=</span><span class="fl">3.7</span>, <span class="at">y=</span><span class="dv">0</span>, <span class="at">label=</span><span class="st">&#39;                         Genome quality&#39;</span>, <span class="at">family=</span><span class="st">&#39;arial&#39;</span>, <span class="at">size=</span><span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb31-52"><a href="#cb31-52" tabindex="-1"></a>        <span class="fu">annotate</span>(<span class="st">&#39;text&#39;</span>, <span class="at">x=</span><span class="fl">4.1</span>, <span class="at">y=</span><span class="dv">0</span>, <span class="at">label=</span><span class="st">&#39;                     Genome size&#39;</span>, <span class="at">family=</span><span class="st">&#39;arial&#39;</span>, <span class="at">size=</span><span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb31-53"><a href="#cb31-53" tabindex="-1"></a>        <span class="fu">annotate</span>(<span class="st">&#39;text&#39;</span>, <span class="at">x=</span><span class="fl">3.4</span>, <span class="at">y=</span><span class="dv">0</span>, <span class="at">label=</span><span class="st">&#39;                     New species&#39;</span>, <span class="at">family=</span><span class="st">&#39;arial&#39;</span>, <span class="at">size=</span><span class="fl">3.5</span>)</span>
<span id="cb31-54"><a href="#cb31-54" tabindex="-1"></a></span>
<span id="cb31-55"><a href="#cb31-55" tabindex="-1"></a><span class="co">#Plot circular tree</span></span>
<span id="cb31-56"><a href="#cb31-56" tabindex="-1"></a>circular_tree <span class="sc">%&gt;%</span> <span class="fu">open_tree</span>(<span class="dv">30</span>) <span class="sc">%&gt;%</span> <span class="fu">rotate_tree</span>(<span class="dv">90</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/gut_genome_phylogeny-1.png" width="960" /></p>
</div>
<div id="taxonomy-overview" class="section level3 hasAnchor" number="2.1.2">
<h3 class="hasAnchor"><span class="header-section-number">2.1.2</span> Taxonomy overview<a href="#taxonomy-overview" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>tax_mag <span class="ot">&lt;-</span>genome_metadata <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>  <span class="fu">group_by</span>(phylum) <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mag_n=</span><span class="fu">n</span>())</span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a>tax_mag <span class="sc">%&gt;%</span></span>
<span id="cb32-6"><a href="#cb32-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percetage_mag=</span><span class="fu">round</span>(mag_n<span class="sc">*</span><span class="dv">100</span><span class="sc">/</span><span class="fu">sum</span>(mag_n), <span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb32-7"><a href="#cb32-7" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>percetage_mag) <span class="sc">%&gt;%</span></span>
<span id="cb32-8"><a href="#cb32-8" tabindex="-1"></a>  <span class="fu">tt</span>()</span></code></pre></div>
<!-- preamble start -->

    <script>

      function styleCell_ndxc2ded3wxq2gpyvsfx(i, j, css_id) {
          var table = document.getElementById("tinytable_ndxc2ded3wxq2gpyvsfx");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_ndxc2ded3wxq2gpyvsfx');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_ndxc2ded3wxq2gpyvsfx(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_ndxc2ded3wxq2gpyvsfx");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 },  ], css_id: 'tinytable_css_ux2ywk34czyo2jf5o8wr',}, 
          { positions: [ { i: 15, j: 0 }, { i: 15, j: 1 }, { i: 15, j: 2 },  ], css_id: 'tinytable_css_jx7flp486c9aaoht2jcw',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_ndxc2ded3wxq2gpyvsfx(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_ux2ywk34czyo2jf5o8wr, .table th.tinytable_css_ux2ywk34czyo2jf5o8wr { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
      .table td.tinytable_css_jx7flp486c9aaoht2jcw, .table th.tinytable_css_jx7flp486c9aaoht2jcw { border-bottom: solid #d3d8dc 0.1em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_ndxc2ded3wxq2gpyvsfx" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing='true'>
        <thead>
        
              <tr>
                <th scope="col">phylum</th>
                <th scope="col">mag_n</th>
                <th scope="col">percetage_mag</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>p__Bacillota_A      </td>
                  <td>1440</td>
                  <td>79.25</td>
                </tr>
                <tr>
                  <td>p__Bacillota        </td>
                  <td>  94</td>
                  <td> 5.17</td>
                </tr>
                <tr>
                  <td>p__Cyanobacteriota  </td>
                  <td>  82</td>
                  <td> 4.51</td>
                </tr>
                <tr>
                  <td>p__Bacteroidota     </td>
                  <td>  71</td>
                  <td> 3.91</td>
                </tr>
                <tr>
                  <td>p__Pseudomonadota   </td>
                  <td>  34</td>
                  <td> 1.87</td>
                </tr>
                <tr>
                  <td>p__Actinomycetota   </td>
                  <td>  25</td>
                  <td> 1.38</td>
                </tr>
                <tr>
                  <td>p__Synergistota     </td>
                  <td>  17</td>
                  <td> 0.94</td>
                </tr>
                <tr>
                  <td>p__Campylobacterota </td>
                  <td>  15</td>
                  <td> 0.83</td>
                </tr>
                <tr>
                  <td>p__Thermoplasmatota </td>
                  <td>  12</td>
                  <td> 0.66</td>
                </tr>
                <tr>
                  <td>p__Desulfobacterota </td>
                  <td>   8</td>
                  <td> 0.44</td>
                </tr>
                <tr>
                  <td>p__Bacillota_C      </td>
                  <td>   7</td>
                  <td> 0.39</td>
                </tr>
                <tr>
                  <td>p__Verrucomicrobiota</td>
                  <td>   6</td>
                  <td> 0.33</td>
                </tr>
                <tr>
                  <td>p__Spirochaetota    </td>
                  <td>   4</td>
                  <td> 0.22</td>
                </tr>
                <tr>
                  <td>p__Bacillota_B      </td>
                  <td>   1</td>
                  <td> 0.06</td>
                </tr>
                <tr>
                  <td>p__Methanobacteriota</td>
                  <td>   1</td>
                  <td> 0.06</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
<div id="mag-size-mb" class="section level3 hasAnchor" number="2.1.3">
<h3 class="hasAnchor"><span class="header-section-number">2.1.3</span> Mag size (MB)<a href="#mag-size-mb" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>genome_metadata <span class="ot">&lt;-</span>genome_metadata <span class="sc">%&gt;%</span></span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">corrected_size=</span><span class="dv">100</span><span class="sc">*</span>length<span class="sc">/</span>completeness) <span class="sc">%&gt;%</span></span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>  <span class="fu">arrange</span>(completeness)</span></code></pre></div>
<p>Mags average size (MB)</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>genome_metadata <span class="sc">%&gt;%</span></span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Average_corrected_size=</span><span class="fu">mean</span>(corrected_size))</span></code></pre></div>
<pre><code>## # A tibble: 1 × 1
##   Average_corrected_size
##                    &lt;dbl&gt;
## 1               2305214.</code></pre>
</div>
<div id="genome-quality" class="section level3 hasAnchor" number="2.1.4">
<h3 class="hasAnchor"><span class="header-section-number">2.1.4</span> Genome quality<a href="#genome-quality" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>genome_metadata <span class="sc">%&gt;%</span></span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">completeness_mean=</span><span class="fu">mean</span>(completeness) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">as.character</span>(),</span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a>              <span class="at">completeness_sd=</span><span class="fu">sd</span>(completeness) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">as.character</span>(),</span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a>              <span class="at">contamination_mean=</span><span class="fu">mean</span>(contamination) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>),</span>
<span id="cb36-5"><a href="#cb36-5" tabindex="-1"></a>              <span class="at">contamination_sd=</span><span class="fu">sd</span>(contamination) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-6"><a href="#cb36-6" tabindex="-1"></a>    <span class="fu">unite</span>(<span class="st">&quot;Completeness&quot;</span>,completeness_mean, completeness_sd, <span class="at">sep =</span> <span class="st">&quot; ± &quot;</span>, <span class="at">remove =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-7"><a href="#cb36-7" tabindex="-1"></a>    <span class="fu">unite</span>(<span class="st">&quot;Contamination&quot;</span>,contamination_mean, contamination_sd, <span class="at">sep =</span> <span class="st">&quot; ± &quot;</span>, <span class="at">remove =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-8"><a href="#cb36-8" tabindex="-1"></a>    <span class="fu">tt</span>()</span></code></pre></div>
<!-- preamble start -->

    <script>

      function styleCell_xucorvi9fmpnytsnz281(i, j, css_id) {
          var table = document.getElementById("tinytable_xucorvi9fmpnytsnz281");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_xucorvi9fmpnytsnz281');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_xucorvi9fmpnytsnz281(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_xucorvi9fmpnytsnz281");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 },  ], css_id: 'tinytable_css_fwaybvwm3l13f4e01c48',}, 
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 },  ], css_id: 'tinytable_css_3ihjwsjx4vay0lsulcbq',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_xucorvi9fmpnytsnz281(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_fwaybvwm3l13f4e01c48, .table th.tinytable_css_fwaybvwm3l13f4e01c48 { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
      .table td.tinytable_css_3ihjwsjx4vay0lsulcbq, .table th.tinytable_css_3ihjwsjx4vay0lsulcbq { border-bottom: solid #d3d8dc 0.1em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_xucorvi9fmpnytsnz281" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing='true'>
        <thead>
        
              <tr>
                <th scope="col">Completeness</th>
                <th scope="col">Contamination</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>79.4 ± 16.77</td>
                  <td>2.79 ± 2.58</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="co">#Generate quality biplot</span></span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a>genome_biplot <span class="ot">&lt;-</span> genome_metadata <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(genome,domain,phylum,completeness,contamination,length)) <span class="sc">%&gt;%</span></span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">match</span>(genome, <span class="fu">rev</span>(genome_tree<span class="sc">$</span>tip.label))) <span class="sc">%&gt;%</span> <span class="co">#sort MAGs according to phylogenetic tree</span></span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>completeness,<span class="at">y=</span>contamination,<span class="at">size=</span>length,<span class="at">color=</span>phylum)) <span class="sc">+</span></span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a>              <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a>                    <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a>                    <span class="fu">scale_color_manual</span>(<span class="at">values=</span>phylum_colors) <span class="sc">+</span></span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a>                    <span class="fu">labs</span>(<span class="at">y=</span> <span class="st">&quot;Contamination&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Completeness&quot;</span>) <span class="sc">+</span></span>
<span id="cb37-10"><a href="#cb37-10" tabindex="-1"></a>                    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb37-11"><a href="#cb37-11" tabindex="-1"></a>                    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb37-12"><a href="#cb37-12" tabindex="-1"></a></span>
<span id="cb37-13"><a href="#cb37-13" tabindex="-1"></a><span class="co">#Generate contamination boxplot</span></span>
<span id="cb37-14"><a href="#cb37-14" tabindex="-1"></a>genome_contamination <span class="ot">&lt;-</span> genome_metadata <span class="sc">%&gt;%</span></span>
<span id="cb37-15"><a href="#cb37-15" tabindex="-1"></a>            <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y=</span>contamination)) <span class="sc">+</span></span>
<span id="cb37-16"><a href="#cb37-16" tabindex="-1"></a>                    <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb37-17"><a href="#cb37-17" tabindex="-1"></a>                    <span class="fu">geom_boxplot</span>(<span class="at">colour =</span> <span class="st">&quot;#999999&quot;</span>, <span class="at">fill=</span><span class="st">&quot;#cccccc&quot;</span>) <span class="sc">+</span></span>
<span id="cb37-18"><a href="#cb37-18" tabindex="-1"></a>                    <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb37-19"><a href="#cb37-19" tabindex="-1"></a>                    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb37-20"><a href="#cb37-20" tabindex="-1"></a>                        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb37-21"><a href="#cb37-21" tabindex="-1"></a>                        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb37-22"><a href="#cb37-22" tabindex="-1"></a>                        <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb37-23"><a href="#cb37-23" tabindex="-1"></a>                        <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb37-24"><a href="#cb37-24" tabindex="-1"></a>                        <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb37-25"><a href="#cb37-25" tabindex="-1"></a>                        <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb37-26"><a href="#cb37-26" tabindex="-1"></a>                        <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">0.40</span>, <span class="dv">0</span>),<span class="st">&quot;inches&quot;</span>)) <span class="co">#add bottom-margin (top, right, bottom, left)</span></span>
<span id="cb37-27"><a href="#cb37-27" tabindex="-1"></a></span>
<span id="cb37-28"><a href="#cb37-28" tabindex="-1"></a><span class="co">#Generate completeness boxplot</span></span>
<span id="cb37-29"><a href="#cb37-29" tabindex="-1"></a>genome_completeness <span class="ot">&lt;-</span> genome_metadata <span class="sc">%&gt;%</span></span>
<span id="cb37-30"><a href="#cb37-30" tabindex="-1"></a>        <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>completeness)) <span class="sc">+</span></span>
<span id="cb37-31"><a href="#cb37-31" tabindex="-1"></a>                <span class="fu">xlim</span>(<span class="fu">c</span>(<span class="dv">50</span>,<span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb37-32"><a href="#cb37-32" tabindex="-1"></a>                <span class="fu">geom_boxplot</span>(<span class="at">colour =</span> <span class="st">&quot;#999999&quot;</span>, <span class="at">fill=</span><span class="st">&quot;#cccccc&quot;</span>) <span class="sc">+</span></span>
<span id="cb37-33"><a href="#cb37-33" tabindex="-1"></a>                <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb37-34"><a href="#cb37-34" tabindex="-1"></a>                <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb37-35"><a href="#cb37-35" tabindex="-1"></a>                    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb37-36"><a href="#cb37-36" tabindex="-1"></a>                    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb37-37"><a href="#cb37-37" tabindex="-1"></a>                    <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb37-38"><a href="#cb37-38" tabindex="-1"></a>                    <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb37-39"><a href="#cb37-39" tabindex="-1"></a>                    <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb37-40"><a href="#cb37-40" tabindex="-1"></a>                    <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb37-41"><a href="#cb37-41" tabindex="-1"></a>                    <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">0.50</span>),<span class="st">&quot;inches&quot;</span>)) <span class="co">#add left-margin (top, right, bottom, left)</span></span>
<span id="cb37-42"><a href="#cb37-42" tabindex="-1"></a></span>
<span id="cb37-43"><a href="#cb37-43" tabindex="-1"></a><span class="co">#Render composite figure</span></span>
<span id="cb37-44"><a href="#cb37-44" tabindex="-1"></a><span class="fu">grid.arrange</span>(<span class="at">grobs =</span> <span class="fu">list</span>(genome_completeness,genome_biplot,genome_contamination),</span>
<span id="cb37-45"><a href="#cb37-45" tabindex="-1"></a>        <span class="at">layout_matrix =</span> <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">4</span>),</span>
<span id="cb37-46"><a href="#cb37-46" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb37-47"><a href="#cb37-47" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb37-48"><a href="#cb37-48" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb37-49"><a href="#cb37-49" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb37-50"><a href="#cb37-50" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb37-51"><a href="#cb37-51" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb37-52"><a href="#cb37-52" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb37-53"><a href="#cb37-53" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb37-54"><a href="#cb37-54" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb37-55"><a href="#cb37-55" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb37-56"><a href="#cb37-56" tabindex="-1"></a>                              <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>)))</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/gut_genome_quality_plot-1.png" width="960" /></p>
</div>
<div id="functional-overview" class="section level3 hasAnchor" number="2.1.5">
<h3 class="hasAnchor"><span class="header-section-number">2.1.5</span> Functional overview<a href="#functional-overview" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong><em>Predicted genes</em></strong></p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>genome_annotations <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/genome_annotations.tsv.xz&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">gene=</span><span class="dv">1</span>, <span class="at">genome=</span><span class="dv">2</span>, <span class="at">contig=</span><span class="dv">3</span>)</span>
<span id="cb38-3"><a href="#cb38-3" tabindex="-1"></a></span>
<span id="cb38-4"><a href="#cb38-4" tabindex="-1"></a><span class="co"># Predicted genes</span></span>
<span id="cb38-5"><a href="#cb38-5" tabindex="-1"></a>pred_genes <span class="ot">&lt;-</span> genome_annotations <span class="sc">%&gt;%</span></span>
<span id="cb38-6"><a href="#cb38-6" tabindex="-1"></a>  <span class="fu">nrow</span>()</span>
<span id="cb38-7"><a href="#cb38-7" tabindex="-1"></a></span>
<span id="cb38-8"><a href="#cb38-8" tabindex="-1"></a><span class="fu">cat</span>(pred_genes)</span></code></pre></div>
<pre><code>## 1914029</code></pre>
<p><strong><em>Number of annotated genes and percentages</em></strong></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a><span class="co"># Some annotation</span></span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a>genome_annota <span class="ot">&lt;-</span> genome_annotations <span class="sc">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">if_any</span>(<span class="fu">c</span>(kegg_id, pfam_hits, cazy_hits), <span class="sc">~</span> <span class="sc">!</span><span class="fu">is.na</span>(.))) <span class="sc">%&gt;%</span></span>
<span id="cb40-4"><a href="#cb40-4" tabindex="-1"></a>  <span class="fu">nrow</span>()</span>
<span id="cb40-5"><a href="#cb40-5" tabindex="-1"></a></span>
<span id="cb40-6"><a href="#cb40-6" tabindex="-1"></a><span class="fu">cat</span>(genome_annota)</span>
<span id="cb40-7"><a href="#cb40-7" tabindex="-1"></a></span>
<span id="cb40-8"><a href="#cb40-8" tabindex="-1"></a><span class="co"># Some annotation percentage</span></span>
<span id="cb40-9"><a href="#cb40-9" tabindex="-1"></a>genome_annota<span class="sc">*</span><span class="dv">100</span><span class="sc">/</span>pred_genes</span></code></pre></div>
<p><strong><em>Number of KEGG annotatated genes and percentages</em></strong></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="co"># KEGG annotation</span></span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>kegg_annota <span class="ot">&lt;-</span> genome_annotations <span class="sc">%&gt;%</span></span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(kegg_id)) <span class="sc">%&gt;%</span></span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a>  <span class="fu">nrow</span>()</span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a><span class="fu">cat</span>(kegg_annota)</span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a><span class="co"># KEGG annotation percentage</span></span>
<span id="cb41-8"><a href="#cb41-8" tabindex="-1"></a>kegg_annota<span class="sc">*</span><span class="dv">100</span><span class="sc">/</span>genome_annota</span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="co"># Aggregate basal GIFT into elements</span></span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a>function_table <span class="ot">&lt;-</span> genome_gifts <span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a>    <span class="fu">to.elements</span>(., GIFT_db)</span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a><span class="co"># Generate  basal tree</span></span>
<span id="cb42-6"><a href="#cb42-6" tabindex="-1"></a>function_tree <span class="ot">&lt;-</span> <span class="fu">force.ultrametric</span>(genome_tree, <span class="at">method=</span><span class="st">&quot;extend&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-7"><a href="#cb42-7" tabindex="-1"></a>                <span class="fu">ggtree</span>(., <span class="at">size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<pre><code>## ***************************************************************
## *                          Note:                              *
## *    force.ultrametric does not include a formal method to    *
## *    ultrametricize a tree &amp; should only be used to coerce    *
## *   a phylogeny that fails is.ultrametric due to rounding --  *
## *    not as a substitute for formal rate-smoothing methods.   *
## ***************************************************************</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="co">#Add phylum colors next to the tree tips</span></span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a>function_tree <span class="ot">&lt;-</span> <span class="fu">gheatmap</span>(function_tree, phylum_heatmap, <span class="at">offset=</span><span class="dv">0</span>, <span class="at">width=</span><span class="fl">0.1</span>, <span class="at">colnames=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb44-3"><a href="#cb44-3" tabindex="-1"></a>            <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>phylum_colors) <span class="sc">+</span></span>
<span id="cb44-4"><a href="#cb44-4" tabindex="-1"></a>            <span class="fu">labs</span>(<span class="at">fill=</span><span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb44-5"><a href="#cb44-5" tabindex="-1"></a></span>
<span id="cb44-6"><a href="#cb44-6" tabindex="-1"></a><span class="co">#Reset fill scale to use a different colour profile in the heatmap</span></span>
<span id="cb44-7"><a href="#cb44-7" tabindex="-1"></a>function_tree <span class="ot">&lt;-</span> function_tree <span class="sc">+</span> <span class="fu">new_scale_fill</span>()</span>
<span id="cb44-8"><a href="#cb44-8" tabindex="-1"></a></span>
<span id="cb44-9"><a href="#cb44-9" tabindex="-1"></a><span class="co">#Add functions heatmap</span></span>
<span id="cb44-10"><a href="#cb44-10" tabindex="-1"></a>function_tree <span class="ot">&lt;-</span> <span class="fu">gheatmap</span>(function_tree, function_table, <span class="at">offset=</span><span class="fl">0.5</span>, <span class="at">width=</span><span class="fl">3.5</span>, <span class="at">colnames=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb44-11"><a href="#cb44-11" tabindex="-1"></a>            <span class="fu">vexpand</span>(.<span class="dv">08</span>) <span class="sc">+</span></span>
<span id="cb44-12"><a href="#cb44-12" tabindex="-1"></a>            <span class="fu">coord_cartesian</span>(<span class="at">clip =</span> <span class="st">&quot;off&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-13"><a href="#cb44-13" tabindex="-1"></a>            <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#f4f4f4&quot;</span>, <span class="at">high =</span> <span class="st">&quot;steelblue&quot;</span>, <span class="at">na.value=</span><span class="st">&quot;white&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-14"><a href="#cb44-14" tabindex="-1"></a>            <span class="fu">labs</span>(<span class="at">fill=</span><span class="st">&quot;GIFT&quot;</span>)</span>
<span id="cb44-15"><a href="#cb44-15" tabindex="-1"></a></span>
<span id="cb44-16"><a href="#cb44-16" tabindex="-1"></a><span class="co">#Reset fill scale to use a different colour profile in the heatmap</span></span>
<span id="cb44-17"><a href="#cb44-17" tabindex="-1"></a>function_tree <span class="ot">&lt;-</span> function_tree <span class="sc">+</span> <span class="fu">new_scale_fill</span>()</span>
<span id="cb44-18"><a href="#cb44-18" tabindex="-1"></a></span>
<span id="cb44-19"><a href="#cb44-19" tabindex="-1"></a><span class="co"># Add completeness barplots</span></span>
<span id="cb44-20"><a href="#cb44-20" tabindex="-1"></a>function_tree <span class="ot">&lt;-</span> function_tree <span class="sc">+</span></span>
<span id="cb44-21"><a href="#cb44-21" tabindex="-1"></a>            <span class="fu">geom_fruit</span>(<span class="at">data=</span>genome_metadata,</span>
<span id="cb44-22"><a href="#cb44-22" tabindex="-1"></a>            <span class="at">geom=</span>geom_bar,</span>
<span id="cb44-23"><a href="#cb44-23" tabindex="-1"></a>            <span class="at">grid.params=</span><span class="fu">list</span>(<span class="at">axis=</span><span class="st">&quot;x&quot;</span>, <span class="at">text.size=</span><span class="dv">2</span>, <span class="at">nbreak =</span> <span class="dv">1</span>),</span>
<span id="cb44-24"><a href="#cb44-24" tabindex="-1"></a>            <span class="at">axis.params=</span><span class="fu">list</span>(<span class="at">vline=</span><span class="cn">TRUE</span>),</span>
<span id="cb44-25"><a href="#cb44-25" tabindex="-1"></a>            <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x=</span>length, <span class="at">y=</span>genome, <span class="at">fill=</span>completeness),</span>
<span id="cb44-26"><a href="#cb44-26" tabindex="-1"></a>                 <span class="at">offset =</span> <span class="fl">3.8</span>,</span>
<span id="cb44-27"><a href="#cb44-27" tabindex="-1"></a>                 <span class="at">orientation=</span><span class="st">&quot;y&quot;</span>,</span>
<span id="cb44-28"><a href="#cb44-28" tabindex="-1"></a>                 <span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-29"><a href="#cb44-29" tabindex="-1"></a>            <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#cf8888&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#a2cc87&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-30"><a href="#cb44-30" tabindex="-1"></a>            <span class="fu">labs</span>(<span class="at">fill=</span><span class="st">&quot;Genome</span><span class="sc">\n</span><span class="st">completeness&quot;</span>)</span>
<span id="cb44-31"><a href="#cb44-31" tabindex="-1"></a></span>
<span id="cb44-32"><a href="#cb44-32" tabindex="-1"></a>function_tree</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/function_heatmap-1.png" width="960" /></p>
</div>
<div id="functional-ordination" class="section level3 hasAnchor" number="2.1.6">
<h3 class="hasAnchor"><span class="header-section-number">2.1.6</span> Functional ordination<a href="#functional-ordination" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="co"># Generate the tSNE ordination</span></span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a>tSNE_function <span class="ot">&lt;-</span> <span class="fu">Rtsne</span>(<span class="at">X=</span>function_table, <span class="at">dims =</span> <span class="dv">2</span>, <span class="at">check_duplicates =</span> <span class="cn">FALSE</span>)</span>
<span id="cb45-3"><a href="#cb45-3" tabindex="-1"></a></span>
<span id="cb45-4"><a href="#cb45-4" tabindex="-1"></a><span class="co"># Plot the ordination</span></span>
<span id="cb45-5"><a href="#cb45-5" tabindex="-1"></a>function_ordination <span class="ot">&lt;-</span> tSNE_function<span class="sc">$</span>Y <span class="sc">%&gt;%</span></span>
<span id="cb45-6"><a href="#cb45-6" tabindex="-1"></a>                <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb45-7"><a href="#cb45-7" tabindex="-1"></a>                <span class="fu">mutate</span>(<span class="at">genome=</span><span class="fu">rownames</span>(function_table)) <span class="sc">%&gt;%</span></span>
<span id="cb45-8"><a href="#cb45-8" tabindex="-1"></a>                <span class="fu">inner_join</span>(genome_metadata, <span class="at">by=</span><span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-9"><a href="#cb45-9" tabindex="-1"></a>                <span class="fu">rename</span>(<span class="at">tSNE1=</span><span class="st">&quot;V1&quot;</span>, <span class="at">tSNE2=</span><span class="st">&quot;V2&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-10"><a href="#cb45-10" tabindex="-1"></a>                <span class="fu">select</span>(genome,phylum,tSNE1,tSNE2, length) <span class="sc">%&gt;%</span></span>
<span id="cb45-11"><a href="#cb45-11" tabindex="-1"></a>                <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> tSNE1, <span class="at">y =</span> tSNE2, <span class="at">color =</span> phylum, <span class="at">size=</span>length))<span class="sc">+</span></span>
<span id="cb45-12"><a href="#cb45-12" tabindex="-1"></a>                            <span class="fu">geom_point</span>(<span class="at">shape=</span><span class="dv">16</span>, <span class="at">alpha=</span><span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb45-13"><a href="#cb45-13" tabindex="-1"></a>                            <span class="fu">scale_color_manual</span>(<span class="at">values=</span>phylum_colors) <span class="sc">+</span></span>
<span id="cb45-14"><a href="#cb45-14" tabindex="-1"></a>                            <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb45-15"><a href="#cb45-15" tabindex="-1"></a>                <span class="fu">labs</span>(<span class="at">color=</span><span class="st">&quot;Phylum&quot;</span>, <span class="at">size=</span><span class="st">&quot;Genome size&quot;</span>) <span class="sc">+</span></span>
<span id="cb45-16"><a href="#cb45-16" tabindex="-1"></a>                <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">5</span>))) <span class="co"># enlarge Phylum dots in legend</span></span>
<span id="cb45-17"><a href="#cb45-17" tabindex="-1"></a></span>
<span id="cb45-18"><a href="#cb45-18" tabindex="-1"></a>function_ordination</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/function_ordination-1.png" width="960" /></p>
<!--chapter:end:03_MAGs_Catalouge.Rmd-->
</div>
</div>
</div>
<div id="community-composition" class="section level1 hasAnchor" number="3">
<h1 class="hasAnchor"><span class="header-section-number">3</span> Community composition<a href="#community-composition" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="gut-microbiota-1" class="section level2 hasAnchor" number="3.1">
<h2 class="hasAnchor"><span class="header-section-number">3.1</span> Gut microbiota<a href="#gut-microbiota-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/data.Rdata&quot;</span>)</span></code></pre></div>
<div id="taxonomy-overview-1" class="section level3 hasAnchor" number="3.1.1">
<h3 class="hasAnchor"><span class="header-section-number">3.1.1</span> Taxonomy overview<a href="#taxonomy-overview-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="phylum-level" class="section level4 hasAnchor" number="3.1.1.1">
<h4 class="hasAnchor"><span class="header-section-number">3.1.1.1</span> Phylum level<a href="#phylum-level" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" tabindex="-1"></a>genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb47-2"><a href="#cb47-2" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>genome),<span class="sc">~</span>.<span class="sc">/</span><span class="fu">sum</span>(.)) <span class="sc">%&gt;%</span> <span class="co">#apply TSS nornalisation</span></span>
<span id="cb47-3"><a href="#cb47-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>genome, <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;count&quot;</span>) <span class="sc">%&gt;%</span> <span class="co">#reduce to minimum number of columns</span></span>
<span id="cb47-4"><a href="#cb47-4" tabindex="-1"></a>  <span class="fu">left_join</span>(., genome_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(genome <span class="sc">==</span> genome)) <span class="sc">%&gt;%</span> <span class="co">#append genome metadata</span></span>
<span id="cb47-5"><a href="#cb47-5" tabindex="-1"></a>  <span class="fu">left_join</span>(., sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span> <span class="co">#append sample metadata</span></span>
<span id="cb47-6"><a href="#cb47-6" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="co">#filter 0 counts</span></span>
<span id="cb47-7"><a href="#cb47-7" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x=</span>sample,<span class="at">y=</span>count, <span class="at">fill=</span>phylum, <span class="at">group=</span>phylum)) <span class="sc">+</span> <span class="co">#grouping enables keeping the same sorting of taxonomic units</span></span>
<span id="cb47-8"><a href="#cb47-8" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">colour=</span><span class="st">&quot;white&quot;</span>, <span class="at">linewidth=</span><span class="fl">0.1</span>) <span class="sc">+</span> <span class="co">#plot stacked bars with white borders</span></span>
<span id="cb47-9"><a href="#cb47-9" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>phylum_colors) <span class="sc">+</span></span>
<span id="cb47-10"><a href="#cb47-10" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb47-11"><a href="#cb47-11" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-12"><a href="#cb47-12" tabindex="-1"></a>          <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-13"><a href="#cb47-13" tabindex="-1"></a>          <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-14"><a href="#cb47-14" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-15"><a href="#cb47-15" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-16"><a href="#cb47-16" tabindex="-1"></a>          <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-17"><a href="#cb47-17" tabindex="-1"></a>          <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-18"><a href="#cb47-18" tabindex="-1"></a>          <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;white&quot;</span>),</span>
<span id="cb47-19"><a href="#cb47-19" tabindex="-1"></a>          <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">lineheight =</span> <span class="fl">0.6</span>,<span class="at">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb47-20"><a href="#cb47-20" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> <span class="fl">0.5</span>, <span class="at">linetype =</span> <span class="st">&quot;solid&quot;</span>, <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>)) <span class="sc">+</span></span>
<span id="cb47-21"><a href="#cb47-21" tabindex="-1"></a>   <span class="fu">labs</span>(<span class="at">fill=</span><span class="st">&quot;Phylum&quot;</span>,<span class="at">y =</span> <span class="st">&quot;Relative abundance&quot;</span>,<span class="at">x=</span><span class="st">&quot;Samples&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/taxonomy_barplot-1.png" width="960" /></p>
<p><strong>Number of MAGs</strong></p>
<pre><code>1817</code></pre>
<p><strong>Number of bacteria phyla</strong></p>
<pre><code>13</code></pre>
<p><strong>Number of Archaea phyla</strong></p>
<pre><code>2</code></pre>
<p><strong><em>Phylum relative abundances</em></strong></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" tabindex="-1"></a>phylum_summary <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb51-2"><a href="#cb51-2" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>genome),<span class="sc">~</span>.<span class="sc">/</span><span class="fu">sum</span>(.)) <span class="sc">%&gt;%</span></span>
<span id="cb51-3"><a href="#cb51-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>genome, <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;count&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-4"><a href="#cb51-4" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb51-5"><a href="#cb51-5" tabindex="-1"></a>  <span class="fu">left_join</span>(genome_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(genome <span class="sc">==</span> genome)) <span class="sc">%&gt;%</span></span>
<span id="cb51-6"><a href="#cb51-6" tabindex="-1"></a>  <span class="fu">group_by</span>(sample,phylum,broad_environment, sex) <span class="sc">%&gt;%</span></span>
<span id="cb51-7"><a href="#cb51-7" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">relabun=</span><span class="fu">sum</span>(count))</span></code></pre></div>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" tabindex="-1"></a>phylum_summary <span class="sc">%&gt;%</span></span>
<span id="cb52-2"><a href="#cb52-2" tabindex="-1"></a>  <span class="fu">group_by</span>(phylum) <span class="sc">%&gt;%</span></span>
<span id="cb52-3"><a href="#cb52-3" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb52-4"><a href="#cb52-4" tabindex="-1"></a>    <span class="at">total_mean =</span> <span class="fu">mean</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb52-5"><a href="#cb52-5" tabindex="-1"></a>    <span class="at">total_sd =</span> <span class="fu">sd</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb52-6"><a href="#cb52-6" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb52-7"><a href="#cb52-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Total =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(total_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(total_sd, <span class="dv">3</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb52-8"><a href="#cb52-8" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>total_mean) <span class="sc">%&gt;%</span></span>
<span id="cb52-9"><a href="#cb52-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(phylum, Total)</span></code></pre></div>
<pre><code>## # A tibble: 15 × 2
##    phylum               Total        
##    &lt;chr&gt;                &lt;chr&gt;        
##  1 p__Bacillota_A       60.713±17.816
##  2 p__Bacteroidota      15.337±12.756
##  3 p__Bacillota         10.651±15.13 
##  4 p__Synergistota      2.856±1.778  
##  5 p__Cyanobacteriota   2.624±3.286  
##  6 p__Verrucomicrobiota 2.074±1.467  
##  7 p__Pseudomonadota    1.207±2.219  
##  8 p__Bacillota_C       1.133±0.683  
##  9 p__Thermoplasmatota  0.868±1.487  
## 10 p__Campylobacterota  0.866±1.568  
## 11 p__Actinomycetota    0.749±1.478  
## 12 p__Desulfobacterota  0.681±0.697  
## 13 p__Spirochaetota     0.189±0.607  
## 14 p__Methanobacteriota 0.03±0.061   
## 15 p__Bacillota_B       0.02±0.043</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" tabindex="-1"></a>phylum_arrange <span class="ot">&lt;-</span> phylum_summary <span class="sc">%&gt;%</span></span>
<span id="cb54-2"><a href="#cb54-2" tabindex="-1"></a>    <span class="fu">group_by</span>(phylum) <span class="sc">%&gt;%</span></span>
<span id="cb54-3"><a href="#cb54-3" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">sum</span>(relabun)) <span class="sc">%&gt;%</span></span>
<span id="cb54-4"><a href="#cb54-4" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>mean) <span class="sc">%&gt;%</span></span>
<span id="cb54-5"><a href="#cb54-5" tabindex="-1"></a>    <span class="fu">select</span>(phylum) <span class="sc">%&gt;%</span></span>
<span id="cb54-6"><a href="#cb54-6" tabindex="-1"></a>    <span class="fu">pull</span>()</span>
<span id="cb54-7"><a href="#cb54-7" tabindex="-1"></a></span>
<span id="cb54-8"><a href="#cb54-8" tabindex="-1"></a>phylum_summary <span class="sc">%&gt;%</span></span>
<span id="cb54-9"><a href="#cb54-9" tabindex="-1"></a>    <span class="fu">left_join</span>(genome_metadata <span class="sc">%&gt;%</span> <span class="fu">select</span>(phylum,phylum) <span class="sc">%&gt;%</span> <span class="fu">unique</span>(),<span class="at">by=</span><span class="fu">join_by</span>(phylum<span class="sc">==</span>phylum)) <span class="sc">%&gt;%</span></span>
<span id="cb54-10"><a href="#cb54-10" tabindex="-1"></a><span class="co">#    left_join(sample_metadata,by=join_by(sample==sample)) %&gt;%</span></span>
<span id="cb54-11"><a href="#cb54-11" tabindex="-1"></a>    <span class="fu">filter</span>(phylum <span class="sc">%in%</span> phylum_arrange[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb54-12"><a href="#cb54-12" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">phylum=</span><span class="fu">factor</span>(phylum,<span class="at">levels=</span><span class="fu">rev</span>(phylum_arrange[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]))) <span class="sc">%&gt;%</span></span>
<span id="cb54-13"><a href="#cb54-13" tabindex="-1"></a>    <span class="fu">filter</span>(relabun <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-14"><a href="#cb54-14" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>relabun, <span class="at">y=</span>phylum, <span class="at">group=</span>phylum, <span class="at">color=</span>phylum)) <span class="sc">+</span></span>
<span id="cb54-15"><a href="#cb54-15" tabindex="-1"></a>        <span class="fu">scale_color_manual</span>(<span class="at">values=</span>phylum_colors[<span class="sc">-</span><span class="dv">8</span>]) <span class="sc">+</span></span>
<span id="cb54-16"><a href="#cb54-16" tabindex="-1"></a>        <span class="fu">geom_jitter</span>(<span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb54-17"><a href="#cb54-17" tabindex="-1"></a>        <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb54-18"><a href="#cb54-18" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;phylum&quot;</span>, <span class="at">x=</span><span class="st">&quot;Relative abundance&quot;</span>, <span class="at">color=</span><span class="st">&quot;Phylum&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/taxonomy_jitterplot_phylum-1.png" width="960" />
<strong>Bacteria phyla in individuals from females</strong></p>
<pre><code>12</code></pre>
<pre><code>13</code></pre>
<p><strong>Bacteria phyla in individuals from Temperate woodland</strong></p>
<pre><code>13</code></pre>
<p><strong>Bacteria phyla in individuals from Xeric shrubland</strong></p>
<pre><code>13</code></pre>
<p><strong>Bacteria phyla in individuals from mixed_forest</strong></p>
<pre><code>11</code></pre>
<p><strong>Bacteria phyla in individuals from cropland</strong></p>
<pre><code>12</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" tabindex="-1"></a>phylum_summary <span class="sc">%&gt;%</span></span>
<span id="cb61-2"><a href="#cb61-2" tabindex="-1"></a>  <span class="fu">group_by</span>(phylum) <span class="sc">%&gt;%</span></span>
<span id="cb61-3"><a href="#cb61-3" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb61-4"><a href="#cb61-4" tabindex="-1"></a>    <span class="at">total_mean =</span> <span class="fu">mean</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb61-5"><a href="#cb61-5" tabindex="-1"></a>    <span class="at">total_sd =</span> <span class="fu">sd</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb61-6"><a href="#cb61-6" tabindex="-1"></a>    <span class="at">le_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000245 - Cropland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb61-7"><a href="#cb61-7" tabindex="-1"></a>    <span class="at">le_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000245 - Cropland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb61-8"><a href="#cb61-8" tabindex="-1"></a>    <span class="at">ha_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000198 - Mixed forest&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb61-9"><a href="#cb61-9" tabindex="-1"></a>    <span class="at">ha_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000198 - Mixed forest&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb61-10"><a href="#cb61-10" tabindex="-1"></a>    <span class="at">er_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000218 - Xeric shrubland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb61-11"><a href="#cb61-11" tabindex="-1"></a>    <span class="at">er_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000218 - Xeric shrubland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb61-12"><a href="#cb61-12" tabindex="-1"></a>    <span class="at">go_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000221 - Temperate woodland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb61-13"><a href="#cb61-13" tabindex="-1"></a>    <span class="at">go_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000221 - Temperate woodland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb61-14"><a href="#cb61-14" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb61-15"><a href="#cb61-15" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb61-16"><a href="#cb61-16" tabindex="-1"></a>    <span class="at">Total =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(total_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(total_sd, <span class="dv">3</span>)),</span>
<span id="cb61-17"><a href="#cb61-17" tabindex="-1"></a>    <span class="at">Cropland =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(le_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(le_sd, <span class="dv">3</span>)),</span>
<span id="cb61-18"><a href="#cb61-18" tabindex="-1"></a>    <span class="at">Mixed_forest =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(ha_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(ha_sd, <span class="dv">3</span>)),</span>
<span id="cb61-19"><a href="#cb61-19" tabindex="-1"></a>    <span class="at">Xeric_shrubland =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(er_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(er_sd, <span class="dv">3</span>)),</span>
<span id="cb61-20"><a href="#cb61-20" tabindex="-1"></a>    <span class="at">Temperate_woodland =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(go_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(go_sd, <span class="dv">3</span>))</span>
<span id="cb61-21"><a href="#cb61-21" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb61-22"><a href="#cb61-22" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>total_mean) <span class="sc">%&gt;%</span></span>
<span id="cb61-23"><a href="#cb61-23" tabindex="-1"></a>  <span class="fu">select</span>(phylum, Total, Cropland, Temperate_woodland, Mixed_forest, Xeric_shrubland)</span></code></pre></div>
<pre><code>## # A tibble: 15 × 6
##    phylum               Total         Cropland      Temperate_woodland Mixed_forest  Xeric_shrubland
##    &lt;chr&gt;                &lt;chr&gt;         &lt;chr&gt;         &lt;chr&gt;              &lt;chr&gt;         &lt;chr&gt;          
##  1 p__Bacillota_A       60.713±17.816 58.875±10.43  63.149±17.119      34.394±18.002 57.552±18.161  
##  2 p__Bacteroidota      15.337±12.756 15.481±9.835  14.668±12.774      10.876±11.507 26.428±17.109  
##  3 p__Bacillota         10.651±15.13  14.427±14.322 8.561±12.074       38.211±25.419 2.336±1.65     
##  4 p__Synergistota      2.856±1.778   2.215±1.373   2.877±1.824        4.395±1.665   3.122±2.054    
##  5 p__Cyanobacteriota   2.624±3.286   2.238±2.31    2.907±3.671        3.894±2.743   0.326±0.377    
##  6 p__Verrucomicrobiota 2.074±1.467   1.534±0.894   1.998±1.116        4.934±2.723   2.134±0.872    
##  7 p__Pseudomonadota    1.207±2.219   0.769±0.588   1.405±2.689        0.769±0.157   0.558±0.257    
##  8 p__Bacillota_C       1.133±0.683   1.154±0.58    1.138±0.762        1.145±0.526   0.894±0.286    
##  9 p__Thermoplasmatota  0.868±1.487   1.874±2.445   0.475±1.01         0.209±0.204   2.817±0.979    
## 10 p__Campylobacterota  0.866±1.568   0.776±0.657   0.746±1.234        0.611±0.499   2.261±4.129    
## 11 p__Actinomycetota    0.749±1.478   0.108±0.227   1.036±1.75         0.286±0.281   0.278±0.522    
## 12 p__Desulfobacterota  0.681±0.697   0.483±0.586   0.773±0.768        0.275±0.289   0.653±0.615    
## 13 p__Spirochaetota     0.189±0.607   0±0           0.211±0.697        0±0           0.62±0.635     
## 14 p__Methanobacteriota 0.03±0.061    0.036±0.046   0.037±0.071        0±0           0±0            
## 15 p__Bacillota_B       0.02±0.043    0.032±0.045   0.017±0.045        0±0           0.022±0.028</code></pre>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" tabindex="-1"></a>phylum_summary <span class="sc">%&gt;%</span></span>
<span id="cb63-2"><a href="#cb63-2" tabindex="-1"></a>  <span class="fu">group_by</span>(phylum) <span class="sc">%&gt;%</span></span>
<span id="cb63-3"><a href="#cb63-3" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb63-4"><a href="#cb63-4" tabindex="-1"></a>    <span class="at">total_mean =</span> <span class="fu">mean</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb63-5"><a href="#cb63-5" tabindex="-1"></a>    <span class="at">total_sd =</span> <span class="fu">sd</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb63-6"><a href="#cb63-6" tabindex="-1"></a>    <span class="at">le_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>(sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb63-7"><a href="#cb63-7" tabindex="-1"></a>    <span class="at">le_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb63-8"><a href="#cb63-8" tabindex="-1"></a>    <span class="at">ha_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>( sex<span class="sc">==</span> <span class="st">&quot;Male&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb63-9"><a href="#cb63-9" tabindex="-1"></a>    <span class="at">ha_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb63-10"><a href="#cb63-10" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb63-11"><a href="#cb63-11" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb63-12"><a href="#cb63-12" tabindex="-1"></a>    <span class="at">Total =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(total_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(total_sd, <span class="dv">3</span>)),</span>
<span id="cb63-13"><a href="#cb63-13" tabindex="-1"></a>    <span class="at">Female =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(le_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(le_sd, <span class="dv">3</span>)),</span>
<span id="cb63-14"><a href="#cb63-14" tabindex="-1"></a>    <span class="at">Male =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(ha_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(ha_sd, <span class="dv">3</span>))</span>
<span id="cb63-15"><a href="#cb63-15" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb63-16"><a href="#cb63-16" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>total_mean) <span class="sc">%&gt;%</span></span>
<span id="cb63-17"><a href="#cb63-17" tabindex="-1"></a>  <span class="fu">select</span>(phylum, Total, Female, Male)</span></code></pre></div>
<pre><code>## # A tibble: 15 × 4
##    phylum               Total         Female        Male         
##    &lt;chr&gt;                &lt;chr&gt;         &lt;chr&gt;         &lt;chr&gt;        
##  1 p__Bacillota_A       60.713±17.816 59.012±21.422 63.183±14.871
##  2 p__Bacteroidota      15.337±12.756 14.648±11.721 13.48±11.248 
##  3 p__Bacillota         10.651±15.13  12.942±17.914 10.184±13.886
##  4 p__Synergistota      2.856±1.778   2.967±1.775   2.65±1.655   
##  5 p__Cyanobacteriota   2.624±3.286   2.659±3.026   2.965±3.089  
##  6 p__Verrucomicrobiota 2.074±1.467   2.067±1.288   2.274±1.798  
##  7 p__Pseudomonadota    1.207±2.219   1.234±1.949   0.951±0.784  
##  8 p__Bacillota_C       1.133±0.683   1.37±0.897    1.025±0.503  
##  9 p__Thermoplasmatota  0.868±1.487   0.766±1.708   0.723±1.287  
## 10 p__Campylobacterota  0.866±1.568   0.53±0.754    0.991±1.341  
## 11 p__Actinomycetota    0.749±1.478   1.128±2.008   0.609±1.113  
## 12 p__Desulfobacterota  0.681±0.697   0.615±0.587   0.652±0.486  
## 13 p__Spirochaetota     0.189±0.607   0±0           0.277±0.83   
## 14 p__Methanobacteriota 0.03±0.061    0.044±0.069   0.013±0.025  
## 15 p__Bacillota_B       0.02±0.043    0.018±0.041   0.023±0.05</code></pre>
</div>
<div id="family-level" class="section level4 hasAnchor" number="3.1.1.2">
<h4 class="hasAnchor"><span class="header-section-number">3.1.1.2</span> Family level<a href="#family-level" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong><em>Percentange of families in each group of environment</em></strong></p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" tabindex="-1"></a>family_summary <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb65-2"><a href="#cb65-2" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>genome),<span class="sc">~</span>.<span class="sc">/</span><span class="fu">sum</span>(.)) <span class="sc">%&gt;%</span> <span class="co">#apply TSS nornalisation</span></span>
<span id="cb65-3"><a href="#cb65-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>genome, <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;count&quot;</span>) <span class="sc">%&gt;%</span> <span class="co">#reduce to minimum number of columns</span></span>
<span id="cb65-4"><a href="#cb65-4" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span> <span class="co">#append sample metadata</span></span>
<span id="cb65-5"><a href="#cb65-5" tabindex="-1"></a>  <span class="fu">left_join</span>(genome_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(genome <span class="sc">==</span> genome)) <span class="sc">%&gt;%</span> <span class="co">#append genome metadata</span></span>
<span id="cb65-6"><a href="#cb65-6" tabindex="-1"></a>  <span class="fu">group_by</span>(sample,family, broad_environment, sex) <span class="sc">%&gt;%</span></span>
<span id="cb65-7"><a href="#cb65-7" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">relabun=</span><span class="fu">sum</span>(count))</span></code></pre></div>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" tabindex="-1"></a>family_summary <span class="sc">%&gt;%</span></span>
<span id="cb66-2"><a href="#cb66-2" tabindex="-1"></a>    <span class="fu">group_by</span>(family) <span class="sc">%&gt;%</span></span>
<span id="cb66-3"><a href="#cb66-3" tabindex="-1"></a>    <span class="fu">summarise</span>(</span>
<span id="cb66-4"><a href="#cb66-4" tabindex="-1"></a>    <span class="at">total_mean =</span> <span class="fu">mean</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb66-5"><a href="#cb66-5" tabindex="-1"></a>    <span class="at">total_sd =</span> <span class="fu">sd</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb66-6"><a href="#cb66-6" tabindex="-1"></a>    <span class="at">le_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000245 - Cropland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb66-7"><a href="#cb66-7" tabindex="-1"></a>    <span class="at">le_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000245 - Cropland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb66-8"><a href="#cb66-8" tabindex="-1"></a>    <span class="at">ha_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000198 - Mixed forest&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb66-9"><a href="#cb66-9" tabindex="-1"></a>    <span class="at">ha_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000198 - Mixed forest&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb66-10"><a href="#cb66-10" tabindex="-1"></a>    <span class="at">er_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000218 - Xeric shrubland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb66-11"><a href="#cb66-11" tabindex="-1"></a>    <span class="at">er_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000218 - Xeric shrubland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb66-12"><a href="#cb66-12" tabindex="-1"></a>    <span class="at">go_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000221 - Temperate woodland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb66-13"><a href="#cb66-13" tabindex="-1"></a>    <span class="at">go_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(broad_environment <span class="sc">==</span> <span class="st">&quot;1000221 - Temperate woodland&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb66-14"><a href="#cb66-14" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb66-15"><a href="#cb66-15" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb66-16"><a href="#cb66-16" tabindex="-1"></a>    <span class="at">Total =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(total_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(total_sd, <span class="dv">3</span>)),</span>
<span id="cb66-17"><a href="#cb66-17" tabindex="-1"></a>    <span class="at">Cropland =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(le_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(le_sd, <span class="dv">3</span>)),</span>
<span id="cb66-18"><a href="#cb66-18" tabindex="-1"></a>    <span class="at">Mixed_forest =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(ha_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(ha_sd, <span class="dv">3</span>)),</span>
<span id="cb66-19"><a href="#cb66-19" tabindex="-1"></a>    <span class="at">Xeric_shrubland =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(er_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(er_sd, <span class="dv">3</span>)),</span>
<span id="cb66-20"><a href="#cb66-20" tabindex="-1"></a>    <span class="at">Temperate_woodland =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(go_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(go_sd, <span class="dv">3</span>))</span>
<span id="cb66-21"><a href="#cb66-21" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb66-22"><a href="#cb66-22" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(family,Total,Cropland, Temperate_woodland, Mixed_forest, Xeric_shrubland)</span></code></pre></div>
<pre><code>## # A tibble: 52 × 6
##    family                Total         Cropland     Temperate_woodland Mixed_forest Xeric_shrubland
##    &lt;chr&gt;                 &lt;chr&gt;         &lt;chr&gt;        &lt;chr&gt;              &lt;chr&gt;        &lt;chr&gt;          
##  1 f__                   0.022±0.064   0.002±0.008  0.033±0.077        0.006±0.014  0±0            
##  2 f__Acidaminococcaceae 1.116±0.658   1.154±0.58   1.112±0.727        1.145±0.526  0.894±0.286    
##  3 f__Acutalibacteraceae 0.437±0.449   0.308±0.215  0.501±0.512        0.204±0.372  0.449±0.299    
##  4 f__Akkermansiaceae    1.932±1.499   1.534±0.894  1.887±1.118        4.934±2.723  1.301±1.184    
##  5 f__Anaerotignaceae    0.044±0.043   0.044±0.041  0.042±0.041        0.005±0.011  0.074±0.067    
##  6 f__Anaerovoracaceae   0.022±0.05    0.009±0.02   0.02±0.042         0.112±0.11   0±0            
##  7 f__Bacteroidaceae     10.696±11.055 9.816±5.594  9.817±11.399       9.473±10.239 22.416±13.948  
##  8 f__Borkfalkiaceae     7.439±5.295   11.275±5.806 7.088±5.26         3.208±2.553  5.841±3.717    
##  9 f__Burkholderiaceae   0.052±0.102   0.039±0.059  0.042±0.095        0.038±0.056  0.174±0.184    
## 10 f__Burkholderiaceae_A 0.577±0.56    0.523±0.471  0.58±0.637         0.699±0.226  0.383±0.242    
## # ℹ 42 more rows</code></pre>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" tabindex="-1"></a>family_arrange <span class="ot">&lt;-</span> family_summary <span class="sc">%&gt;%</span></span>
<span id="cb68-2"><a href="#cb68-2" tabindex="-1"></a>    <span class="fu">group_by</span>(family) <span class="sc">%&gt;%</span></span>
<span id="cb68-3"><a href="#cb68-3" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">sum</span>(relabun)) <span class="sc">%&gt;%</span></span>
<span id="cb68-4"><a href="#cb68-4" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>mean) <span class="sc">%&gt;%</span></span>
<span id="cb68-5"><a href="#cb68-5" tabindex="-1"></a>    <span class="fu">select</span>(family) <span class="sc">%&gt;%</span></span>
<span id="cb68-6"><a href="#cb68-6" tabindex="-1"></a>    <span class="fu">pull</span>()</span>
<span id="cb68-7"><a href="#cb68-7" tabindex="-1"></a></span>
<span id="cb68-8"><a href="#cb68-8" tabindex="-1"></a><span class="co"># Per environment</span></span>
<span id="cb68-9"><a href="#cb68-9" tabindex="-1"></a>family_summary <span class="sc">%&gt;%</span></span>
<span id="cb68-10"><a href="#cb68-10" tabindex="-1"></a>    <span class="fu">left_join</span>(genome_metadata <span class="sc">%&gt;%</span> <span class="fu">select</span>(family,phylum) <span class="sc">%&gt;%</span> <span class="fu">unique</span>(),<span class="at">by=</span><span class="fu">join_by</span>(family<span class="sc">==</span>family)) <span class="sc">%&gt;%</span></span>
<span id="cb68-11"><a href="#cb68-11" tabindex="-1"></a>    <span class="fu">filter</span>(family <span class="sc">%in%</span> family_arrange[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb68-12"><a href="#cb68-12" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">family=</span><span class="fu">factor</span>(family,<span class="at">levels=</span><span class="fu">rev</span>(family_arrange[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]))) <span class="sc">%&gt;%</span></span>
<span id="cb68-13"><a href="#cb68-13" tabindex="-1"></a>    <span class="fu">filter</span>(relabun <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb68-14"><a href="#cb68-14" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>relabun, <span class="at">y=</span>family, <span class="at">group=</span>family, <span class="at">color=</span>phylum)) <span class="sc">+</span></span>
<span id="cb68-15"><a href="#cb68-15" tabindex="-1"></a>        <span class="fu">scale_color_manual</span>(<span class="at">values=</span>phylum_colors[<span class="sc">-</span><span class="dv">8</span>]) <span class="sc">+</span></span>
<span id="cb68-16"><a href="#cb68-16" tabindex="-1"></a>        <span class="fu">geom_jitter</span>(<span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb68-17"><a href="#cb68-17" tabindex="-1"></a>        <span class="fu">facet_grid</span>(.<span class="sc">~</span>broad_environment)<span class="sc">+</span></span>
<span id="cb68-18"><a href="#cb68-18" tabindex="-1"></a>        <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb68-19"><a href="#cb68-19" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Family&quot;</span>, <span class="at">x=</span><span class="st">&quot;Relative abundance&quot;</span>, <span class="at">color=</span><span class="st">&quot;Phylum&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/taxonomy_jitterplot_family-1.png" width="960" /></p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" tabindex="-1"></a>family_summary <span class="sc">%&gt;%</span></span>
<span id="cb69-2"><a href="#cb69-2" tabindex="-1"></a>    <span class="fu">group_by</span>(family) <span class="sc">%&gt;%</span></span>
<span id="cb69-3"><a href="#cb69-3" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">total_mean=</span><span class="fu">mean</span>(relabun<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb69-4"><a href="#cb69-4" tabindex="-1"></a>              <span class="at">total_sd=</span><span class="fu">sd</span>(relabun<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb69-5"><a href="#cb69-5" tabindex="-1"></a>              <span class="at">le_mean=</span><span class="fu">mean</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000245 - Cropland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb69-6"><a href="#cb69-6" tabindex="-1"></a>             <span class="at">le_sd=</span><span class="fu">sd</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000245 - Cropland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb69-7"><a href="#cb69-7" tabindex="-1"></a>              <span class="at">ha_mean=</span><span class="fu">mean</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000198 - Mixed forest&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb69-8"><a href="#cb69-8" tabindex="-1"></a>              <span class="at">ha_sd=</span><span class="fu">sd</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000198 - Mixed forest&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb69-9"><a href="#cb69-9" tabindex="-1"></a>              <span class="at">er_mean=</span><span class="fu">mean</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000218 - Xeric shrubland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb69-10"><a href="#cb69-10" tabindex="-1"></a>              <span class="at">er_sd=</span><span class="fu">sd</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000218 - Xeric shrubland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb69-11"><a href="#cb69-11" tabindex="-1"></a>              <span class="at">go_mean=</span><span class="fu">mean</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000221 - Temperate woodland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb69-12"><a href="#cb69-12" tabindex="-1"></a>              <span class="at">go_sd=</span><span class="fu">sd</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000221 - Temperate woodland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T)) <span class="sc">%&gt;%</span></span>
<span id="cb69-13"><a href="#cb69-13" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Total=</span><span class="fu">str_c</span>(<span class="fu">round</span>(total_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(total_sd,<span class="dv">3</span>)),</span>
<span id="cb69-14"><a href="#cb69-14" tabindex="-1"></a>          <span class="at">Cropland=</span><span class="fu">str_c</span>(<span class="fu">round</span>(le_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(le_sd,<span class="dv">3</span>)),</span>
<span id="cb69-15"><a href="#cb69-15" tabindex="-1"></a>          <span class="at">Mixed_forest=</span><span class="fu">str_c</span>(<span class="fu">round</span>(ha_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(ha_sd,<span class="dv">3</span>)),</span>
<span id="cb69-16"><a href="#cb69-16" tabindex="-1"></a>          <span class="at">Xeric_shrubland=</span><span class="fu">str_c</span>(<span class="fu">round</span>(er_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(er_sd,<span class="dv">3</span>)),</span>
<span id="cb69-17"><a href="#cb69-17" tabindex="-1"></a>          <span class="at">Temperate_woodland=</span><span class="fu">str_c</span>(<span class="fu">round</span>(go_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(go_sd,<span class="dv">3</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb69-18"><a href="#cb69-18" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>total_mean) <span class="sc">%&gt;%</span> </span>
<span id="cb69-19"><a href="#cb69-19" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(family,Total,Cropland,Temperate_woodland, Mixed_forest,Xeric_shrubland)</span></code></pre></div>
<pre><code>## # A tibble: 52 × 6
##    family                     Total         Cropland      Temperate_woodland Mixed_forest  Xeric_shrubland
##    &lt;chr&gt;                      &lt;chr&gt;         &lt;chr&gt;         &lt;chr&gt;              &lt;chr&gt;         &lt;chr&gt;          
##  1 f__Lachnospiraceae         28.205±9.998  25.865±6.414  29.657±10.487      22.929±12.444 22.276±6.399   
##  2 f__Oscillospiraceae        11.223±8.158  6.852±2.912   12.307±8.693       4.256±3.422   17.81±7.479    
##  3 f__Bacteroidaceae          10.696±11.055 9.816±5.594   9.817±11.399       9.473±10.239  22.416±13.948  
##  4 f__CAG-288                 8.635±15.251  11.701±14.385 6.449±11.978       37.695±25.73  0.245±0.538    
##  5 f__Borkfalkiaceae          7.439±5.295   11.275±5.806  7.088±5.26         3.208±2.553   5.841±3.717    
##  6 f__Ruminococcaceae         5.429±4.762   6.017±5.302   5.389±4.49         1.722±1.147   3.322±2.259    
##  7 f__CAG-508                 4.042±4.359   4.664±4.633   4.107±4.212        0.146±0.267   3.826±3.273    
##  8 f__Dethiosulfovibrionaceae 2.454±1.626   2.049±1.294   2.405±1.661        3.993±1.662   2.626±1.872    
##  9 f__Muribaculaceae          2.053±4.249   2.813±5.308   2.341±4.499        0.386±0.863   0±0            
## 10 f__UBA932                  1.966±3.772   2.294±3.61    1.763±3.655        0.915±2.046   3.7±6.208      
## # ℹ 42 more rows</code></pre>
<p><strong><em>Family summary grouped by sex</em></strong></p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" tabindex="-1"></a>family_summary <span class="sc">%&gt;%</span></span>
<span id="cb71-2"><a href="#cb71-2" tabindex="-1"></a>    <span class="fu">group_by</span>(family) <span class="sc">%&gt;%</span></span>
<span id="cb71-3"><a href="#cb71-3" tabindex="-1"></a>   <span class="fu">summarise</span>(</span>
<span id="cb71-4"><a href="#cb71-4" tabindex="-1"></a>    <span class="at">total_mean =</span> <span class="fu">mean</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb71-5"><a href="#cb71-5" tabindex="-1"></a>    <span class="at">total_sd =</span> <span class="fu">sd</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb71-6"><a href="#cb71-6" tabindex="-1"></a>    <span class="at">le_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>(sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb71-7"><a href="#cb71-7" tabindex="-1"></a>    <span class="at">le_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb71-8"><a href="#cb71-8" tabindex="-1"></a>    <span class="at">ha_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>( sex<span class="sc">==</span> <span class="st">&quot;Male&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb71-9"><a href="#cb71-9" tabindex="-1"></a>    <span class="at">ha_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb71-10"><a href="#cb71-10" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb71-11"><a href="#cb71-11" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb71-12"><a href="#cb71-12" tabindex="-1"></a>    <span class="at">Total =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(total_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(total_sd, <span class="dv">3</span>)),</span>
<span id="cb71-13"><a href="#cb71-13" tabindex="-1"></a>    <span class="at">Female =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(le_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(le_sd, <span class="dv">3</span>)),</span>
<span id="cb71-14"><a href="#cb71-14" tabindex="-1"></a>    <span class="at">Male =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(ha_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(ha_sd, <span class="dv">3</span>))</span>
<span id="cb71-15"><a href="#cb71-15" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb71-16"><a href="#cb71-16" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>total_mean) <span class="sc">%&gt;%</span></span>
<span id="cb71-17"><a href="#cb71-17" tabindex="-1"></a>  <span class="fu">select</span>(family, Total, Female, Male)</span></code></pre></div>
<pre><code>## # A tibble: 52 × 4
##    family                     Total         Female        Male        
##    &lt;chr&gt;                      &lt;chr&gt;         &lt;chr&gt;         &lt;chr&gt;       
##  1 f__Lachnospiraceae         28.205±9.998  29.818±13.448 28.926±7.093
##  2 f__Oscillospiraceae        11.223±8.158  8.451±7.695   11.752±6.237
##  3 f__Bacteroidaceae          10.696±11.055 9.231±9.11    9.336±9.589 
##  4 f__CAG-288                 8.635±15.251  11.469±18.156 7.94±13.683 
##  5 f__Borkfalkiaceae          7.439±5.295   7.529±6.085   7.343±4.351 
##  6 f__Ruminococcaceae         5.429±4.762   5.14±3.989    6.359±5.72  
##  7 f__CAG-508                 4.042±4.359   3.495±4.278   4.605±4.655 
##  8 f__Dethiosulfovibrionaceae 2.454±1.626   2.599±1.648   2.294±1.494 
##  9 f__Muribaculaceae          2.053±4.249   2.968±5.555   1.326±2.19  
## 10 f__UBA932                  1.966±3.772   1.695±3.403   2.367±3.922 
## # ℹ 42 more rows</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" tabindex="-1"></a>family_arrange <span class="ot">&lt;-</span> family_summary <span class="sc">%&gt;%</span></span>
<span id="cb73-2"><a href="#cb73-2" tabindex="-1"></a>    <span class="fu">group_by</span>(family) <span class="sc">%&gt;%</span></span>
<span id="cb73-3"><a href="#cb73-3" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">sum</span>(relabun)) <span class="sc">%&gt;%</span></span>
<span id="cb73-4"><a href="#cb73-4" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>mean) <span class="sc">%&gt;%</span></span>
<span id="cb73-5"><a href="#cb73-5" tabindex="-1"></a>    <span class="fu">select</span>(family) <span class="sc">%&gt;%</span></span>
<span id="cb73-6"><a href="#cb73-6" tabindex="-1"></a>    <span class="fu">pull</span>()</span>
<span id="cb73-7"><a href="#cb73-7" tabindex="-1"></a></span>
<span id="cb73-8"><a href="#cb73-8" tabindex="-1"></a><span class="co"># Per environment</span></span>
<span id="cb73-9"><a href="#cb73-9" tabindex="-1"></a>family_summary <span class="sc">%&gt;%</span></span>
<span id="cb73-10"><a href="#cb73-10" tabindex="-1"></a>    <span class="fu">left_join</span>(genome_metadata <span class="sc">%&gt;%</span> <span class="fu">select</span>(family,phylum) <span class="sc">%&gt;%</span> <span class="fu">unique</span>(),<span class="at">by=</span><span class="fu">join_by</span>(family<span class="sc">==</span>family)) <span class="sc">%&gt;%</span></span>
<span id="cb73-11"><a href="#cb73-11" tabindex="-1"></a>    <span class="fu">filter</span>(family <span class="sc">%in%</span> family_arrange[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb73-12"><a href="#cb73-12" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">family=</span><span class="fu">factor</span>(family,<span class="at">levels=</span><span class="fu">rev</span>(family_arrange[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]))) <span class="sc">%&gt;%</span></span>
<span id="cb73-13"><a href="#cb73-13" tabindex="-1"></a>    <span class="fu">filter</span>(relabun <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb73-14"><a href="#cb73-14" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>relabun, <span class="at">y=</span>family, <span class="at">group=</span>family, <span class="at">color=</span>phylum)) <span class="sc">+</span></span>
<span id="cb73-15"><a href="#cb73-15" tabindex="-1"></a>        <span class="fu">scale_color_manual</span>(<span class="at">values=</span>phylum_colors[<span class="sc">-</span><span class="dv">8</span>]) <span class="sc">+</span></span>
<span id="cb73-16"><a href="#cb73-16" tabindex="-1"></a>        <span class="fu">geom_jitter</span>(<span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb73-17"><a href="#cb73-17" tabindex="-1"></a>        <span class="fu">facet_grid</span>(.<span class="sc">~</span>sex)<span class="sc">+</span></span>
<span id="cb73-18"><a href="#cb73-18" tabindex="-1"></a>        <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb73-19"><a href="#cb73-19" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Family&quot;</span>, <span class="at">x=</span><span class="st">&quot;Relative abundance&quot;</span>, <span class="at">color=</span><span class="st">&quot;Phylum&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/taxonomy_jitterplot_family_sex-1.png" width="960" /></p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" tabindex="-1"></a>family_summary <span class="sc">%&gt;%</span></span>
<span id="cb74-2"><a href="#cb74-2" tabindex="-1"></a>    <span class="fu">group_by</span>(family) <span class="sc">%&gt;%</span></span>
<span id="cb74-3"><a href="#cb74-3" tabindex="-1"></a>    <span class="fu">summarise</span>(</span>
<span id="cb74-4"><a href="#cb74-4" tabindex="-1"></a>    <span class="at">total_mean =</span> <span class="fu">mean</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb74-5"><a href="#cb74-5" tabindex="-1"></a>    <span class="at">total_sd =</span> <span class="fu">sd</span>(relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb74-6"><a href="#cb74-6" tabindex="-1"></a>    <span class="at">le_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>(sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb74-7"><a href="#cb74-7" tabindex="-1"></a>    <span class="at">le_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(sex <span class="sc">==</span> <span class="st">&quot;Female&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb74-8"><a href="#cb74-8" tabindex="-1"></a>    <span class="at">ha_mean =</span> <span class="fu">mean</span>(<span class="fu">ifelse</span>( sex<span class="sc">==</span> <span class="st">&quot;Male&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb74-9"><a href="#cb74-9" tabindex="-1"></a>    <span class="at">ha_sd =</span> <span class="fu">sd</span>(<span class="fu">ifelse</span>(sex <span class="sc">==</span> <span class="st">&quot;Male&quot;</span>, relabun <span class="sc">*</span> <span class="dv">100</span>, <span class="cn">NA</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb74-10"><a href="#cb74-10" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb74-11"><a href="#cb74-11" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb74-12"><a href="#cb74-12" tabindex="-1"></a>    <span class="at">Total =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(total_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(total_sd, <span class="dv">3</span>)),</span>
<span id="cb74-13"><a href="#cb74-13" tabindex="-1"></a>    <span class="at">Female =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(le_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(le_sd, <span class="dv">3</span>)),</span>
<span id="cb74-14"><a href="#cb74-14" tabindex="-1"></a>    <span class="at">Male =</span> <span class="fu">str_c</span>(<span class="fu">round</span>(ha_mean, <span class="dv">3</span>), <span class="st">&quot;±&quot;</span>, <span class="fu">round</span>(ha_sd, <span class="dv">3</span>))</span>
<span id="cb74-15"><a href="#cb74-15" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb74-16"><a href="#cb74-16" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>total_mean) <span class="sc">%&gt;%</span></span>
<span id="cb74-17"><a href="#cb74-17" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(family,Total,Female, Male)</span></code></pre></div>
<pre><code>## # A tibble: 52 × 4
##    family                     Total         Female        Male        
##    &lt;chr&gt;                      &lt;chr&gt;         &lt;chr&gt;         &lt;chr&gt;       
##  1 f__Lachnospiraceae         28.205±9.998  29.818±13.448 28.926±7.093
##  2 f__Oscillospiraceae        11.223±8.158  8.451±7.695   11.752±6.237
##  3 f__Bacteroidaceae          10.696±11.055 9.231±9.11    9.336±9.589 
##  4 f__CAG-288                 8.635±15.251  11.469±18.156 7.94±13.683 
##  5 f__Borkfalkiaceae          7.439±5.295   7.529±6.085   7.343±4.351 
##  6 f__Ruminococcaceae         5.429±4.762   5.14±3.989    6.359±5.72  
##  7 f__CAG-508                 4.042±4.359   3.495±4.278   4.605±4.655 
##  8 f__Dethiosulfovibrionaceae 2.454±1.626   2.599±1.648   2.294±1.494 
##  9 f__Muribaculaceae          2.053±4.249   2.968±5.555   1.326±2.19  
## 10 f__UBA932                  1.966±3.772   1.695±3.403   2.367±3.922 
## # ℹ 42 more rows</code></pre>
</div>
<div id="genus-level" class="section level4 hasAnchor" number="3.1.1.3">
<h4 class="hasAnchor"><span class="header-section-number">3.1.1.3</span> Genus level<a href="#genus-level" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>*** Percetange of genera in each group of environment***</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" tabindex="-1"></a>genus_summary <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb76-2"><a href="#cb76-2" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>genome),<span class="sc">~</span>.<span class="sc">/</span><span class="fu">sum</span>(.)) <span class="sc">%&gt;%</span> <span class="co">#apply TSS nornalisation</span></span>
<span id="cb76-3"><a href="#cb76-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>genome, <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;count&quot;</span>) <span class="sc">%&gt;%</span> <span class="co">#reduce to minimum number of columns</span></span>
<span id="cb76-4"><a href="#cb76-4" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span> <span class="co">#append sample metadata</span></span>
<span id="cb76-5"><a href="#cb76-5" tabindex="-1"></a>  <span class="fu">left_join</span>(genome_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(genome <span class="sc">==</span> genome)) <span class="sc">%&gt;%</span> <span class="co">#append genome metadata</span></span>
<span id="cb76-6"><a href="#cb76-6" tabindex="-1"></a>  <span class="fu">group_by</span>(sample,phylum,genus,broad_environment) <span class="sc">%&gt;%</span></span>
<span id="cb76-7"><a href="#cb76-7" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">relabun=</span><span class="fu">sum</span>(count)) </span>
<span id="cb76-8"><a href="#cb76-8" tabindex="-1"></a><span class="co"># %&gt;%</span></span>
<span id="cb76-9"><a href="#cb76-9" tabindex="-1"></a><span class="co">#   filter(genus != &quot;g__&quot;) %&gt;%</span></span>
<span id="cb76-10"><a href="#cb76-10" tabindex="-1"></a><span class="co">#   mutate(genus= sub(&quot;^g__&quot;, &quot;&quot;, genus))</span></span>
<span id="cb76-11"><a href="#cb76-11" tabindex="-1"></a></span>
<span id="cb76-12"><a href="#cb76-12" tabindex="-1"></a>genus_summary <span class="sc">%&gt;%</span></span>
<span id="cb76-13"><a href="#cb76-13" tabindex="-1"></a>    <span class="fu">group_by</span>(genus) <span class="sc">%&gt;%</span></span>
<span id="cb76-14"><a href="#cb76-14" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">total_mean=</span><span class="fu">mean</span>(relabun<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb76-15"><a href="#cb76-15" tabindex="-1"></a>              <span class="at">total_sd=</span><span class="fu">sd</span>(relabun<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb76-16"><a href="#cb76-16" tabindex="-1"></a>              <span class="at">le_mean=</span><span class="fu">mean</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000245 - Cropland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb76-17"><a href="#cb76-17" tabindex="-1"></a>             <span class="at">le_sd=</span><span class="fu">sd</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000245 - Cropland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb76-18"><a href="#cb76-18" tabindex="-1"></a>              <span class="at">ha_mean=</span><span class="fu">mean</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000198 - Mixed forest&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb76-19"><a href="#cb76-19" tabindex="-1"></a>              <span class="at">ha_sd=</span><span class="fu">sd</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000198 - Mixed forest&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb76-20"><a href="#cb76-20" tabindex="-1"></a>              <span class="at">er_mean=</span><span class="fu">mean</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000218 - Xeric shrubland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb76-21"><a href="#cb76-21" tabindex="-1"></a>              <span class="at">er_sd=</span><span class="fu">sd</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000218 - Xeric shrubland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb76-22"><a href="#cb76-22" tabindex="-1"></a>              <span class="at">go_mean=</span><span class="fu">mean</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000221 - Temperate woodland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb76-23"><a href="#cb76-23" tabindex="-1"></a>              <span class="at">go_sd=</span><span class="fu">sd</span>(relabun[broad_environment<span class="sc">==</span><span class="st">&quot;1000221 - Temperate woodland&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T)) <span class="sc">%&gt;%</span></span>
<span id="cb76-24"><a href="#cb76-24" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Total=</span><span class="fu">str_c</span>(<span class="fu">round</span>(total_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(total_sd,<span class="dv">3</span>)),</span>
<span id="cb76-25"><a href="#cb76-25" tabindex="-1"></a>          <span class="at">Cropland=</span><span class="fu">str_c</span>(<span class="fu">round</span>(le_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(le_sd,<span class="dv">3</span>)),</span>
<span id="cb76-26"><a href="#cb76-26" tabindex="-1"></a>          <span class="at">Mixed_forest=</span><span class="fu">str_c</span>(<span class="fu">round</span>(ha_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(ha_sd,<span class="dv">3</span>)),</span>
<span id="cb76-27"><a href="#cb76-27" tabindex="-1"></a>          <span class="at">Xeric_shrubland=</span><span class="fu">str_c</span>(<span class="fu">round</span>(er_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(er_sd,<span class="dv">3</span>)),</span>
<span id="cb76-28"><a href="#cb76-28" tabindex="-1"></a>          <span class="at">Temperate_woodland=</span><span class="fu">str_c</span>(<span class="fu">round</span>(go_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(go_sd,<span class="dv">3</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb76-29"><a href="#cb76-29" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>total_mean) <span class="sc">%&gt;%</span> </span>
<span id="cb76-30"><a href="#cb76-30" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(genus,Total, Cropland,Temperate_woodland, Mixed_forest,Xeric_shrubland) </span></code></pre></div>
<pre><code>## # A tibble: 173 × 6
##    genus            Total        Cropland      Temperate_woodland Mixed_forest Xeric_shrubland
##    &lt;chr&gt;            &lt;chr&gt;        &lt;chr&gt;         &lt;chr&gt;              &lt;chr&gt;        &lt;chr&gt;          
##  1 g__Prevotella    9.081±10.534 9.381±5.844   8.141±10.756       4.235±6.183  21.397±13.482  
##  2 g__Enterosoma    8.635±15.251 11.701±14.385 6.449±11.978       37.695±25.73 0.245±0.538    
##  3 g__Coproplasma   6.807±5.002  10.681±5.315  6.328±4.94         3.196±2.54   5.313±3.765    
##  4 g__RGIG6307      5.319±4.84   4.127±2.083   5.707±5.151        0.477±0.289  6.745±2.756    
##  5 g__Onthomonas    4.353±3.228  3.628±2.036   4.237±3.041        2.906±2.533  7.746±5.474    
##  6 g__Ventrimonas   4.342±5.43   3.449±4.552   4.925±6.094        5.197±3.825  0.795±0.458    
##  7 g__Eubacterium_I 3.269±2.549  2.489±3.267   3.597±2.571        2.763±2.006  2.887±1.925    
##  8 g__Choladousia   2.622±1.668  2.029±0.97    2.638±1.7          1.956±1.192  2.869±1.766    
##  9 g__Faeciplasma   2.6±3.197    3.158±4.909   2.748±3.049        1.153±0.718  0.746±1.353    
## 10 g__Agathobacter  2.501±2.807  1.917±2.231   2.575±3.063        2.888±1.871  1.333±1.002    
## # ℹ 163 more rows</code></pre>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" tabindex="-1"></a>genus_summary_sort <span class="ot">&lt;-</span> genus_summary <span class="sc">%&gt;%</span></span>
<span id="cb78-2"><a href="#cb78-2" tabindex="-1"></a>    <span class="fu">group_by</span>(genus) <span class="sc">%&gt;%</span></span>
<span id="cb78-3"><a href="#cb78-3" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(relabun, <span class="at">na.rm=</span>T),<span class="at">sd=</span><span class="fu">sd</span>(relabun, <span class="at">na.rm=</span>T)) <span class="sc">%&gt;%</span></span>
<span id="cb78-4"><a href="#cb78-4" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>mean) </span>
<span id="cb78-5"><a href="#cb78-5" tabindex="-1"></a></span>
<span id="cb78-6"><a href="#cb78-6" tabindex="-1"></a>genus_summary <span class="sc">%&gt;%</span></span>
<span id="cb78-7"><a href="#cb78-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genus=</span><span class="fu">factor</span>(genus, <span class="at">levels=</span><span class="fu">rev</span>(genus_summary_sort <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genus)))) <span class="sc">%&gt;%</span></span>
<span id="cb78-8"><a href="#cb78-8" tabindex="-1"></a>  <span class="fu">filter</span>(relabun <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb78-9"><a href="#cb78-9" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>relabun, <span class="at">y=</span>genus, <span class="at">group=</span>genus, <span class="at">color=</span>phylum)) <span class="sc">+</span></span>
<span id="cb78-10"><a href="#cb78-10" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>phylum_colors) <span class="sc">+</span></span>
<span id="cb78-11"><a href="#cb78-11" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb78-12"><a href="#cb78-12" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>broad_environment)<span class="sc">+</span></span>
<span id="cb78-13"><a href="#cb78-13" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb78-14"><a href="#cb78-14" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>))<span class="sc">+</span></span>
<span id="cb78-15"><a href="#cb78-15" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Family&quot;</span>, <span class="at">x=</span><span class="st">&quot;Relative abundance&quot;</span>, <span class="at">color=</span><span class="st">&quot;Phylum&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/taxonomy_jitterplot_genus-1.png" width="960" />
*** Percetange of genera in each group of sex***</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" tabindex="-1"></a>genus_summary <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb79-2"><a href="#cb79-2" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="sc">-</span>genome),<span class="sc">~</span>.<span class="sc">/</span><span class="fu">sum</span>(.)) <span class="sc">%&gt;%</span> <span class="co">#apply TSS nornalisation</span></span>
<span id="cb79-3"><a href="#cb79-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>genome, <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;count&quot;</span>) <span class="sc">%&gt;%</span> <span class="co">#reduce to minimum number of columns</span></span>
<span id="cb79-4"><a href="#cb79-4" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span> <span class="co">#append sample metadata</span></span>
<span id="cb79-5"><a href="#cb79-5" tabindex="-1"></a>  <span class="fu">left_join</span>(genome_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(genome <span class="sc">==</span> genome)) <span class="sc">%&gt;%</span> <span class="co">#append genome metadata</span></span>
<span id="cb79-6"><a href="#cb79-6" tabindex="-1"></a>  <span class="fu">group_by</span>(sample,phylum,genus,sex) <span class="sc">%&gt;%</span></span>
<span id="cb79-7"><a href="#cb79-7" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">relabun=</span><span class="fu">sum</span>(count)) </span>
<span id="cb79-8"><a href="#cb79-8" tabindex="-1"></a><span class="co"># %&gt;%</span></span>
<span id="cb79-9"><a href="#cb79-9" tabindex="-1"></a><span class="co">#   filter(genus != &quot;g__&quot;) %&gt;%</span></span>
<span id="cb79-10"><a href="#cb79-10" tabindex="-1"></a><span class="co">#   mutate(genus= sub(&quot;^g__&quot;, &quot;&quot;, genus))</span></span>
<span id="cb79-11"><a href="#cb79-11" tabindex="-1"></a></span>
<span id="cb79-12"><a href="#cb79-12" tabindex="-1"></a>genus_summary <span class="sc">%&gt;%</span></span>
<span id="cb79-13"><a href="#cb79-13" tabindex="-1"></a>    <span class="fu">group_by</span>(genus) <span class="sc">%&gt;%</span></span>
<span id="cb79-14"><a href="#cb79-14" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">total_mean=</span><span class="fu">mean</span>(relabun<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb79-15"><a href="#cb79-15" tabindex="-1"></a>              <span class="at">total_sd=</span><span class="fu">sd</span>(relabun<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb79-16"><a href="#cb79-16" tabindex="-1"></a>              <span class="at">le_mean=</span><span class="fu">mean</span>(relabun[sex<span class="sc">==</span><span class="st">&quot;Female&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb79-17"><a href="#cb79-17" tabindex="-1"></a>             <span class="at">le_sd=</span><span class="fu">sd</span>(relabun[sex<span class="sc">==</span><span class="st">&quot;Female&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb79-18"><a href="#cb79-18" tabindex="-1"></a>              <span class="at">ha_mean=</span><span class="fu">mean</span>(relabun[sex<span class="sc">==</span><span class="st">&quot;Male&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T),</span>
<span id="cb79-19"><a href="#cb79-19" tabindex="-1"></a>              <span class="at">ha_sd=</span><span class="fu">sd</span>(relabun[sex<span class="sc">==</span><span class="st">&quot;Male&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">na.rm=</span>T)) <span class="sc">%&gt;%</span></span>
<span id="cb79-20"><a href="#cb79-20" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Total=</span><span class="fu">str_c</span>(<span class="fu">round</span>(total_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(total_sd,<span class="dv">3</span>)),</span>
<span id="cb79-21"><a href="#cb79-21" tabindex="-1"></a>          <span class="at">Female=</span><span class="fu">str_c</span>(<span class="fu">round</span>(le_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(le_sd,<span class="dv">3</span>)),</span>
<span id="cb79-22"><a href="#cb79-22" tabindex="-1"></a>          <span class="at">Male=</span><span class="fu">str_c</span>(<span class="fu">round</span>(ha_mean,<span class="dv">3</span>),<span class="st">&quot;±&quot;</span>,<span class="fu">round</span>(ha_sd,<span class="dv">3</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb79-23"><a href="#cb79-23" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>total_mean) <span class="sc">%&gt;%</span> </span>
<span id="cb79-24"><a href="#cb79-24" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(genus,Total,Male,Female) </span></code></pre></div>
<pre><code>## # A tibble: 173 × 4
##    genus            Total        Male        Female       
##    &lt;chr&gt;            &lt;chr&gt;        &lt;chr&gt;       &lt;chr&gt;        
##  1 g__Prevotella    9.081±10.534 6.761±8.201 8.435±8.906  
##  2 g__Enterosoma    8.635±15.251 7.94±13.683 11.469±18.156
##  3 g__Coproplasma   6.807±5.002  6.852±4.084 6.752±5.903  
##  4 g__RGIG6307      5.319±4.84   6.178±5.285 4.467±5.172  
##  5 g__Onthomonas    4.353±3.228  5.021±2.992 3.351±2.586  
##  6 g__Ventrimonas   4.342±5.43   3.935±3.668 6.022±7.312  
##  7 g__Eubacterium_I 3.269±2.549  3.336±1.804 3.328±2.91   
##  8 g__Choladousia   2.622±1.668  3.109±1.545 2.266±1.754  
##  9 g__Faeciplasma   2.6±3.197    3.338±3.991 2.463±2.538  
## 10 g__Agathobacter  2.501±2.807  2.548±1.862 2.905±3.867  
## # ℹ 163 more rows</code></pre>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" tabindex="-1"></a>genus_summary_sort <span class="ot">&lt;-</span> genus_summary <span class="sc">%&gt;%</span></span>
<span id="cb81-2"><a href="#cb81-2" tabindex="-1"></a>    <span class="fu">group_by</span>(genus) <span class="sc">%&gt;%</span></span>
<span id="cb81-3"><a href="#cb81-3" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(relabun, <span class="at">na.rm=</span>T),<span class="at">sd=</span><span class="fu">sd</span>(relabun, <span class="at">na.rm=</span>T)) <span class="sc">%&gt;%</span></span>
<span id="cb81-4"><a href="#cb81-4" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>mean) </span>
<span id="cb81-5"><a href="#cb81-5" tabindex="-1"></a></span>
<span id="cb81-6"><a href="#cb81-6" tabindex="-1"></a>genus_summary <span class="sc">%&gt;%</span></span>
<span id="cb81-7"><a href="#cb81-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genus=</span><span class="fu">factor</span>(genus, <span class="at">levels=</span><span class="fu">rev</span>(genus_summary_sort <span class="sc">%&gt;%</span> <span class="fu">pull</span>(genus)))) <span class="sc">%&gt;%</span></span>
<span id="cb81-8"><a href="#cb81-8" tabindex="-1"></a>  <span class="fu">filter</span>(relabun <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb81-9"><a href="#cb81-9" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>relabun, <span class="at">y=</span>genus, <span class="at">group=</span>genus, <span class="at">color=</span>phylum)) <span class="sc">+</span></span>
<span id="cb81-10"><a href="#cb81-10" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>phylum_colors) <span class="sc">+</span></span>
<span id="cb81-11"><a href="#cb81-11" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb81-12"><a href="#cb81-12" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>sex)<span class="sc">+</span></span>
<span id="cb81-13"><a href="#cb81-13" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb81-14"><a href="#cb81-14" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>))<span class="sc">+</span></span>
<span id="cb81-15"><a href="#cb81-15" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Family&quot;</span>, <span class="at">x=</span><span class="st">&quot;Relative abundance&quot;</span>, <span class="at">color=</span><span class="st">&quot;Phylum&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/taxonomy_jitterplot_genus_sex-1.png" width="960" /></p>
<p><strong>Number of MAGs without genera-level annotation</strong></p>
<pre><code>84</code></pre>
<!-- preamble start -->

    <script>

      function styleCell_gz2zispzxtfa6iqvy7l1(i, j, css_id) {
          var table = document.getElementById("tinytable_gz2zispzxtfa6iqvy7l1");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_gz2zispzxtfa6iqvy7l1');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_gz2zispzxtfa6iqvy7l1(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_gz2zispzxtfa6iqvy7l1");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_x7v20wfxm88fm0c8jlvk',}, 
          { positions: [ { i: 6, j: 0 }, { i: 6, j: 1 }, { i: 6, j: 2 }, { i: 6, j: 3 },  ], css_id: 'tinytable_css_jc2nsv3dlu4fs6ekvzo0',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_gz2zispzxtfa6iqvy7l1(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_x7v20wfxm88fm0c8jlvk, .table th.tinytable_css_x7v20wfxm88fm0c8jlvk { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
      .table td.tinytable_css_jc2nsv3dlu4fs6ekvzo0, .table th.tinytable_css_jc2nsv3dlu4fs6ekvzo0 { border-bottom: solid #d3d8dc 0.1em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_gz2zispzxtfa6iqvy7l1" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing='true'>
        <thead>
        
              <tr>
                <th scope="col">phylum</th>
                <th scope="col">count_nogene</th>
                <th scope="col">count_total</th>
                <th scope="col">percentage</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>p__Actinomycetota </td>
                  <td> 5</td>
                  <td>  25</td>
                  <td>20.000000</td>
                </tr>
                <tr>
                  <td>p__Bacillota      </td>
                  <td> 3</td>
                  <td>  94</td>
                  <td> 3.191489</td>
                </tr>
                <tr>
                  <td>p__Bacillota_A    </td>
                  <td>71</td>
                  <td>1440</td>
                  <td> 4.930556</td>
                </tr>
                <tr>
                  <td>p__Bacteroidota   </td>
                  <td> 1</td>
                  <td>  71</td>
                  <td> 1.408451</td>
                </tr>
                <tr>
                  <td>p__Cyanobacteriota</td>
                  <td> 3</td>
                  <td>  82</td>
                  <td> 3.658537</td>
                </tr>
                <tr>
                  <td>p__Spirochaetota  </td>
                  <td> 1</td>
                  <td>   4</td>
                  <td>25.000000</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
<p><strong>Percentage of MAGs without genus-level annotation</strong></p>
<pre><code>4.623005</code></pre>
<p><strong>Number of MAGs without species-level annotation</strong></p>
<pre><code># A tibble: 1 × 1
  Mag_nospecies
          &lt;int&gt;
1          1547</code></pre>
<!-- preamble start -->

    <script>

      function styleCell_3dcwdwp9mmxaw492hqcs(i, j, css_id) {
          var table = document.getElementById("tinytable_3dcwdwp9mmxaw492hqcs");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_3dcwdwp9mmxaw492hqcs');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_3dcwdwp9mmxaw492hqcs(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_3dcwdwp9mmxaw492hqcs");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 }, { i: 0, j: 4 }, { i: 0, j: 5 },  ], css_id: 'tinytable_css_i0m4rbrdwiriqi9grg88',}, 
          { positions: [ { i: 15, j: 0 }, { i: 15, j: 1 }, { i: 15, j: 2 }, { i: 15, j: 3 }, { i: 15, j: 4 }, { i: 15, j: 5 },  ], css_id: 'tinytable_css_d3ozmhasdudhgi8uj1f6',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_3dcwdwp9mmxaw492hqcs(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_i0m4rbrdwiriqi9grg88, .table th.tinytable_css_i0m4rbrdwiriqi9grg88 { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
      .table td.tinytable_css_d3ozmhasdudhgi8uj1f6, .table th.tinytable_css_d3ozmhasdudhgi8uj1f6 { border-bottom: solid #d3d8dc 0.1em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_3dcwdwp9mmxaw492hqcs" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing='true'>
        <thead>
        
              <tr>
                <th scope="col">phylum</th>
                <th scope="col">count_nospecies</th>
                <th scope="col">count_total</th>
                <th scope="col">species_annotated</th>
                <th scope="col">percentage</th>
                <th scope="col">percentage_species</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>p__Actinomycetota   </td>
                  <td>  25</td>
                  <td>  25</td>
                  <td>  0</td>
                  <td>100.00000</td>
                  <td> 0.000000</td>
                </tr>
                <tr>
                  <td>p__Bacillota        </td>
                  <td>  93</td>
                  <td>  94</td>
                  <td>  1</td>
                  <td> 98.93617</td>
                  <td> 1.063830</td>
                </tr>
                <tr>
                  <td>p__Bacillota_A      </td>
                  <td>1211</td>
                  <td>1440</td>
                  <td>229</td>
                  <td> 84.09722</td>
                  <td>15.902778</td>
                </tr>
                <tr>
                  <td>p__Bacillota_B      </td>
                  <td>   1</td>
                  <td>   1</td>
                  <td>  0</td>
                  <td>100.00000</td>
                  <td> 0.000000</td>
                </tr>
                <tr>
                  <td>p__Bacillota_C      </td>
                  <td>   7</td>
                  <td>   7</td>
                  <td>  0</td>
                  <td>100.00000</td>
                  <td> 0.000000</td>
                </tr>
                <tr>
                  <td>p__Bacteroidota     </td>
                  <td>  53</td>
                  <td>  71</td>
                  <td> 18</td>
                  <td> 74.64789</td>
                  <td>25.352113</td>
                </tr>
                <tr>
                  <td>p__Campylobacterota </td>
                  <td>  13</td>
                  <td>  15</td>
                  <td>  2</td>
                  <td> 86.66667</td>
                  <td>13.333333</td>
                </tr>
                <tr>
                  <td>p__Cyanobacteriota  </td>
                  <td>  74</td>
                  <td>  82</td>
                  <td>  8</td>
                  <td> 90.24390</td>
                  <td> 9.756098</td>
                </tr>
                <tr>
                  <td>p__Desulfobacterota </td>
                  <td>   8</td>
                  <td>   8</td>
                  <td>  0</td>
                  <td>100.00000</td>
                  <td> 0.000000</td>
                </tr>
                <tr>
                  <td>p__Methanobacteriota</td>
                  <td>   1</td>
                  <td>   1</td>
                  <td>  0</td>
                  <td>100.00000</td>
                  <td> 0.000000</td>
                </tr>
                <tr>
                  <td>p__Pseudomonadota   </td>
                  <td>  24</td>
                  <td>  34</td>
                  <td> 10</td>
                  <td> 70.58824</td>
                  <td>29.411765</td>
                </tr>
                <tr>
                  <td>p__Spirochaetota    </td>
                  <td>   4</td>
                  <td>   4</td>
                  <td>  0</td>
                  <td>100.00000</td>
                  <td> 0.000000</td>
                </tr>
                <tr>
                  <td>p__Synergistota     </td>
                  <td>  17</td>
                  <td>  17</td>
                  <td>  0</td>
                  <td>100.00000</td>
                  <td> 0.000000</td>
                </tr>
                <tr>
                  <td>p__Thermoplasmatota </td>
                  <td>  11</td>
                  <td>  12</td>
                  <td>  1</td>
                  <td> 91.66667</td>
                  <td> 8.333333</td>
                </tr>
                <tr>
                  <td>p__Verrucomicrobiota</td>
                  <td>   5</td>
                  <td>   6</td>
                  <td>  1</td>
                  <td> 83.33333</td>
                  <td>16.666667</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
<p><strong>Percentage of MAGs without species-level annotation</strong></p>
<pre><code>85.14034</code></pre>
<!--chapter:end:04_Community.Rmd-->
</div>
</div>
</div>
</div>
<div id="alpha-diversity" class="section level1 hasAnchor" number="4">
<h1 class="hasAnchor"><span class="header-section-number">4</span> Alpha diversity<a href="#alpha-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="gut-microbiota-2" class="section level2 hasAnchor" number="4.1">
<h2 class="hasAnchor"><span class="header-section-number">4.1</span> Gut microbiota<a href="#gut-microbiota-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/data.Rdata&quot;</span>)</span>
<span id="cb86-2"><a href="#cb86-2" tabindex="-1"></a>treatment_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#f56042&quot;</span>,<span class="st">&quot;#429ef5&quot;</span>, <span class="st">&quot;#42f58d&quot;</span>, <span class="st">&quot;#b142f5&quot;</span>)</span>
<span id="cb86-3"><a href="#cb86-3" tabindex="-1"></a>sample_metadata<span class="sc">$</span>broad_environment <span class="ot">&lt;-</span> <span class="fu">factor</span>(sample_metadata<span class="sc">$</span>broad_environment, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;1000198 - Mixed forest&quot;</span>, <span class="st">&quot;1000221 - Temperate woodland&quot;</span>, <span class="st">&quot;1000218 - Xeric shrubland&quot;</span>, <span class="st">&quot;1000245 - Cropland&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" tabindex="-1"></a><span class="co"># Calculate Hill numbers</span></span>
<span id="cb87-2"><a href="#cb87-2" tabindex="-1"></a>richness <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb87-3"><a href="#cb87-3" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-4"><a href="#cb87-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">all</span>(. <span class="sc">==</span> <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb87-5"><a href="#cb87-5" tabindex="-1"></a>  <span class="fu">hilldiv</span>(., <span class="at">q =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-6"><a href="#cb87-6" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb87-7"><a href="#cb87-7" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb87-8"><a href="#cb87-8" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">richness =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-9"><a href="#cb87-9" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;sample&quot;</span>)</span>
<span id="cb87-10"><a href="#cb87-10" tabindex="-1"></a></span>
<span id="cb87-11"><a href="#cb87-11" tabindex="-1"></a>neutral <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb87-12"><a href="#cb87-12" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-13"><a href="#cb87-13" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">all</span>(. <span class="sc">==</span> <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb87-14"><a href="#cb87-14" tabindex="-1"></a>  <span class="fu">hilldiv</span>(., <span class="at">q =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-15"><a href="#cb87-15" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb87-16"><a href="#cb87-16" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb87-17"><a href="#cb87-17" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">neutral =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-18"><a href="#cb87-18" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;sample&quot;</span>)</span>
<span id="cb87-19"><a href="#cb87-19" tabindex="-1"></a></span>
<span id="cb87-20"><a href="#cb87-20" tabindex="-1"></a>phylogenetic <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb87-21"><a href="#cb87-21" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-22"><a href="#cb87-22" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">all</span>(. <span class="sc">==</span> <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb87-23"><a href="#cb87-23" tabindex="-1"></a>  <span class="fu">hilldiv</span>(., <span class="at">q =</span> <span class="dv">1</span>, <span class="at">tree =</span> genome_tree) <span class="sc">%&gt;%</span></span>
<span id="cb87-24"><a href="#cb87-24" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb87-25"><a href="#cb87-25" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb87-26"><a href="#cb87-26" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">phylogenetic =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-27"><a href="#cb87-27" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;sample&quot;</span>)</span>
<span id="cb87-28"><a href="#cb87-28" tabindex="-1"></a></span>
<span id="cb87-29"><a href="#cb87-29" tabindex="-1"></a><span class="co"># Aggregate basal GIFT into elements</span></span>
<span id="cb87-30"><a href="#cb87-30" tabindex="-1"></a>dist <span class="ot">&lt;-</span> genome_gifts <span class="sc">%&gt;%</span></span>
<span id="cb87-31"><a href="#cb87-31" tabindex="-1"></a>  <span class="fu">to.elements</span>(., GIFT_db) <span class="sc">%&gt;%</span></span>
<span id="cb87-32"><a href="#cb87-32" tabindex="-1"></a>  <span class="fu">traits2dist</span>(., <span class="at">method =</span> <span class="st">&quot;gower&quot;</span>)</span>
<span id="cb87-33"><a href="#cb87-33" tabindex="-1"></a></span>
<span id="cb87-34"><a href="#cb87-34" tabindex="-1"></a>functional <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb87-35"><a href="#cb87-35" tabindex="-1"></a>  <span class="fu">filter</span>(genome <span class="sc">%in%</span> <span class="fu">rownames</span>(dist)) <span class="sc">%&gt;%</span> </span>
<span id="cb87-36"><a href="#cb87-36" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-37"><a href="#cb87-37" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">all</span>(. <span class="sc">==</span> <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb87-38"><a href="#cb87-38" tabindex="-1"></a>  <span class="fu">hilldiv</span>(., <span class="at">q =</span> <span class="dv">1</span>, <span class="at">dist =</span> dist) <span class="sc">%&gt;%</span></span>
<span id="cb87-39"><a href="#cb87-39" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb87-40"><a href="#cb87-40" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb87-41"><a href="#cb87-41" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">functional =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-42"><a href="#cb87-42" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-43"><a href="#cb87-43" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">functional =</span> <span class="fu">if_else</span>(<span class="fu">is.nan</span>(functional), <span class="dv">1</span>, functional))</span>
<span id="cb87-44"><a href="#cb87-44" tabindex="-1"></a></span>
<span id="cb87-45"><a href="#cb87-45" tabindex="-1"></a><span class="co"># Merge all metrics</span></span>
<span id="cb87-46"><a href="#cb87-46" tabindex="-1"></a>alpha_div <span class="ot">&lt;-</span> richness <span class="sc">%&gt;%</span></span>
<span id="cb87-47"><a href="#cb87-47" tabindex="-1"></a>  <span class="fu">full_join</span>(neutral, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb87-48"><a href="#cb87-48" tabindex="-1"></a>  <span class="fu">full_join</span>(phylogenetic, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb87-49"><a href="#cb87-49" tabindex="-1"></a>  <span class="fu">full_join</span>(functional, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample))</span></code></pre></div>
<!-- preamble start -->

    <script>

      function styleCell_h0erwlurjphs9uc7kt8z(i, j, css_id) {
          var table = document.getElementById("tinytable_h0erwlurjphs9uc7kt8z");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_h0erwlurjphs9uc7kt8z');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_h0erwlurjphs9uc7kt8z(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_h0erwlurjphs9uc7kt8z");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 4, j: 0 }, { i: 4, j: 1 }, { i: 4, j: 2 }, { i: 4, j: 3 }, { i: 4, j: 4 }, { i: 4, j: 5 },  ], css_id: 'tinytable_css_ricqu7uag7yv7xg7dowp',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 }, { i: 0, j: 4 }, { i: 0, j: 5 },  ], css_id: 'tinytable_css_8ipgkd5skpn8qt7zw00k',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_h0erwlurjphs9uc7kt8z(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_ricqu7uag7yv7xg7dowp, .table th.tinytable_css_ricqu7uag7yv7xg7dowp { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_8ipgkd5skpn8qt7zw00k, .table th.tinytable_css_8ipgkd5skpn8qt7zw00k { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_h0erwlurjphs9uc7kt8z" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing='true'>
        <thead>
        
              <tr>
                <th scope="col">alpha</th>
                <th scope="col">Total</th>
                <th scope="col">Xeric_shrubland</th>
                <th scope="col">Temperate_woodland</th>
                <th scope="col">Cropland</th>
                <th scope="col">Mixed_forest</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>richness    </td>
                  <td>307.14±137.95</td>
                  <td>331.67±156.38</td>
                  <td>290.87±124.42</td>
                  <td>423.73±156.53</td>
                  <td>196.6±87.35</td>
                </tr>
                <tr>
                  <td>neutral     </td>
                  <td>115.2±68.51  </td>
                  <td>125.56±81.45 </td>
                  <td>115.36±64.88 </td>
                  <td>143.35±80.62 </td>
                  <td>41.57±21.29</td>
                </tr>
                <tr>
                  <td>phylogenetic</td>
                  <td>7.35±1.43    </td>
                  <td>7.66±1.29    </td>
                  <td>7.17±1.38    </td>
                  <td>8.68±0.97    </td>
                  <td>6.41±1.48  </td>
                </tr>
                <tr>
                  <td>functional  </td>
                  <td>1.37±0.06    </td>
                  <td>1.33±0.03    </td>
                  <td>1.38±0.07    </td>
                  <td>1.36±0.03    </td>
                  <td>1.33±0.04  </td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" tabindex="-1"></a><span class="co">#Richness</span></span>
<span id="cb88-2"><a href="#cb88-2" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> alpha_div <span class="sc">%&gt;%</span></span>
<span id="cb88-3"><a href="#cb88-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb88-4"><a href="#cb88-4" tabindex="-1"></a>  <span class="fu">left_join</span>(., sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb88-5"><a href="#cb88-5" tabindex="-1"></a>  <span class="fu">filter</span>(metric<span class="sc">==</span><span class="st">&quot;richness&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb88-6"><a href="#cb88-6" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> value, <span class="at">x =</span> broad_environment, <span class="at">group=</span>broad_environment, <span class="at">color=</span>broad_environment, <span class="at">fill=</span>broad_environment)) <span class="sc">+</span></span>
<span id="cb88-7"><a href="#cb88-7" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb88-8"><a href="#cb88-8" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.5</span>,<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb88-9"><a href="#cb88-9" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>treatment_colors)<span class="sc">+</span></span>
<span id="cb88-10"><a href="#cb88-10" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>treatment_colors) <span class="sc">+</span></span>
<span id="cb88-11"><a href="#cb88-11" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;1000218 - Xeric shrubland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Xeric shrubland&quot;</span>, <span class="st">&quot;1000245 - Cropland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Cropland&quot;</span>, <span class="st">&quot;1000198 - Mixed forest&quot;</span> <span class="ot">=</span> <span class="st">&quot;Mixed forest&quot;</span>, <span class="st">&quot;1000221 - Temperate woodland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Temperate woodland&quot;</span>)) <span class="sc">+</span></span>
<span id="cb88-12"><a href="#cb88-12" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>, <span class="at">show.legend =</span> F, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">label.y =</span> <span class="fu">c</span>(<span class="dv">300</span>), <span class="at">label.x =</span> <span class="fu">c</span>(<span class="fl">1.5</span>))<span class="sc">+</span></span>
<span id="cb88-13"><a href="#cb88-13" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="cn">NA</span>)) <span class="sc">+</span></span>
<span id="cb88-14"><a href="#cb88-14" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb88-15"><a href="#cb88-15" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb88-16"><a href="#cb88-16" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-17"><a href="#cb88-17" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-18"><a href="#cb88-18" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_line</span>(<span class="at">size =</span> .<span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb88-19"><a href="#cb88-19" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb88-20"><a href="#cb88-20" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">0</span>, <span class="at">r =</span> <span class="dv">20</span>, <span class="at">b =</span> <span class="dv">0</span>, <span class="at">l =</span> <span class="dv">0</span>))</span>
<span id="cb88-21"><a href="#cb88-21" tabindex="-1"></a>    )<span class="sc">+</span></span>
<span id="cb88-22"><a href="#cb88-22" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Richness&quot;</span>)</span>
<span id="cb88-23"><a href="#cb88-23" tabindex="-1"></a></span>
<span id="cb88-24"><a href="#cb88-24" tabindex="-1"></a></span>
<span id="cb88-25"><a href="#cb88-25" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> alpha_div <span class="sc">%&gt;%</span></span>
<span id="cb88-26"><a href="#cb88-26" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb88-27"><a href="#cb88-27" tabindex="-1"></a>  <span class="fu">left_join</span>(., sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb88-28"><a href="#cb88-28" tabindex="-1"></a>  <span class="fu">filter</span>(metric<span class="sc">==</span><span class="st">&quot;neutral&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb88-29"><a href="#cb88-29" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> value, <span class="at">x =</span> broad_environment, <span class="at">group=</span>broad_environment, <span class="at">color=</span>broad_environment, <span class="at">fill=</span>broad_environment)) <span class="sc">+</span></span>
<span id="cb88-30"><a href="#cb88-30" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb88-31"><a href="#cb88-31" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.5</span>,<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb88-32"><a href="#cb88-32" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>treatment_colors)<span class="sc">+</span></span>
<span id="cb88-33"><a href="#cb88-33" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>treatment_colors) <span class="sc">+</span></span>
<span id="cb88-34"><a href="#cb88-34" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;1000218 - Xeric shrubland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Xeric shrubland&quot;</span>, <span class="st">&quot;1000245 - Cropland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Cropland&quot;</span>, <span class="st">&quot;1000198 - Mixed forest&quot;</span> <span class="ot">=</span> <span class="st">&quot;Mixed forest&quot;</span>, <span class="st">&quot;1000221 - Temperate woodland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Temperate woodland&quot;</span>)) <span class="sc">+</span></span>
<span id="cb88-35"><a href="#cb88-35" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>, <span class="at">show.legend =</span> F, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">label.y =</span> <span class="fu">c</span>(<span class="dv">190</span>), <span class="at">label.x =</span> <span class="fu">c</span>(<span class="fl">1.5</span>))<span class="sc">+</span></span>
<span id="cb88-36"><a href="#cb88-36" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="cn">NA</span>)) <span class="sc">+</span></span>
<span id="cb88-37"><a href="#cb88-37" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb88-38"><a href="#cb88-38" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb88-39"><a href="#cb88-39" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-40"><a href="#cb88-40" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-41"><a href="#cb88-41" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_line</span>(<span class="at">size =</span> .<span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb88-42"><a href="#cb88-42" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb88-43"><a href="#cb88-43" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">0</span>, <span class="at">r =</span> <span class="dv">20</span>, <span class="at">b =</span> <span class="dv">0</span>, <span class="at">l =</span> <span class="dv">0</span>))</span>
<span id="cb88-44"><a href="#cb88-44" tabindex="-1"></a>    )<span class="sc">+</span></span>
<span id="cb88-45"><a href="#cb88-45" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Neutral alpha diversity&quot;</span>)</span>
<span id="cb88-46"><a href="#cb88-46" tabindex="-1"></a></span>
<span id="cb88-47"><a href="#cb88-47" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> alpha_div <span class="sc">%&gt;%</span></span>
<span id="cb88-48"><a href="#cb88-48" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb88-49"><a href="#cb88-49" tabindex="-1"></a>  <span class="fu">left_join</span>(., sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb88-50"><a href="#cb88-50" tabindex="-1"></a>  <span class="fu">filter</span>(metric<span class="sc">==</span><span class="st">&quot;phylogenetic&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb88-51"><a href="#cb88-51" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> value, <span class="at">x =</span> broad_environment, <span class="at">group=</span>broad_environment, <span class="at">color=</span>broad_environment, <span class="at">fill=</span>broad_environment)) <span class="sc">+</span></span>
<span id="cb88-52"><a href="#cb88-52" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb88-53"><a href="#cb88-53" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.5</span>,<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb88-54"><a href="#cb88-54" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>treatment_colors)<span class="sc">+</span></span>
<span id="cb88-55"><a href="#cb88-55" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>treatment_colors) <span class="sc">+</span></span>
<span id="cb88-56"><a href="#cb88-56" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;1000218 - Xeric shrubland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Xeric shrubland&quot;</span>, <span class="st">&quot;1000245 - Cropland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Cropland&quot;</span>, <span class="st">&quot;1000198 - Mixed forest&quot;</span> <span class="ot">=</span> <span class="st">&quot;Mixed forest&quot;</span>, <span class="st">&quot;1000221 - Temperate woodland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Temperate woodland&quot;</span>)) <span class="sc">+</span></span>
<span id="cb88-57"><a href="#cb88-57" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>, <span class="at">show.legend =</span> F, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">label.y =</span> <span class="fu">c</span>(<span class="dv">12</span>), <span class="at">label.x =</span> <span class="fu">c</span>(<span class="fl">1.5</span>))<span class="sc">+</span></span>
<span id="cb88-58"><a href="#cb88-58" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="cn">NA</span>)) <span class="sc">+</span></span>
<span id="cb88-59"><a href="#cb88-59" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb88-60"><a href="#cb88-60" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb88-61"><a href="#cb88-61" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-62"><a href="#cb88-62" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-63"><a href="#cb88-63" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_line</span>(<span class="at">size =</span> .<span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb88-64"><a href="#cb88-64" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb88-65"><a href="#cb88-65" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">0</span>, <span class="at">r =</span> <span class="dv">20</span>, <span class="at">b =</span> <span class="dv">0</span>, <span class="at">l =</span> <span class="dv">0</span>))</span>
<span id="cb88-66"><a href="#cb88-66" tabindex="-1"></a>    )<span class="sc">+</span></span>
<span id="cb88-67"><a href="#cb88-67" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Phylogenetic alpha diversity&quot;</span>)</span>
<span id="cb88-68"><a href="#cb88-68" tabindex="-1"></a></span>
<span id="cb88-69"><a href="#cb88-69" tabindex="-1"></a></span>
<span id="cb88-70"><a href="#cb88-70" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span> alpha_div <span class="sc">%&gt;%</span></span>
<span id="cb88-71"><a href="#cb88-71" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb88-72"><a href="#cb88-72" tabindex="-1"></a>  <span class="fu">left_join</span>(., sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb88-73"><a href="#cb88-73" tabindex="-1"></a>  <span class="fu">filter</span>(metric<span class="sc">==</span><span class="st">&quot;functional&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb88-74"><a href="#cb88-74" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> value, <span class="at">x =</span> broad_environment, <span class="at">group=</span>broad_environment, <span class="at">color=</span>broad_environment, <span class="at">fill=</span>broad_environment)) <span class="sc">+</span></span>
<span id="cb88-75"><a href="#cb88-75" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb88-76"><a href="#cb88-76" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.5</span>,<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb88-77"><a href="#cb88-77" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>treatment_colors)<span class="sc">+</span></span>
<span id="cb88-78"><a href="#cb88-78" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>treatment_colors) <span class="sc">+</span></span>
<span id="cb88-79"><a href="#cb88-79" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;1000218 - Xeric shrubland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Xeric shrubland&quot;</span>, <span class="st">&quot;1000245 - Cropland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Cropland&quot;</span>, <span class="st">&quot;1000198 - Mixed forest&quot;</span> <span class="ot">=</span> <span class="st">&quot;Mixed forest&quot;</span>, <span class="st">&quot;1000221 - Temperate woodland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Temperate woodland&quot;</span>)) <span class="sc">+</span></span>
<span id="cb88-80"><a href="#cb88-80" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>, <span class="at">show.legend =</span> F, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">label.y =</span> <span class="fu">c</span>(<span class="dv">2</span>), <span class="at">label.x =</span> <span class="fu">c</span>(<span class="fl">1.5</span>))<span class="sc">+</span></span>
<span id="cb88-81"><a href="#cb88-81" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="cn">NA</span>)) <span class="sc">+</span></span>
<span id="cb88-82"><a href="#cb88-82" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb88-83"><a href="#cb88-83" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb88-84"><a href="#cb88-84" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-85"><a href="#cb88-85" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb88-86"><a href="#cb88-86" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_line</span>(<span class="at">size =</span> .<span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb88-87"><a href="#cb88-87" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb88-88"><a href="#cb88-88" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">0</span>, <span class="at">r =</span> <span class="dv">20</span>, <span class="at">b =</span> <span class="dv">0</span>, <span class="at">l =</span> <span class="dv">0</span>))</span>
<span id="cb88-89"><a href="#cb88-89" tabindex="-1"></a>    )<span class="sc">+</span></span>
<span id="cb88-90"><a href="#cb88-90" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Functional alpha diversity&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/div_plot_together-1.png" width="960" /></p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" tabindex="-1"></a><span class="co">#Richness</span></span>
<span id="cb89-2"><a href="#cb89-2" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> alpha_div <span class="sc">%&gt;%</span></span>
<span id="cb89-3"><a href="#cb89-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-4"><a href="#cb89-4" tabindex="-1"></a>  <span class="fu">left_join</span>(., sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb89-5"><a href="#cb89-5" tabindex="-1"></a>  <span class="fu">filter</span>(metric<span class="sc">==</span><span class="st">&quot;richness&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-6"><a href="#cb89-6" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> value, <span class="at">x =</span> sex, <span class="at">group=</span>sex, <span class="at">color=</span>broad_environment, <span class="at">fill=</span>broad_environment)) <span class="sc">+</span></span>
<span id="cb89-7"><a href="#cb89-7" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb89-8"><a href="#cb89-8" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.5</span>,<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb89-9"><a href="#cb89-9" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>treatment_colors)<span class="sc">+</span></span>
<span id="cb89-10"><a href="#cb89-10" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>treatment_colors) <span class="sc">+</span></span>
<span id="cb89-11"><a href="#cb89-11" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span><span class="fu">factor</span>(broad_environment, <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;1000218 - Xeric shrubland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Xeric shrubland&quot;</span>, <span class="st">&quot;1000245 - Cropland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Cropland&quot;</span>, <span class="st">&quot;1000198 - Mixed forest&quot;</span> <span class="ot">=</span> <span class="st">&quot;Mixed forest&quot;</span>, <span class="st">&quot;1000221 - Temperate woodland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Temperate woodland&quot;</span>)), <span class="at">scale=</span><span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>)<span class="sc">+</span></span>
<span id="cb89-12"><a href="#cb89-12" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="cn">NA</span>)) <span class="sc">+</span></span>
<span id="cb89-13"><a href="#cb89-13" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb89-14"><a href="#cb89-14" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb89-15"><a href="#cb89-15" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-16"><a href="#cb89-16" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-17"><a href="#cb89-17" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">lineheight =</span> <span class="fl">0.6</span>),</span>
<span id="cb89-18"><a href="#cb89-18" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_line</span>(<span class="at">size =</span> .<span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb89-19"><a href="#cb89-19" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb89-20"><a href="#cb89-20" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">0</span>, <span class="at">r =</span> <span class="dv">20</span>, <span class="at">b =</span> <span class="dv">0</span>, <span class="at">l =</span> <span class="dv">0</span>))</span>
<span id="cb89-21"><a href="#cb89-21" tabindex="-1"></a>    )<span class="sc">+</span></span>
<span id="cb89-22"><a href="#cb89-22" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Richness&quot;</span>)</span>
<span id="cb89-23"><a href="#cb89-23" tabindex="-1"></a></span>
<span id="cb89-24"><a href="#cb89-24" tabindex="-1"></a></span>
<span id="cb89-25"><a href="#cb89-25" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> alpha_div <span class="sc">%&gt;%</span></span>
<span id="cb89-26"><a href="#cb89-26" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-27"><a href="#cb89-27" tabindex="-1"></a>  <span class="fu">left_join</span>(., sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb89-28"><a href="#cb89-28" tabindex="-1"></a>  <span class="fu">filter</span>(metric<span class="sc">==</span><span class="st">&quot;neutral&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-29"><a href="#cb89-29" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> value, <span class="at">x =</span> sex, <span class="at">group=</span>sex, <span class="at">color=</span>broad_environment, <span class="at">fill=</span>broad_environment)) <span class="sc">+</span></span>
<span id="cb89-30"><a href="#cb89-30" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb89-31"><a href="#cb89-31" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.5</span>,<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb89-32"><a href="#cb89-32" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>treatment_colors)<span class="sc">+</span></span>
<span id="cb89-33"><a href="#cb89-33" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>treatment_colors) <span class="sc">+</span></span>
<span id="cb89-34"><a href="#cb89-34" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span><span class="fu">factor</span>(broad_environment, <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;1000218 - Xeric shrubland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Xeric shrubland&quot;</span>, <span class="st">&quot;1000245 - Cropland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Cropland&quot;</span>, <span class="st">&quot;1000198 - Mixed forest&quot;</span> <span class="ot">=</span> <span class="st">&quot;Mixed forest&quot;</span>, <span class="st">&quot;1000221 - Temperate woodland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Temperate woodland&quot;</span>)), <span class="at">scale=</span><span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>)<span class="sc">+</span></span>
<span id="cb89-35"><a href="#cb89-35" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="cn">NA</span>)) <span class="sc">+</span></span>
<span id="cb89-36"><a href="#cb89-36" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb89-37"><a href="#cb89-37" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb89-38"><a href="#cb89-38" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-39"><a href="#cb89-39" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-40"><a href="#cb89-40" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">lineheight =</span> <span class="fl">0.6</span>),</span>
<span id="cb89-41"><a href="#cb89-41" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_line</span>(<span class="at">size =</span> .<span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb89-42"><a href="#cb89-42" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb89-43"><a href="#cb89-43" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">0</span>, <span class="at">r =</span> <span class="dv">20</span>, <span class="at">b =</span> <span class="dv">0</span>, <span class="at">l =</span> <span class="dv">0</span>))</span>
<span id="cb89-44"><a href="#cb89-44" tabindex="-1"></a>    )<span class="sc">+</span></span>
<span id="cb89-45"><a href="#cb89-45" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Neutral alpha diversity&quot;</span>)</span>
<span id="cb89-46"><a href="#cb89-46" tabindex="-1"></a></span>
<span id="cb89-47"><a href="#cb89-47" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> alpha_div <span class="sc">%&gt;%</span></span>
<span id="cb89-48"><a href="#cb89-48" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-49"><a href="#cb89-49" tabindex="-1"></a>  <span class="fu">left_join</span>(., sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb89-50"><a href="#cb89-50" tabindex="-1"></a>  <span class="fu">filter</span>(metric<span class="sc">==</span><span class="st">&quot;phylogenetic&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-51"><a href="#cb89-51" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> value, <span class="at">x =</span> sex, <span class="at">group=</span>sex, <span class="at">color=</span>broad_environment, <span class="at">fill=</span>broad_environment)) <span class="sc">+</span></span>
<span id="cb89-52"><a href="#cb89-52" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb89-53"><a href="#cb89-53" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.5</span>,<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb89-54"><a href="#cb89-54" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>treatment_colors)<span class="sc">+</span></span>
<span id="cb89-55"><a href="#cb89-55" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>treatment_colors) <span class="sc">+</span></span>
<span id="cb89-56"><a href="#cb89-56" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span><span class="fu">factor</span>(broad_environment, <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;1000218 - Xeric shrubland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Xeric shrubland&quot;</span>, <span class="st">&quot;1000245 - Cropland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Cropland&quot;</span>, <span class="st">&quot;1000198 - Mixed forest&quot;</span> <span class="ot">=</span> <span class="st">&quot;Mixed forest&quot;</span>, <span class="st">&quot;1000221 - Temperate woodland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Temperate woodland&quot;</span>)), <span class="at">scale=</span><span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>)<span class="sc">+</span></span>
<span id="cb89-57"><a href="#cb89-57" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="cn">NA</span>)) <span class="sc">+</span></span>
<span id="cb89-58"><a href="#cb89-58" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb89-59"><a href="#cb89-59" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb89-60"><a href="#cb89-60" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-61"><a href="#cb89-61" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-62"><a href="#cb89-62" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">lineheight =</span> <span class="fl">0.6</span>),</span>
<span id="cb89-63"><a href="#cb89-63" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_line</span>(<span class="at">size =</span> .<span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb89-64"><a href="#cb89-64" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb89-65"><a href="#cb89-65" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">0</span>, <span class="at">r =</span> <span class="dv">20</span>, <span class="at">b =</span> <span class="dv">0</span>, <span class="at">l =</span> <span class="dv">0</span>))</span>
<span id="cb89-66"><a href="#cb89-66" tabindex="-1"></a>    )<span class="sc">+</span></span>
<span id="cb89-67"><a href="#cb89-67" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Phylogenetic alpha diversity&quot;</span>)</span>
<span id="cb89-68"><a href="#cb89-68" tabindex="-1"></a></span>
<span id="cb89-69"><a href="#cb89-69" tabindex="-1"></a></span>
<span id="cb89-70"><a href="#cb89-70" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span> alpha_div <span class="sc">%&gt;%</span></span>
<span id="cb89-71"><a href="#cb89-71" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">&quot;metric&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-72"><a href="#cb89-72" tabindex="-1"></a>  <span class="fu">left_join</span>(., sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb89-73"><a href="#cb89-73" tabindex="-1"></a>  <span class="fu">filter</span>(metric<span class="sc">==</span><span class="st">&quot;functional&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-74"><a href="#cb89-74" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> value, <span class="at">x =</span> sex, <span class="at">group=</span>sex, <span class="at">color=</span>broad_environment, <span class="at">fill=</span>broad_environment)) <span class="sc">+</span></span>
<span id="cb89-75"><a href="#cb89-75" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb89-76"><a href="#cb89-76" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.5</span>,<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb89-77"><a href="#cb89-77" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>treatment_colors)<span class="sc">+</span></span>
<span id="cb89-78"><a href="#cb89-78" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>treatment_colors) <span class="sc">+</span></span>
<span id="cb89-79"><a href="#cb89-79" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span><span class="fu">factor</span>(broad_environment, <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;1000218 - Xeric shrubland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Xeric shrubland&quot;</span>, <span class="st">&quot;1000245 - Cropland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Cropland&quot;</span>, <span class="st">&quot;1000198 - Mixed forest&quot;</span> <span class="ot">=</span> <span class="st">&quot;Mixed forest&quot;</span>, <span class="st">&quot;1000221 - Temperate woodland&quot;</span> <span class="ot">=</span> <span class="st">&quot;Temperate woodland&quot;</span>)), <span class="at">scale=</span><span class="st">&quot;free&quot;</span>, <span class="at">space =</span> <span class="st">&quot;free&quot;</span>)<span class="sc">+</span></span>
<span id="cb89-80"><a href="#cb89-80" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="cn">NA</span>)) <span class="sc">+</span></span>
<span id="cb89-81"><a href="#cb89-81" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb89-82"><a href="#cb89-82" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb89-83"><a href="#cb89-83" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-84"><a href="#cb89-84" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb89-85"><a href="#cb89-85" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">lineheight =</span> <span class="fl">0.6</span>),</span>
<span id="cb89-86"><a href="#cb89-86" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_line</span>(<span class="at">size =</span> .<span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb89-87"><a href="#cb89-87" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>),</span>
<span id="cb89-88"><a href="#cb89-88" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">0</span>, <span class="at">r =</span> <span class="dv">20</span>, <span class="at">b =</span> <span class="dv">0</span>, <span class="at">l =</span> <span class="dv">0</span>))</span>
<span id="cb89-89"><a href="#cb89-89" tabindex="-1"></a>    )<span class="sc">+</span></span>
<span id="cb89-90"><a href="#cb89-90" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Functional alpha diversity&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/div_plot_together_sex-1.png" width="960" /></p>
<div id="mixed-models" class="section level3 hasAnchor" number="4.1.1">
<h3 class="hasAnchor"><span class="header-section-number">4.1.1</span> Mixed models<a href="#mixed-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" tabindex="-1"></a>alpha_div_meta <span class="ot">&lt;-</span> alpha_div <span class="sc">%&gt;%</span></span>
<span id="cb90-2"><a href="#cb90-2" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample))</span>
<span id="cb90-3"><a href="#cb90-3" tabindex="-1"></a></span>
<span id="cb90-4"><a href="#cb90-4" tabindex="-1"></a>alpha_div_meta_clean <span class="ot">&lt;-</span> alpha_div_meta <span class="sc">%&gt;%</span></span>
<span id="cb90-5"><a href="#cb90-5" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(richness), <span class="sc">!</span><span class="fu">is.na</span>(broad_environment), <span class="sc">!</span><span class="fu">is.na</span>(sex))</span>
<span id="cb90-6"><a href="#cb90-6" tabindex="-1"></a></span>
<span id="cb90-7"><a href="#cb90-7" tabindex="-1"></a>alpha_dic_meta_filtered_out <span class="ot">&lt;-</span> <span class="fu">anti_join</span>(alpha_div_meta, alpha_div_meta_clean)</span>
<span id="cb90-8"><a href="#cb90-8" tabindex="-1"></a></span>
<span id="cb90-9"><a href="#cb90-9" tabindex="-1"></a>Model_richness_random <span class="ot">&lt;-</span> <span class="fu">lme</span>(<span class="at">fixed =</span> richness <span class="sc">~</span> broad_environment, <span class="at">data =</span> alpha_div_meta_clean,</span>
<span id="cb90-10"><a href="#cb90-10" tabindex="-1"></a>               <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> sex)</span>
<span id="cb90-11"><a href="#cb90-11" tabindex="-1"></a><span class="fu">summary</span>(Model_richness_random)</span></code></pre></div>
<pre><code>Linear mixed-effects model fit by REML
  Data: alpha_div_meta_clean 
       AIC      BIC    logLik
  660.2575 670.0137 -325.1288

Random effects:
 Formula: ~1 | sex
        (Intercept) Residual
StdDev: 0.008756482 117.1185

Fixed effects:  richness ~ broad_environment 
                                                 Value Std.Error DF  t-value p-value
(Intercept)                                   196.6000  52.37697 51 3.753558  0.0004
broad_environment1000221 - Temperate woodland 101.6326  55.33842 51 1.836564  0.0721
broad_environment1000245 - Cropland           227.6857  68.57756 51 3.320120  0.0017
 Correlation: 
                                              (Intr) b_1-Tw
broad_environment1000221 - Temperate woodland -0.946       
broad_environment1000245 - Cropland           -0.764  0.723

Standardized Within-Group Residuals:
       Min         Q1        Med         Q3        Max 
-1.7950418 -0.6850547 -0.1129844  0.5485118  2.9437495 

Number of Observations: 55
Number of Groups: 2 </code></pre>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" tabindex="-1"></a>Model_richness<span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="at">formula =</span> richness <span class="sc">~</span> broad_environment<span class="sc">+</span>sex, <span class="at">data =</span> alpha_div_meta_clean) </span>
<span id="cb92-2"><a href="#cb92-2" tabindex="-1"></a><span class="fu">summary</span>(Model_richness)</span></code></pre></div>
<pre><code>
Call:
lm(formula = richness ~ broad_environment + sex, data = alpha_div_meta_clean)

Residuals:
    Min      1Q  Median      3Q     Max 
-198.26  -87.96  -12.64   60.05  334.36 

Coefficients:
                                              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                                     205.55      54.20   3.793 0.000396 ***
broad_environment1000221 - Temperate woodland   103.09      55.66   1.852 0.069778 .  
broad_environment1000245 - Cropland             234.72      69.66   3.369 0.001442 ** 
sexFemale                                       -22.38      32.24  -0.694 0.490722    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 117.7 on 51 degrees of freedom
Multiple R-squared:  0.1912,    Adjusted R-squared:  0.1436 
F-statistic: 4.018 on 3 and 51 DF,  p-value: 0.01215</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" tabindex="-1"></a>Model_richness_sex_random <span class="ot">&lt;-</span> <span class="fu">lme</span>(<span class="at">fixed =</span> richness <span class="sc">~</span> sex, <span class="at">data =</span> alpha_div_meta_clean,</span>
<span id="cb94-2"><a href="#cb94-2" tabindex="-1"></a>               <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> broad_environment)</span>
<span id="cb94-3"><a href="#cb94-3" tabindex="-1"></a><span class="fu">summary</span>(Model_richness_sex_random)</span></code></pre></div>
<pre><code>Linear mixed-effects model fit by REML
  Data: alpha_div_meta_clean 
       AIC      BIC    logLik
  674.7345 682.6157 -333.3673

Random effects:
 Formula: ~1 | broad_environment
        (Intercept) Residual
StdDev:    103.5142 117.8367

Fixed effects:  richness ~ sex 
               Value Std.Error DF   t-value p-value
(Intercept) 318.2455  66.28983 51  4.800820  0.0000
sexFemale   -19.3594  32.20735 51 -0.601086  0.5504
 Correlation: 
          (Intr)
sexFemale -0.256

Standardized Within-Group Residuals:
       Min         Q1        Med         Q3        Max 
-1.6989576 -0.7600262 -0.1150659  0.5803378  2.8466596 

Number of Observations: 55
Number of Groups: 3 </code></pre>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" tabindex="-1"></a><span class="fu">emmeans</span>(Model_richness_sex_random, pairwise <span class="sc">~</span> sex)</span></code></pre></div>
<pre><code>$emmeans
 sex    emmean   SE df lower.CL upper.CL
 Male      318 66.3  2     33.0      603
 Female    299 65.9  2     15.4      582

Degrees-of-freedom method: containment 
Confidence level used: 0.95 

$contrasts
 contrast      estimate   SE df t.ratio p.value
 Male - Female     19.4 32.2 51   0.601  0.5504

Degrees-of-freedom method: containment </code></pre>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" tabindex="-1"></a>Model_neutral_random <span class="ot">&lt;-</span> <span class="fu">lme</span>(<span class="at">fixed =</span> neutral <span class="sc">~</span> broad_environment, <span class="at">data =</span> alpha_div_meta_clean,</span>
<span id="cb98-2"><a href="#cb98-2" tabindex="-1"></a>               <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> sex)</span>
<span id="cb98-3"><a href="#cb98-3" tabindex="-1"></a><span class="fu">summary</span>(Model_neutral_random)</span></code></pre></div>
<pre><code>Linear mixed-effects model fit by REML
  Data: alpha_div_meta_clean 
      AIC      BIC   logLik
  591.136 600.8923 -290.568

Random effects:
 Formula: ~1 | sex
        (Intercept) Residual
StdDev:    3.155776 60.21234

Fixed effects:  neutral ~ broad_environment 
                                                 Value Std.Error DF  t-value p-value
(Intercept)                                   41.45285  27.02351 51 1.533955  0.1312
broad_environment1000221 - Temperate woodland 78.47859  28.45168 51 2.758311  0.0080
broad_environment1000245 - Cropland           89.64625  35.28272 51 2.540797  0.0141
 Correlation: 
                                              (Intr) b_1-Tw
broad_environment1000221 - Temperate woodland -0.943       
broad_environment1000245 - Cropland           -0.761  0.723

Standardized Within-Group Residuals:
       Min         Q1        Med         Q3        Max 
-1.4967057 -0.6852481 -0.2208246  0.5080001  2.9856016 

Number of Observations: 55
Number of Groups: 2 </code></pre>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" tabindex="-1"></a>Model_neutral<span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="at">formula =</span> neutral <span class="sc">~</span> broad_environment<span class="sc">+</span>sex, <span class="at">data =</span> alpha_div_meta_clean) </span>
<span id="cb100-2"><a href="#cb100-2" tabindex="-1"></a><span class="fu">summary</span>(Model_neutral)</span></code></pre></div>
<pre><code>
Call:
lm(formula = neutral ~ broad_environment + sex, data = alpha_div_meta_clean)

Residuals:
   Min     1Q Median     3Q    Max 
-91.26 -41.01 -18.18  33.87 172.37 

Coefficients:
                                              Estimate Std. Error t value Pr(&gt;|t|)   
(Intercept)                                      48.40      27.72   1.746  0.08685 . 
broad_environment1000221 - Temperate woodland    79.52      28.47   2.793  0.00734 **
broad_environment1000245 - Cropland              94.65      35.64   2.656  0.01053 * 
sexFemale                                       -17.09      16.49  -1.036  0.30511   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 60.21 on 51 degrees of freedom
Multiple R-squared:  0.1539,    Adjusted R-squared:  0.1042 
F-statistic: 3.093 on 3 and 51 DF,  p-value: 0.03501</code></pre>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" tabindex="-1"></a>Model_neutral_sex_random <span class="ot">&lt;-</span> <span class="fu">lme</span>(<span class="at">fixed =</span> neutral <span class="sc">~</span> sex, <span class="at">data =</span> alpha_div_meta_clean,</span>
<span id="cb102-2"><a href="#cb102-2" tabindex="-1"></a>               <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> broad_environment)</span>
<span id="cb102-3"><a href="#cb102-3" tabindex="-1"></a><span class="fu">summary</span>(Model_neutral_sex_random)</span></code></pre></div>
<pre><code>Linear mixed-effects model fit by REML
  Data: alpha_div_meta_clean 
       AIC      BIC    logLik
  602.9621 610.8433 -297.4811

Random effects:
 Formula: ~1 | broad_environment
        (Intercept) Residual
StdDev:    42.01245 60.31076

Fixed effects:  neutral ~ sex 
                Value Std.Error DF   t-value p-value
(Intercept) 108.84953  28.27875 51  3.849164  0.0003
sexFemale   -15.66423  16.46757 51 -0.951217  0.3460
 Correlation: 
          (Intr)
sexFemale -0.306

Standardized Within-Group Residuals:
       Min         Q1        Med         Q3        Max 
-1.4882909 -0.6737556 -0.3183183  0.6753650  2.8828753 

Number of Observations: 55
Number of Groups: 3 </code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" tabindex="-1"></a><span class="fu">emmeans</span>(Model_neutral_sex_random, pairwise <span class="sc">~</span> sex)</span></code></pre></div>
<pre><code>$emmeans
 sex    emmean   SE df lower.CL upper.CL
 Male    108.8 28.3  2    -12.8      231
 Female   93.2 28.0  2    -27.4      214

Degrees-of-freedom method: containment 
Confidence level used: 0.95 

$contrasts
 contrast      estimate   SE df t.ratio p.value
 Male - Female     15.7 16.5 51   0.951  0.3460

Degrees-of-freedom method: containment </code></pre>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" tabindex="-1"></a>Model_phylogenetic<span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="at">formula =</span> phylogenetic <span class="sc">~</span> broad_environment<span class="sc">+</span>sex, <span class="at">data =</span> alpha_div_meta_clean) </span>
<span id="cb106-2"><a href="#cb106-2" tabindex="-1"></a><span class="fu">summary</span>(Model_phylogenetic)</span></code></pre></div>
<pre><code>
Call:
lm(formula = phylogenetic ~ broad_environment + sex, data = alpha_div_meta_clean)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.9295 -0.8603 -0.1014  0.9895  2.8013 

Coefficients:
                                              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                                     6.6133     0.6269  10.549 2.02e-14 ***
broad_environment1000221 - Temperate woodland   0.8873     0.6438   1.378   0.1741    
broad_environment1000245 - Cropland             2.5411     0.8058   3.154   0.0027 ** 
sexFemale                                      -0.5038     0.3730  -1.351   0.1827    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 1.362 on 51 degrees of freedom
Multiple R-squared:  0.1932,    Adjusted R-squared:  0.1457 
F-statistic:  4.07 on 3 and 51 DF,  p-value: 0.01146</code></pre>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" tabindex="-1"></a>Model_phylogenetic_sex_random <span class="ot">&lt;-</span> <span class="fu">lme</span>(<span class="at">fixed =</span> phylogenetic <span class="sc">~</span> sex, <span class="at">data =</span> alpha_div_meta_clean,</span>
<span id="cb108-2"><a href="#cb108-2" tabindex="-1"></a>               <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> broad_environment)</span>
<span id="cb108-3"><a href="#cb108-3" tabindex="-1"></a></span>
<span id="cb108-4"><a href="#cb108-4" tabindex="-1"></a>Model_phylogenetic_sex_random <span class="ot">&lt;-</span> <span class="fu">lme</span>(<span class="at">fixed =</span> phylogenetic <span class="sc">~</span> sex, <span class="at">data =</span> alpha_div_meta_clean,</span>
<span id="cb108-5"><a href="#cb108-5" tabindex="-1"></a>               <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> broad_environment)</span>
<span id="cb108-6"><a href="#cb108-6" tabindex="-1"></a><span class="fu">summary</span>(Model_phylogenetic_sex_random)</span></code></pre></div>
<pre><code>Linear mixed-effects model fit by REML
  Data: alpha_div_meta_clean 
      AIC      BIC    logLik
  201.804 209.6852 -96.90202

Random effects:
 Formula: ~1 | broad_environment
        (Intercept) Residual
StdDev:     1.12689 1.362951

Fixed effects:  phylogenetic ~ sex 
                Value Std.Error DF   t-value p-value
(Intercept)  7.746185 0.7299749 51 10.611578  0.0000
sexFemale   -0.466249 0.3724362 51 -1.251891  0.2163
 Correlation: 
          (Intr)
sexFemale -0.268

Standardized Within-Group Residuals:
        Min          Q1         Med          Q3         Max 
-2.17047880 -0.65844821 -0.01077506  0.80316831  2.03418595 

Number of Observations: 55
Number of Groups: 3 </code></pre>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" tabindex="-1"></a><span class="fu">emmeans</span>(Model_phylogenetic_sex_random, pairwise <span class="sc">~</span> sex)</span></code></pre></div>
<pre><code>$emmeans
 sex    emmean    SE df lower.CL upper.CL
 Male     7.75 0.730  2     4.61     10.9
 Female   7.28 0.725  2     4.16     10.4

Degrees-of-freedom method: containment 
Confidence level used: 0.95 

$contrasts
 contrast      estimate    SE df t.ratio p.value
 Male - Female    0.466 0.372 51   1.252  0.2163

Degrees-of-freedom method: containment </code></pre>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" tabindex="-1"></a>Model_functional_random <span class="ot">&lt;-</span> <span class="fu">lme</span>(<span class="at">fixed =</span> functional <span class="sc">~</span> broad_environment, <span class="at">data =</span> alpha_div_meta_clean,</span>
<span id="cb112-2"><a href="#cb112-2" tabindex="-1"></a>               <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> sex)</span>
<span id="cb112-3"><a href="#cb112-3" tabindex="-1"></a><span class="fu">summary</span>(Model_functional_random)</span></code></pre></div>
<pre><code>Linear mixed-effects model fit by REML
  Data: alpha_div_meta_clean 
        AIC       BIC   logLik
  -154.3579 -144.6017 82.17895

Random effects:
 Formula: ~1 | sex
         (Intercept)  Residual
StdDev: 8.821509e-07 0.0464372

Fixed effects:  functional ~ broad_environment 
                                                  Value  Std.Error DF  t-value p-value
(Intercept)                                   1.3345237 0.02076735 51 64.26067  0.0000
broad_environment1000221 - Temperate woodland 0.0382453 0.02194156 51  1.74306  0.0874
broad_environment1000245 - Cropland           0.0217147 0.02719084 51  0.79860  0.4282
 Correlation: 
                                              (Intr) b_1-Tw
broad_environment1000221 - Temperate woodland -0.946       
broad_environment1000245 - Cropland           -0.764  0.723

Standardized Within-Group Residuals:
        Min          Q1         Med          Q3         Max 
-1.56046208 -0.45545337  0.02617571  0.41673185  5.77462530 

Number of Observations: 55
Number of Groups: 2 </code></pre>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" tabindex="-1"></a>Model_functional<span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="at">formula =</span> functional <span class="sc">~</span> broad_environment<span class="sc">+</span>sex, <span class="at">data =</span> alpha_div_meta_clean) </span>
<span id="cb114-2"><a href="#cb114-2" tabindex="-1"></a><span class="fu">summary</span>(Model_functional)</span></code></pre></div>
<pre><code>
Call:
lm(formula = functional ~ broad_environment + sex, data = alpha_div_meta_clean)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.071081 -0.021395  0.000071  0.020129  0.266568 

Coefficients:
                                              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                                   1.333335   0.021579  61.789   &lt;2e-16 ***
broad_environment1000221 - Temperate woodland 0.038052   0.022160   1.717    0.092 .  
broad_environment1000245 - Cropland           0.020781   0.027737   0.749    0.457    
sexFemale                                     0.002971   0.012838   0.231    0.818    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.04687 on 51 degrees of freedom
Multiple R-squared:  0.06378,   Adjusted R-squared:  0.008708 
F-statistic: 1.158 on 3 and 51 DF,  p-value: 0.3348</code></pre>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" tabindex="-1"></a>Model_functional_sex_random <span class="ot">&lt;-</span> <span class="fu">lme</span>(<span class="at">fixed =</span> functional <span class="sc">~</span> sex, <span class="at">data =</span> alpha_div_meta_clean,</span>
<span id="cb116-2"><a href="#cb116-2" tabindex="-1"></a>               <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> broad_environment)</span>
<span id="cb116-3"><a href="#cb116-3" tabindex="-1"></a><span class="fu">summary</span>(Model_functional_sex_random)</span></code></pre></div>
<pre><code>Linear mixed-effects model fit by REML
  Data: alpha_div_meta_clean 
        AIC       BIC   logLik
  -158.4514 -150.5703 83.22572

Random effects:
 Formula: ~1 | broad_environment
        (Intercept)   Residual
StdDev:  0.01309153 0.04678291

Fixed effects:  functional ~ sex 
                Value  Std.Error DF   t-value p-value
(Intercept) 1.3594219 0.01282884 51 105.96608  0.0000
sexFemale   0.0029804 0.01269615 51   0.23475  0.8153
 Correlation: 
          (Intr)
sexFemale -0.509

Standardized Within-Group Residuals:
        Min          Q1         Med          Q3         Max 
-1.84791978 -0.40639674  0.02278814  0.31350399  5.75641981 

Number of Observations: 55
Number of Groups: 3 </code></pre>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" tabindex="-1"></a><span class="fu">emmeans</span>(Model_functional_sex_random, pairwise <span class="sc">~</span> sex)</span></code></pre></div>
<pre><code>$emmeans
 sex    emmean     SE df lower.CL upper.CL
 Male     1.36 0.0128  2     1.30     1.41
 Female   1.36 0.0126  2     1.31     1.42

Degrees-of-freedom method: containment 
Confidence level used: 0.95 

$contrasts
 contrast      estimate     SE df t.ratio p.value
 Male - Female -0.00298 0.0127 51  -0.235  0.8153

Degrees-of-freedom method: containment </code></pre>
<!--chapter:end:05_Differential_abundance.Rmd-->
</div>
</div>
</div>
<div id="beta-diversity" class="section level1 hasAnchor" number="5">
<h1 class="hasAnchor"><span class="header-section-number">5</span> Beta diversity<a href="#beta-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="gut-microbiota-3" class="section level2 hasAnchor" number="5.1">
<h2 class="hasAnchor"><span class="header-section-number">5.1</span> Gut microbiota<a href="#gut-microbiota-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/data.Rdata&quot;</span>)</span>
<span id="cb120-2"><a href="#cb120-2" tabindex="-1"></a><span class="co">#load(&quot;data/beta_all.Rdata&quot;)</span></span>
<span id="cb120-3"><a href="#cb120-3" tabindex="-1"></a></span>
<span id="cb120-4"><a href="#cb120-4" tabindex="-1"></a>treatment_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#f56042&quot;</span>,<span class="st">&quot;#429ef5&quot;</span>, <span class="st">&quot;#42f58d&quot;</span>, <span class="st">&quot;#b142f5&quot;</span>, <span class="st">&quot;#f5e642&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" tabindex="-1"></a>sample_metadata_without_NA <span class="ot">&lt;-</span> sample_metadata <span class="sc">%&gt;%</span></span>
<span id="cb121-2"><a href="#cb121-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(broad_environment))</span></code></pre></div>
<p><strong><em>beta_div</em></strong></p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" tabindex="-1"></a><span class="co">#Presence/Absence</span></span>
<span id="cb122-2"><a href="#cb122-2" tabindex="-1"></a>beta_q0n <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span> </span>
<span id="cb122-3"><a href="#cb122-3" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(., <span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-4"><a href="#cb122-4" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">rowSums</span>(. <span class="sc">!=</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-5"><a href="#cb122-5" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~!</span><span class="fu">all</span>(. <span class="sc">==</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb122-6"><a href="#cb122-6" tabindex="-1"></a>  <span class="fu">hillpair</span>(., <span class="at">q =</span> <span class="dv">0</span>)</span>
<span id="cb122-7"><a href="#cb122-7" tabindex="-1"></a><span class="co">#Abundance-sensitive</span></span>
<span id="cb122-8"><a href="#cb122-8" tabindex="-1"></a>beta_q1n <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb122-9"><a href="#cb122-9" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(., <span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-10"><a href="#cb122-10" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">rowSums</span>(. <span class="sc">!=</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-11"><a href="#cb122-11" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~!</span><span class="fu">all</span>(. <span class="sc">==</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb122-12"><a href="#cb122-12" tabindex="-1"></a>  <span class="fu">hillpair</span>(., <span class="at">q =</span> <span class="dv">1</span>)</span>
<span id="cb122-13"><a href="#cb122-13" tabindex="-1"></a><span class="co">#Abundance + Phylogeny</span></span>
<span id="cb122-14"><a href="#cb122-14" tabindex="-1"></a>genome_counts_filt_beta <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb122-15"><a href="#cb122-15" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(., <span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb122-16"><a href="#cb122-16" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">rowSums</span>(. <span class="sc">!=</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-17"><a href="#cb122-17" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~!</span><span class="fu">all</span>(. <span class="sc">==</span> <span class="dv">0</span>))<span class="sc">%&gt;%</span></span>
<span id="cb122-18"><a href="#cb122-18" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(., <span class="st">&quot;genome&quot;</span>)</span>
<span id="cb122-19"><a href="#cb122-19" tabindex="-1"></a><span class="co">#Abundance + Trait Distance</span></span>
<span id="cb122-20"><a href="#cb122-20" tabindex="-1"></a>genome_tree <span class="ot">&lt;-</span> <span class="fu">keep.tip</span>(genome_tree, <span class="at">tip=</span>genome_counts_filt_beta<span class="sc">$</span>genome)</span>
<span id="cb122-21"><a href="#cb122-21" tabindex="-1"></a>beta_q1p <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span>  </span>
<span id="cb122-22"><a href="#cb122-22" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(., <span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-23"><a href="#cb122-23" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">rowSums</span>(. <span class="sc">!=</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-24"><a href="#cb122-24" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~!</span><span class="fu">all</span>(. <span class="sc">==</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb122-25"><a href="#cb122-25" tabindex="-1"></a>  <span class="fu">hillpair</span>(., <span class="at">q =</span> <span class="dv">1</span>, <span class="at">tree =</span> genome_tree)</span></code></pre></div>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" tabindex="-1"></a><span class="co"># 1. Only use genomes that also occur in genome_gifts</span></span>
<span id="cb123-2"><a href="#cb123-2" tabindex="-1"></a>valid_genomes <span class="ot">&lt;-</span> genome_counts_filt_beta<span class="sc">$</span>genome[genome_counts_filt_beta<span class="sc">$</span>genome <span class="sc">%in%</span> <span class="fu">rownames</span>(genome_gifts)]</span>
<span id="cb123-3"><a href="#cb123-3" tabindex="-1"></a></span>
<span id="cb123-4"><a href="#cb123-4" tabindex="-1"></a><span class="co"># 2. filter genome_gifts on these genomes</span></span>
<span id="cb123-5"><a href="#cb123-5" tabindex="-1"></a>genome_gifts1 <span class="ot">&lt;-</span> genome_gifts[valid_genomes, ]</span>
<span id="cb123-6"><a href="#cb123-6" tabindex="-1"></a>genome_gifts1 <span class="ot">&lt;-</span> genome_gifts1[, <span class="fu">colSums</span>(genome_gifts1 <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb123-7"><a href="#cb123-7" tabindex="-1"></a></span>
<span id="cb123-8"><a href="#cb123-8" tabindex="-1"></a><span class="co"># 3. Calculate functional distance matrix</span></span>
<span id="cb123-9"><a href="#cb123-9" tabindex="-1"></a>dist <span class="ot">&lt;-</span> genome_gifts1 <span class="sc">%&gt;%</span></span>
<span id="cb123-10"><a href="#cb123-10" tabindex="-1"></a>  <span class="fu">to.elements</span>(., GIFT_db) <span class="sc">%&gt;%</span></span>
<span id="cb123-11"><a href="#cb123-11" tabindex="-1"></a>  <span class="fu">traits2dist</span>(., <span class="at">method =</span> <span class="st">&quot;gower&quot;</span>)</span>
<span id="cb123-12"><a href="#cb123-12" tabindex="-1"></a></span>
<span id="cb123-13"><a href="#cb123-13" tabindex="-1"></a><span class="co"># 4. Determine genomes that exist in both genome_counts_filt_beta and dist</span></span>
<span id="cb123-14"><a href="#cb123-14" tabindex="-1"></a>common_genomes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(</span>
<span id="cb123-15"><a href="#cb123-15" tabindex="-1"></a>  genome_counts_filt_beta<span class="sc">$</span>genome,</span>
<span id="cb123-16"><a href="#cb123-16" tabindex="-1"></a>  <span class="fu">rownames</span>(dist)</span>
<span id="cb123-17"><a href="#cb123-17" tabindex="-1"></a>)</span>
<span id="cb123-18"><a href="#cb123-18" tabindex="-1"></a></span>
<span id="cb123-19"><a href="#cb123-19" tabindex="-1"></a><span class="co"># 5. genome_counts_filt to filter and sort the shared genomes</span></span>
<span id="cb123-20"><a href="#cb123-20" tabindex="-1"></a>genome_counts_filt_matched <span class="ot">&lt;-</span> genome_counts_filt <span class="sc">%&gt;%</span></span>
<span id="cb123-21"><a href="#cb123-21" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb123-22"><a href="#cb123-22" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">rownames</span>(.) <span class="sc">%in%</span> common_genomes) <span class="sc">%&gt;%</span></span>
<span id="cb123-23"><a href="#cb123-23" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~!</span><span class="fu">all</span>(. <span class="sc">==</span> <span class="dv">0</span>))  <span class="co"># optional: leere Spalten raus</span></span>
<span id="cb123-24"><a href="#cb123-24" tabindex="-1"></a></span>
<span id="cb123-25"><a href="#cb123-25" tabindex="-1"></a><span class="co"># 6. Limit and sort dist matrix to common genomes</span></span>
<span id="cb123-26"><a href="#cb123-26" tabindex="-1"></a>dist_matched <span class="ot">&lt;-</span> dist[common_genomes, common_genomes]</span>
<span id="cb123-27"><a href="#cb123-27" tabindex="-1"></a></span>
<span id="cb123-28"><a href="#cb123-28" tabindex="-1"></a></span>
<span id="cb123-29"><a href="#cb123-29" tabindex="-1"></a>beta_q1f <span class="ot">&lt;-</span> genome_counts_filt_matched <span class="sc">%&gt;%</span></span>
<span id="cb123-30"><a href="#cb123-30" tabindex="-1"></a>  <span class="fu">hillpair</span>(<span class="at">q =</span> <span class="dv">1</span>, <span class="at">dist =</span> dist_matched)</span></code></pre></div>
<div id="permanova" class="section level3 hasAnchor" number="5.1.1">
<h3 class="hasAnchor"><span class="header-section-number">5.1.1</span> Permanova<a href="#permanova" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" tabindex="-1"></a>run_permanova <span class="ot">&lt;-</span> <span class="cf">function</span>(beta_matrix, metadata, <span class="at">sample_col =</span> <span class="st">&quot;sample&quot;</span>, group_col, group_vars, <span class="at">pairwise =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb124-2"><a href="#cb124-2" tabindex="-1"></a>  </span>
<span id="cb124-3"><a href="#cb124-3" tabindex="-1"></a>  <span class="co"># 1. Sample labels from Beta-Matrix</span></span>
<span id="cb124-4"><a href="#cb124-4" tabindex="-1"></a>  sample_labels <span class="ot">&lt;-</span> <span class="fu">labels</span>(beta_matrix<span class="sc">$</span>S)</span>
<span id="cb124-5"><a href="#cb124-5" tabindex="-1"></a>  </span>
<span id="cb124-6"><a href="#cb124-6" tabindex="-1"></a>  <span class="co"># 2. Metadata filtered and sorted</span></span>
<span id="cb124-7"><a href="#cb124-7" tabindex="-1"></a>  metadata_matched <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb124-8"><a href="#cb124-8" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!!</span><span class="fu">sym</span>(sample_col) <span class="sc">%in%</span> sample_labels) <span class="sc">%&gt;%</span></span>
<span id="cb124-9"><a href="#cb124-9" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">match</span>(<span class="sc">!!</span><span class="fu">sym</span>(sample_col), sample_labels))</span>
<span id="cb124-10"><a href="#cb124-10" tabindex="-1"></a>  </span>
<span id="cb124-11"><a href="#cb124-11" tabindex="-1"></a>  <span class="co"># 3. Synchronize beta_matrix$S to metadata</span></span>
<span id="cb124-12"><a href="#cb124-12" tabindex="-1"></a>  common_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(sample_labels, metadata_matched[[sample_col]])</span>
<span id="cb124-13"><a href="#cb124-13" tabindex="-1"></a>  beta_matrix<span class="sc">$</span>S <span class="ot">&lt;-</span> beta_matrix<span class="sc">$</span>S[common_samples, common_samples]</span>
<span id="cb124-14"><a href="#cb124-14" tabindex="-1"></a>  metadata_matched <span class="ot">&lt;-</span> metadata_matched <span class="sc">%&gt;%</span></span>
<span id="cb124-15"><a href="#cb124-15" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!!</span><span class="fu">sym</span>(sample_col) <span class="sc">%in%</span> common_samples) <span class="sc">%&gt;%</span></span>
<span id="cb124-16"><a href="#cb124-16" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">match</span>(<span class="sc">!!</span><span class="fu">sym</span>(sample_col), common_samples))</span>
<span id="cb124-17"><a href="#cb124-17" tabindex="-1"></a>  </span>
<span id="cb124-18"><a href="#cb124-18" tabindex="-1"></a>  <span class="co"># 4. Remove samples with NA in group_col</span></span>
<span id="cb124-19"><a href="#cb124-19" tabindex="-1"></a>  valid_idx <span class="ot">&lt;-</span> <span class="sc">!</span><span class="fu">is.na</span>(metadata_matched[[group_col]])</span>
<span id="cb124-20"><a href="#cb124-20" tabindex="-1"></a>  beta_matrix<span class="sc">$</span>S <span class="ot">&lt;-</span> beta_matrix<span class="sc">$</span>S[valid_idx, valid_idx]</span>
<span id="cb124-21"><a href="#cb124-21" tabindex="-1"></a>  metadata_matched <span class="ot">&lt;-</span> metadata_matched[valid_idx, ]</span>
<span id="cb124-22"><a href="#cb124-22" tabindex="-1"></a>  </span>
<span id="cb124-23"><a href="#cb124-23" tabindex="-1"></a>  <span class="co"># 5. Betadisper and Permutest</span></span>
<span id="cb124-24"><a href="#cb124-24" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">===== Beta Dispersion Test =====</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb124-25"><a href="#cb124-25" tabindex="-1"></a>  dispersion_result <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(beta_matrix<span class="sc">$</span>S, metadata_matched[[group_col]])</span>
<span id="cb124-26"><a href="#cb124-26" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">permutest</span>(dispersion_result))</span>
<span id="cb124-27"><a href="#cb124-27" tabindex="-1"></a>  </span>
<span id="cb124-28"><a href="#cb124-28" tabindex="-1"></a>  <span class="co"># 6. Dynamic formula building</span></span>
<span id="cb124-29"><a href="#cb124-29" tabindex="-1"></a>  formula_str <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;beta_matrix$S ~ &quot;</span>, <span class="fu">paste</span>(group_vars, <span class="at">collapse =</span> <span class="st">&quot; + &quot;</span>))</span>
<span id="cb124-30"><a href="#cb124-30" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">===== PERMANOVA Test (adonis2) =====</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb124-31"><a href="#cb124-31" tabindex="-1"></a>  permanova_result <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(<span class="fu">as.formula</span>(formula_str),</span>
<span id="cb124-32"><a href="#cb124-32" tabindex="-1"></a>                              <span class="at">data =</span> metadata_matched,</span>
<span id="cb124-33"><a href="#cb124-33" tabindex="-1"></a>                              <span class="at">permutations =</span> <span class="dv">999</span>,</span>
<span id="cb124-34"><a href="#cb124-34" tabindex="-1"></a>                              <span class="at">by =</span> <span class="st">&quot;terms&quot;</span>)</span>
<span id="cb124-35"><a href="#cb124-35" tabindex="-1"></a>  <span class="fu">print</span>(broom<span class="sc">::</span><span class="fu">tidy</span>(permanova_result))</span>
<span id="cb124-36"><a href="#cb124-36" tabindex="-1"></a>  </span>
<span id="cb124-37"><a href="#cb124-37" tabindex="-1"></a>  <span class="co"># 7. (Optional) Pairwise Adonis</span></span>
<span id="cb124-38"><a href="#cb124-38" tabindex="-1"></a>  <span class="cf">if</span> (pairwise) {</span>
<span id="cb124-39"><a href="#cb124-39" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">===== Pairwise PERMANOVA =====</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb124-40"><a href="#cb124-40" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">pairwise.adonis</span>(beta_matrix<span class="sc">$</span>S, metadata_matched[[group_col]], <span class="at">perm =</span> <span class="dv">999</span>))</span>
<span id="cb124-41"><a href="#cb124-41" tabindex="-1"></a>  }</span>
<span id="cb124-42"><a href="#cb124-42" tabindex="-1"></a>  </span>
<span id="cb124-43"><a href="#cb124-43" tabindex="-1"></a>  <span class="co"># 8. Return outputs</span></span>
<span id="cb124-44"><a href="#cb124-44" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(</span>
<span id="cb124-45"><a href="#cb124-45" tabindex="-1"></a>    <span class="at">dispersion =</span> dispersion_result,</span>
<span id="cb124-46"><a href="#cb124-46" tabindex="-1"></a>    <span class="at">permanova =</span> permanova_result</span>
<span id="cb124-47"><a href="#cb124-47" tabindex="-1"></a>  ))</span>
<span id="cb124-48"><a href="#cb124-48" tabindex="-1"></a>}</span></code></pre></div>
</div>
</div>
</div>
<div id="richness" class="section level1 hasAnchor" number="6">
<h1 class="hasAnchor"><span class="header-section-number">6</span> Richness<a href="#richness" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" tabindex="-1"></a><span class="fu">run_permanova</span>(</span>
<span id="cb125-2"><a href="#cb125-2" tabindex="-1"></a>  <span class="at">beta_matrix =</span> beta_q0n<span class="sc">$</span>S,</span>
<span id="cb125-3"><a href="#cb125-3" tabindex="-1"></a>  <span class="at">metadata =</span> sample_metadata_without_NA,</span>
<span id="cb125-4"><a href="#cb125-4" tabindex="-1"></a>  <span class="at">sample_col =</span> <span class="st">&quot;sample&quot;</span>,</span>
<span id="cb125-5"><a href="#cb125-5" tabindex="-1"></a>  <span class="at">group_col =</span> <span class="st">&quot;broad_environment&quot;</span>,</span>
<span id="cb125-6"><a href="#cb125-6" tabindex="-1"></a>  <span class="at">group_vars =</span> <span class="fu">c</span>(<span class="st">&quot;broad_environment&quot;</span>,<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;broad_environment:sex&quot;</span>),</span>
<span id="cb125-7"><a href="#cb125-7" tabindex="-1"></a>  <span class="at">pairwise =</span> <span class="cn">FALSE</span></span>
<span id="cb125-8"><a href="#cb125-8" tabindex="-1"></a>)</span></code></pre></div>
<p>#Neutral diversity</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" tabindex="-1"></a><span class="fu">run_permanova</span>(</span>
<span id="cb126-2"><a href="#cb126-2" tabindex="-1"></a>  <span class="at">beta_matrix =</span> beta_q1n,</span>
<span id="cb126-3"><a href="#cb126-3" tabindex="-1"></a>  <span class="at">metadata =</span> sample_metadata_without_NA,</span>
<span id="cb126-4"><a href="#cb126-4" tabindex="-1"></a>  <span class="at">sample_col =</span> <span class="st">&quot;sample&quot;</span>,</span>
<span id="cb126-5"><a href="#cb126-5" tabindex="-1"></a>  <span class="at">group_col =</span> <span class="st">&quot;broad_environment&quot;</span>,</span>
<span id="cb126-6"><a href="#cb126-6" tabindex="-1"></a>  <span class="at">group_vars =</span> <span class="fu">c</span>(<span class="st">&quot;broad_environment&quot;</span>,<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;broad_environment:sex&quot;</span>),</span>
<span id="cb126-7"><a href="#cb126-7" tabindex="-1"></a>  <span class="at">pairwise =</span> <span class="cn">TRUE</span></span>
<span id="cb126-8"><a href="#cb126-8" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="phylogenetic-diversity-beta_q1p" class="section level1 hasAnchor" number="7">
<h1 class="hasAnchor"><span class="header-section-number">7</span> Phylogenetic diversity (beta_q1p)<a href="#phylogenetic-diversity-beta_q1p" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" tabindex="-1"></a><span class="fu">run_permanova</span>(</span>
<span id="cb127-2"><a href="#cb127-2" tabindex="-1"></a>  <span class="at">beta_matrix =</span> beta_q1p,</span>
<span id="cb127-3"><a href="#cb127-3" tabindex="-1"></a>  <span class="at">metadata =</span> sample_metadata_without_NA,</span>
<span id="cb127-4"><a href="#cb127-4" tabindex="-1"></a>  <span class="at">sample_col =</span> <span class="st">&quot;sample&quot;</span>,</span>
<span id="cb127-5"><a href="#cb127-5" tabindex="-1"></a>  <span class="at">group_col =</span> <span class="st">&quot;broad_environment&quot;</span>,</span>
<span id="cb127-6"><a href="#cb127-6" tabindex="-1"></a>  <span class="at">group_vars =</span> <span class="fu">c</span>(<span class="st">&quot;broad_environment&quot;</span>,<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;broad_environment:sex&quot;</span>),</span>
<span id="cb127-7"><a href="#cb127-7" tabindex="-1"></a>  <span class="at">pairwise =</span> <span class="cn">FALSE</span>  <span class="co"># no pairwise comparisons here</span></span>
<span id="cb127-8"><a href="#cb127-8" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="functional-diversity-beta_q1f" class="section level1 hasAnchor" number="8">
<h1 class="hasAnchor"><span class="header-section-number">8</span> Functional diversity (beta_q1f)<a href="#functional-diversity-beta_q1f" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" tabindex="-1"></a><span class="fu">run_permanova</span>(</span>
<span id="cb128-2"><a href="#cb128-2" tabindex="-1"></a>  <span class="at">beta_matrix =</span> beta_q1f,</span>
<span id="cb128-3"><a href="#cb128-3" tabindex="-1"></a>  <span class="at">metadata =</span> sample_metadata_without_NA,</span>
<span id="cb128-4"><a href="#cb128-4" tabindex="-1"></a>  <span class="at">sample_col =</span> <span class="st">&quot;sample&quot;</span>,</span>
<span id="cb128-5"><a href="#cb128-5" tabindex="-1"></a>  <span class="at">group_col =</span> <span class="st">&quot;broad_environment&quot;</span>,</span>
<span id="cb128-6"><a href="#cb128-6" tabindex="-1"></a>  <span class="at">group_vars =</span> <span class="fu">c</span>(<span class="st">&quot;broad_environment&quot;</span>,<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;broad_environment:sex&quot;</span>),</span>
<span id="cb128-7"><a href="#cb128-7" tabindex="-1"></a>  <span class="at">pairwise =</span> <span class="cn">TRUE</span>  </span>
<span id="cb128-8"><a href="#cb128-8" tabindex="-1"></a>)</span></code></pre></div>
<div id="nmds" class="section level3 hasAnchor" number="8.0.1">
<h3 class="hasAnchor"><span class="header-section-number">8.0.1</span> NMDS<a href="#nmds" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>####Richness</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" tabindex="-1"></a>beta_q0n<span class="sc">$</span>S <span class="sc">%&gt;%</span></span>
<span id="cb129-2"><a href="#cb129-2" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">metaMDS</span>(., <span class="at">trymax =</span> <span class="dv">500</span>, <span class="at">k =</span> <span class="dv">2</span>, <span class="at">trace=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb129-3"><a href="#cb129-3" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">scores</span>() <span class="sc">%&gt;%</span></span>
<span id="cb129-4"><a href="#cb129-4" tabindex="-1"></a>  <span class="fu">as_tibble</span>(., <span class="at">rownames =</span> <span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb129-5"><a href="#cb129-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb129-6"><a href="#cb129-6" tabindex="-1"></a>  <span class="fu">group_by</span>(broad_environment) <span class="sc">%&gt;%</span></span>
<span id="cb129-7"><a href="#cb129-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">x_cen =</span> <span class="fu">mean</span>(NMDS1, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb129-8"><a href="#cb129-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y_cen =</span> <span class="fu">mean</span>(NMDS2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb129-9"><a href="#cb129-9" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb129-10"><a href="#cb129-10" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> NMDS1, <span class="at">y =</span> NMDS2, <span class="at">color =</span> broad_environment, <span class="at">shape =</span> <span class="fu">as.factor</span>(sex))) <span class="sc">+</span></span>
<span id="cb129-11"><a href="#cb129-11" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb129-12"><a href="#cb129-12" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> treatment_colors,<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;high&quot;</span> <span class="ot">=</span> <span class="st">&quot;High-altitude&quot;</span>, <span class="st">&quot;low&quot;</span> <span class="ot">=</span> <span class="st">&quot;Low-altitude&quot;</span>)) <span class="sc">+</span></span>
<span id="cb129-13"><a href="#cb129-13" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> x_cen, <span class="at">y =</span> y_cen, <span class="at">xend =</span> NMDS1, <span class="at">yend =</span> NMDS2), <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb129-14"><a href="#cb129-14" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb129-15"><a href="#cb129-15" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb129-16"><a href="#cb129-16" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb129-17"><a href="#cb129-17" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb129-18"><a href="#cb129-18" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fl">0.5</span>, <span class="at">linetype =</span> <span class="st">&quot;solid&quot;</span>, <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb129-19"><a href="#cb129-19" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb129-20"><a href="#cb129-20" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb129-21"><a href="#cb129-21" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span></span>
<span id="cb129-22"><a href="#cb129-22" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb129-23"><a href="#cb129-23" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color=</span><span class="st">&quot;Environmental context&quot;</span>, <span class="at">shape=</span><span class="st">&quot;sex&quot;</span>)<span class="sc">+</span><span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> sample), <span class="at">size=</span><span class="dv">3</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/beta_div_nmds_richness_plot-1.png" width="960" /></p>
<div id="neutral-diversity" class="section level4 hasAnchor" number="8.0.1.1">
<h4 class="hasAnchor"><span class="header-section-number">8.0.1.1</span> Neutral diversity<a href="#neutral-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" tabindex="-1"></a>beta_q1n<span class="sc">$</span>S <span class="sc">%&gt;%</span></span>
<span id="cb130-2"><a href="#cb130-2" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">metaMDS</span>(., <span class="at">trymax =</span> <span class="dv">500</span>, <span class="at">k =</span> <span class="dv">2</span>, <span class="at">trace=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb130-3"><a href="#cb130-3" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">scores</span>() <span class="sc">%&gt;%</span></span>
<span id="cb130-4"><a href="#cb130-4" tabindex="-1"></a>  <span class="fu">as_tibble</span>(., <span class="at">rownames =</span> <span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb130-5"><a href="#cb130-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb130-6"><a href="#cb130-6" tabindex="-1"></a>  <span class="fu">group_by</span>(broad_environment) <span class="sc">%&gt;%</span></span>
<span id="cb130-7"><a href="#cb130-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">x_cen =</span> <span class="fu">mean</span>(NMDS1, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb130-8"><a href="#cb130-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y_cen =</span> <span class="fu">mean</span>(NMDS2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb130-9"><a href="#cb130-9" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb130-10"><a href="#cb130-10" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> NMDS1, <span class="at">y =</span> NMDS2, <span class="at">color =</span> broad_environment, <span class="at">shape =</span> <span class="fu">as.factor</span>(sex))) <span class="sc">+</span></span>
<span id="cb130-11"><a href="#cb130-11" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb130-12"><a href="#cb130-12" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> treatment_colors,<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;high&quot;</span> <span class="ot">=</span> <span class="st">&quot;High-altitude&quot;</span>, <span class="st">&quot;low&quot;</span> <span class="ot">=</span> <span class="st">&quot;Low-altitude&quot;</span>)) <span class="sc">+</span></span>
<span id="cb130-13"><a href="#cb130-13" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> x_cen, <span class="at">y =</span> y_cen, <span class="at">xend =</span> NMDS1, <span class="at">yend =</span> NMDS2), <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb130-14"><a href="#cb130-14" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb130-15"><a href="#cb130-15" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb130-16"><a href="#cb130-16" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb130-17"><a href="#cb130-17" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb130-18"><a href="#cb130-18" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fl">0.5</span>, <span class="at">linetype =</span> <span class="st">&quot;solid&quot;</span>, <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb130-19"><a href="#cb130-19" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb130-20"><a href="#cb130-20" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb130-21"><a href="#cb130-21" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span></span>
<span id="cb130-22"><a href="#cb130-22" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb130-23"><a href="#cb130-23" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color=</span><span class="st">&quot;Evironmental context&quot;</span>, <span class="at">shape=</span><span class="st">&quot;sex&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/beta_div_nmds_neutral_plot-1.png" width="960" /></p>
</div>
<div id="phylogenetic-diversity" class="section level4 hasAnchor" number="8.0.1.2">
<h4 class="hasAnchor"><span class="header-section-number">8.0.1.2</span> Phylogenetic diversity<a href="#phylogenetic-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" tabindex="-1"></a>beta_q1p<span class="sc">$</span>S <span class="sc">%&gt;%</span></span>
<span id="cb131-2"><a href="#cb131-2" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">metaMDS</span>(., <span class="at">trymax =</span> <span class="dv">500</span>, <span class="at">k =</span> <span class="dv">2</span>, <span class="at">trace=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb131-3"><a href="#cb131-3" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">scores</span>() <span class="sc">%&gt;%</span></span>
<span id="cb131-4"><a href="#cb131-4" tabindex="-1"></a>  <span class="fu">as_tibble</span>(., <span class="at">rownames =</span> <span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb131-5"><a href="#cb131-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb131-6"><a href="#cb131-6" tabindex="-1"></a>  <span class="fu">group_by</span>(broad_environment) <span class="sc">%&gt;%</span></span>
<span id="cb131-7"><a href="#cb131-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">x_cen =</span> <span class="fu">mean</span>(NMDS1, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb131-8"><a href="#cb131-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y_cen =</span> <span class="fu">mean</span>(NMDS2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb131-9"><a href="#cb131-9" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb131-10"><a href="#cb131-10" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> NMDS1, <span class="at">y =</span> NMDS2, <span class="at">color =</span> broad_environment, <span class="at">shape =</span> <span class="fu">as.factor</span>(sex))) <span class="sc">+</span></span>
<span id="cb131-11"><a href="#cb131-11" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb131-12"><a href="#cb131-12" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> treatment_colors,<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;high&quot;</span> <span class="ot">=</span> <span class="st">&quot;High-altitude&quot;</span>, <span class="st">&quot;low&quot;</span> <span class="ot">=</span> <span class="st">&quot;Low-altitude&quot;</span>)) <span class="sc">+</span></span>
<span id="cb131-13"><a href="#cb131-13" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> x_cen, <span class="at">y =</span> y_cen, <span class="at">xend =</span> NMDS1, <span class="at">yend =</span> NMDS2), <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb131-14"><a href="#cb131-14" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb131-15"><a href="#cb131-15" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb131-16"><a href="#cb131-16" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb131-17"><a href="#cb131-17" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb131-18"><a href="#cb131-18" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fl">0.5</span>, <span class="at">linetype =</span> <span class="st">&quot;solid&quot;</span>, <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb131-19"><a href="#cb131-19" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb131-20"><a href="#cb131-20" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb131-21"><a href="#cb131-21" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span></span>
<span id="cb131-22"><a href="#cb131-22" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb131-23"><a href="#cb131-23" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color=</span><span class="st">&quot;environment context&quot;</span>, <span class="at">shape=</span><span class="st">&quot;sex&quot;</span>)<span class="sc">+</span><span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> sample), <span class="at">size=</span><span class="dv">3</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/beta_div_nmds_phylo_plot-1.png" width="960" /></p>
</div>
<div id="functional-diversity" class="section level4 hasAnchor" number="8.0.1.3">
<h4 class="hasAnchor"><span class="header-section-number">8.0.1.3</span> Functional diversity<a href="#functional-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" tabindex="-1"></a>beta_q1f<span class="sc">$</span>S <span class="sc">%&gt;%</span></span>
<span id="cb132-2"><a href="#cb132-2" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">metaMDS</span>(., <span class="at">trymax =</span> <span class="dv">500</span>, <span class="at">k =</span> <span class="dv">2</span>, <span class="at">trace=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb132-3"><a href="#cb132-3" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">scores</span>() <span class="sc">%&gt;%</span></span>
<span id="cb132-4"><a href="#cb132-4" tabindex="-1"></a>  <span class="fu">as_tibble</span>(., <span class="at">rownames =</span> <span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb132-5"><a href="#cb132-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb132-6"><a href="#cb132-6" tabindex="-1"></a>  <span class="fu">group_by</span>(broad_environment) <span class="sc">%&gt;%</span></span>
<span id="cb132-7"><a href="#cb132-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">x_cen =</span> <span class="fu">mean</span>(NMDS1, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb132-8"><a href="#cb132-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y_cen =</span> <span class="fu">mean</span>(NMDS2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb132-9"><a href="#cb132-9" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb132-10"><a href="#cb132-10" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> NMDS1, <span class="at">y =</span> NMDS2, <span class="at">color =</span> broad_environment, <span class="at">shape =</span> <span class="fu">as.factor</span>(sex))) <span class="sc">+</span></span>
<span id="cb132-11"><a href="#cb132-11" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb132-12"><a href="#cb132-12" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> treatment_colors,<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;high&quot;</span> <span class="ot">=</span> <span class="st">&quot;High-altitude&quot;</span>, <span class="st">&quot;low&quot;</span> <span class="ot">=</span> <span class="st">&quot;Low-altitude&quot;</span>)) <span class="sc">+</span></span>
<span id="cb132-13"><a href="#cb132-13" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> x_cen, <span class="at">y =</span> y_cen, <span class="at">xend =</span> NMDS1, <span class="at">yend =</span> NMDS2), <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb132-14"><a href="#cb132-14" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb132-15"><a href="#cb132-15" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb132-16"><a href="#cb132-16" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb132-17"><a href="#cb132-17" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb132-18"><a href="#cb132-18" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fl">0.5</span>, <span class="at">linetype =</span> <span class="st">&quot;solid&quot;</span>, <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb132-19"><a href="#cb132-19" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb132-20"><a href="#cb132-20" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb132-21"><a href="#cb132-21" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>, <span class="at">legend.box =</span> <span class="st">&quot;vertical&quot;</span></span>
<span id="cb132-22"><a href="#cb132-22" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb132-23"><a href="#cb132-23" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color=</span><span class="st">&quot;broad_environment&quot;</span>, <span class="at">shape=</span><span class="st">&quot;sex&quot;</span>)<span class="sc">+</span><span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> sample), <span class="at">size=</span><span class="dv">3</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/beta_div_nmds_funct_plot-1.png" width="960" /></p>
<!--chapter:end:06_Functional_diffrences.Rmd-->
</div>
</div>
</div>
<div id="gut-microbiota-differential-abundance-analysis" class="section level1 hasAnchor" number="9">
<h1 class="hasAnchor"><span class="header-section-number">9</span> Gut microbiota: Differential abundance analysis<a href="#gut-microbiota-differential-abundance-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="structural-zeros" class="section level2 hasAnchor" number="9.1">
<h2 class="hasAnchor"><span class="header-section-number">9.1</span> Structural zeros<a href="#structural-zeros" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>#complett absence of a microbe/genome/function in one group</p>
</div>
<div id="mags-in-different-locations-and-shared-among-locations" class="section level2 hasAnchor" number="9.2">
<h2 class="hasAnchor"><span class="header-section-number">9.2</span> MAGs in different locations and shared among locations<a href="#mags-in-different-locations-and-shared-among-locations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="enrichment-analysis-between-male-and-female-ancombc2" class="section level2 hasAnchor" number="9.3">
<h2 class="hasAnchor"><span class="header-section-number">9.3</span> Enrichment analysis between male and female: Ancombc2<a href="#enrichment-analysis-between-male-and-female-ancombc2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="with-random-effect" class="section level3 hasAnchor" number="9.3.1">
<h3 class="hasAnchor"><span class="header-section-number">9.3.1</span> With random effect<a href="#with-random-effect" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="mag-level" class="section level4 hasAnchor" number="9.3.1.1">
<h4 class="hasAnchor"><span class="header-section-number">9.3.1.1</span> MAG level<a href="#mag-level" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="controling-structural-components-in-ancombc" class="section level5 hasAnchor" number="9.3.1.1.1">
<h5 class="hasAnchor"><span class="header-section-number">9.3.1.1.1</span> Controling structural components in Ancombc<a href="#controling-structural-components-in-ancombc" class="anchor-section" aria-label="Anchor link to header"></a></h5>
</div>
</div>
<div id="phylum-level-1" class="section level4 hasAnchor" number="9.3.1.2">
<h4 class="hasAnchor"><span class="header-section-number">9.3.1.2</span> Phylum level<a href="#phylum-level-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
<div id="family-level-1" class="section level4 hasAnchor" number="9.3.1.3">
<h4 class="hasAnchor"><span class="header-section-number">9.3.1.3</span> Family level<a href="#family-level-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
<div id="genera" class="section level4 hasAnchor" number="9.3.1.4">
<h4 class="hasAnchor"><span class="header-section-number">9.3.1.4</span> Genera<a href="#genera" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
</div>
<div id="without-random-effect" class="section level3 hasAnchor" number="9.3.2">
<h3 class="hasAnchor"><span class="header-section-number">9.3.2</span> Without random effect<a href="#without-random-effect" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="mag-level-1" class="section level4 hasAnchor" number="9.3.2.1">
<h4 class="hasAnchor"><span class="header-section-number">9.3.2.1</span> MAG level<a href="#mag-level-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="controling-structural-components-in-ancombc-1" class="section level5 hasAnchor" number="9.3.2.1.1">
<h5 class="hasAnchor"><span class="header-section-number">9.3.2.1.1</span> Controling structural components in Ancombc<a href="#controling-structural-components-in-ancombc-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
</div>
</div>
</div>
<div id="phylum-level-2" class="section level3 hasAnchor" number="9.3.3">
<h3 class="hasAnchor"><span class="header-section-number">9.3.3</span> Phylum level<a href="#phylum-level-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="family-level-2" class="section level3 hasAnchor" number="9.3.4">
<h3 class="hasAnchor"><span class="header-section-number">9.3.4</span> Family level<a href="#family-level-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="genera-1" class="section level3 hasAnchor" number="9.3.5">
<h3 class="hasAnchor"><span class="header-section-number">9.3.5</span> Genera<a href="#genera-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<!--chapter:end:07_microbiota_abundance_analysis_sex.Rmd-->
</div>
</div>
</div>
<div id="gut-microbiota-functional-analysis" class="section level1 hasAnchor" number="10">
<h1 class="hasAnchor"><span class="header-section-number">10</span> Gut microbiota: functional analysis<a href="#gut-microbiota-functional-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="mci" class="section level2 hasAnchor" number="10.1">
<h2 class="hasAnchor"><span class="header-section-number">10.1</span> MCI<a href="#mci" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" tabindex="-1"></a>GIFTs_functions_community <span class="sc">%&gt;%</span></span>
<span id="cb133-2"><a href="#cb133-2" tabindex="-1"></a>  <span class="fu">rowMeans</span>() <span class="sc">%&gt;%</span></span>
<span id="cb133-3"><a href="#cb133-3" tabindex="-1"></a>  <span class="fu">as_tibble</span>(., <span class="at">rownames =</span> <span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb133-4"><a href="#cb133-4" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb133-5"><a href="#cb133-5" tabindex="-1"></a><span class="co">#  filter(time_point==&quot;0_Wild&quot;) %&gt;%</span></span>
<span id="cb133-6"><a href="#cb133-6" tabindex="-1"></a>  <span class="fu">group_by</span>(sex) <span class="sc">%&gt;%</span></span>
<span id="cb133-7"><a href="#cb133-7" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">MCI =</span> <span class="fu">mean</span>(value), <span class="at">sd =</span> <span class="fu">sd</span>(value))</span></code></pre></div>
<pre><code>## # A tibble: 3 × 3
##   sex        MCI      sd
##   &lt;fct&gt;    &lt;dbl&gt;   &lt;dbl&gt;
## 1 Male     0.170  0.0141
## 2 Female   0.158  0.0199
## 3 &lt;NA&gt;   NaN     NA</code></pre>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" tabindex="-1"></a>MCI <span class="ot">&lt;-</span> GIFTs_functions_community <span class="sc">%&gt;%</span></span>
<span id="cb135-2"><a href="#cb135-2" tabindex="-1"></a>  <span class="fu">rowMeans</span>() <span class="sc">%&gt;%</span></span>
<span id="cb135-3"><a href="#cb135-3" tabindex="-1"></a>  <span class="fu">as_tibble</span>(., <span class="at">rownames =</span> <span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb135-4"><a href="#cb135-4" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) </span>
<span id="cb135-5"><a href="#cb135-5" tabindex="-1"></a><span class="co"># %&gt;% </span></span>
<span id="cb135-6"><a href="#cb135-6" tabindex="-1"></a><span class="co">#   filter(diet!=&quot;Post_grass&quot;)</span></span>
<span id="cb135-7"><a href="#cb135-7" tabindex="-1"></a></span>
<span id="cb135-8"><a href="#cb135-8" tabindex="-1"></a><span class="fu">shapiro.test</span>(MCI<span class="sc">$</span>value)</span></code></pre></div>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  MCI$value
## W = 0.98643, p-value = 0.6341</code></pre>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" tabindex="-1"></a><span class="fu">wilcox.test</span>(value <span class="sc">~</span> sex, <span class="at">data=</span>MCI)</span></code></pre></div>
<pre><code>## 
##  Wilcoxon rank sum exact test
## 
## data:  value by sex
## W = 587, p-value = 0.008114
## alternative hypothesis: true location shift is not equal to 0</code></pre>
<p><img src="Microbiome_Study_files/figure-html/comunity_elem_plot-1.png" width="960" />
<img src="Microbiome_Study_files/figure-html/comunity_funct_plot-1.png" width="960" /></p>
<p><img src="Microbiome_Study_files/figure-html/comunity_dom_plot-1.png" width="960" /></p>
</div>
<div id="wilcoxon" class="section level2 hasAnchor" number="10.2">
<h2 class="hasAnchor"><span class="header-section-number">10.2</span> Wilcoxon<a href="#wilcoxon" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="community-elements-differences" class="section level3 hasAnchor" number="10.2.1">
<h3 class="hasAnchor"><span class="header-section-number">10.2.1</span> Community elements differences:<a href="#community-elements-differences" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" tabindex="-1"></a>difference_table <span class="sc">%&gt;%</span></span>
<span id="cb139-2"><a href="#cb139-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>forcats<span class="sc">::</span><span class="fu">fct_reorder</span>(Function,Difference), <span class="at">y=</span>Difference, <span class="at">fill=</span>group_color)) <span class="sc">+</span> </span>
<span id="cb139-3"><a href="#cb139-3" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb139-4"><a href="#cb139-4" tabindex="-1"></a><span class="co">#  geom_point(size=4) + </span></span>
<span id="cb139-5"><a href="#cb139-5" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>treatment_colors) <span class="sc">+</span></span>
<span id="cb139-6"><a href="#cb139-6" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb139-7"><a href="#cb139-7" tabindex="-1"></a>  <span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb139-8"><a href="#cb139-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb139-9"><a href="#cb139-9" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb139-10"><a href="#cb139-10" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>, </span>
<span id="cb139-11"><a href="#cb139-11" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb139-12"><a href="#cb139-12" tabindex="-1"></a>          <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fl">0.15</span>, <span class="at">linetype =</span> <span class="st">&#39;solid&#39;</span>,</span>
<span id="cb139-13"><a href="#cb139-13" tabindex="-1"></a>                                <span class="at">colour =</span> <span class="st">&quot;grey&quot;</span>))<span class="sc">+</span></span>
<span id="cb139-14"><a href="#cb139-14" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Microbial Functional Capacity&quot;</span>) <span class="sc">+</span> </span>
<span id="cb139-15"><a href="#cb139-15" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Mean difference&quot;</span>)<span class="sc">+</span></span>
<span id="cb139-16"><a href="#cb139-16" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill=</span><span class="st">&quot;Elevation&quot;</span>)</span></code></pre></div>
<p><img src="Microbiome_Study_files/figure-html/plot3-1.png" width="1344" /></p>
</div>
<div id="community-functions-differences" class="section level3 hasAnchor" number="10.2.2">
<h3 class="hasAnchor"><span class="header-section-number">10.2.2</span> Community functions differences<a href="#community-functions-differences" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" tabindex="-1"></a>function_gift <span class="ot">&lt;-</span> GIFTs_functions_community <span class="sc">%&gt;%</span> </span>
<span id="cb140-2"><a href="#cb140-2" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb140-3"><a href="#cb140-3" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(., <span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb140-4"><a href="#cb140-4" tabindex="-1"></a>  <span class="fu">merge</span>(., sample_metadata <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,sex,broad_environment), <span class="at">by=</span><span class="st">&quot;sample&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" tabindex="-1"></a>unique_funct_db<span class="ot">&lt;-</span> GIFT_db[<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>)] <span class="sc">%&gt;%</span> </span>
<span id="cb141-2"><a href="#cb141-2" tabindex="-1"></a>  <span class="fu">distinct</span>(Code_function, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb141-3"><a href="#cb141-3" tabindex="-1"></a></span>
<span id="cb141-4"><a href="#cb141-4" tabindex="-1"></a>significant_functional <span class="ot">&lt;-</span> function_gift <span class="sc">%&gt;%</span></span>
<span id="cb141-5"><a href="#cb141-5" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(sample,sex,broad_environment), <span class="at">names_to =</span> <span class="st">&quot;trait&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb141-6"><a href="#cb141-6" tabindex="-1"></a>    <span class="fu">group_by</span>(trait) <span class="sc">%&gt;%</span></span>
<span id="cb141-7"><a href="#cb141-7" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">p_value =</span> <span class="fu">wilcox.test</span>(value <span class="sc">~</span> sex)<span class="sc">$</span>p.value) <span class="sc">%&gt;%</span></span>
<span id="cb141-8"><a href="#cb141-8" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">p_adjust=</span><span class="fu">p.adjust</span>(p_value, <span class="at">method=</span><span class="st">&quot;BH&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb141-9"><a href="#cb141-9" tabindex="-1"></a>    <span class="fu">filter</span>(p_adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>)<span class="sc">%&gt;%</span></span>
<span id="cb141-10"><a href="#cb141-10" tabindex="-1"></a>  <span class="fu">left_join</span>(.,unique_funct_db[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)],<span class="at">by =</span> <span class="fu">join_by</span>(trait <span class="sc">==</span> Code_function))</span></code></pre></div>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" tabindex="-1"></a>function_gift_t <span class="ot">&lt;-</span> function_gift  <span class="sc">%&gt;%</span> </span>
<span id="cb142-2"><a href="#cb142-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>sex)  <span class="sc">%&gt;%</span> </span>
<span id="cb142-3"><a href="#cb142-3" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb142-4"><a href="#cb142-4" tabindex="-1"></a>  <span class="fu">row_to_names</span>(<span class="at">row_number =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb142-5"><a href="#cb142-5" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb142-6"><a href="#cb142-6" tabindex="-1"></a>  <span class="fu">mutate_if</span>(is.character, as.numeric)  <span class="sc">%&gt;%</span></span>
<span id="cb142-7"><a href="#cb142-7" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(., <span class="st">&quot;trait&quot;</span>)</span>
<span id="cb142-8"><a href="#cb142-8" tabindex="-1"></a></span>
<span id="cb142-9"><a href="#cb142-9" tabindex="-1"></a>function_gift_filt <span class="ot">&lt;-</span> <span class="fu">subset</span>(function_gift_t, trait <span class="sc">%in%</span> significant_functional<span class="sc">$</span>trait) <span class="sc">%&gt;%</span> </span>
<span id="cb142-10"><a href="#cb142-10" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb142-11"><a href="#cb142-11" tabindex="-1"></a>  <span class="fu">row_to_names</span>(<span class="at">row_number =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb142-12"><a href="#cb142-12" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb142-13"><a href="#cb142-13" tabindex="-1"></a>  <span class="fu">mutate_if</span>(is.character, as.numeric)  <span class="sc">%&gt;%</span></span>
<span id="cb142-14"><a href="#cb142-14" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(., <span class="st">&quot;sample&quot;</span>)<span class="sc">%&gt;%</span> </span>
<span id="cb142-15"><a href="#cb142-15" tabindex="-1"></a>  <span class="fu">left_join</span>(., sample_metadata <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,sex,broad_environment), <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample)) <span class="sc">%&gt;%</span></span>
<span id="cb142-16"><a href="#cb142-16" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(sex))</span>
<span id="cb142-17"><a href="#cb142-17" tabindex="-1"></a></span>
<span id="cb142-18"><a href="#cb142-18" tabindex="-1"></a>function_gift_filt <span class="sc">%&gt;%</span></span>
<span id="cb142-19"><a href="#cb142-19" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(sample,broad_environment)) <span class="sc">%&gt;%</span></span>
<span id="cb142-20"><a href="#cb142-20" tabindex="-1"></a>  <span class="fu">group_by</span>(sex)  <span class="sc">%&gt;%</span></span>
<span id="cb142-21"><a href="#cb142-21" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), mean))<span class="sc">%&gt;%</span></span>
<span id="cb142-22"><a href="#cb142-22" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb142-23"><a href="#cb142-23" tabindex="-1"></a>  <span class="fu">row_to_names</span>(<span class="at">row_number =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb142-24"><a href="#cb142-24" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb142-25"><a href="#cb142-25" tabindex="-1"></a>  <span class="fu">mutate_if</span>(is.character, as.numeric)  <span class="sc">%&gt;%</span></span>
<span id="cb142-26"><a href="#cb142-26" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(., <span class="st">&quot;Code_function&quot;</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb142-27"><a href="#cb142-27" tabindex="-1"></a>  <span class="fu">left_join</span>(unique_funct_db[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)],<span class="at">by =</span> <span class="fu">join_by</span>(Code_function <span class="sc">==</span> Code_function))</span></code></pre></div>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" tabindex="-1"></a>function_gift_names <span class="ot">&lt;-</span> function_gift_filt<span class="sc">%&gt;%</span></span>
<span id="cb143-2"><a href="#cb143-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(sex,broad_environment)) <span class="sc">%&gt;%</span></span>
<span id="cb143-3"><a href="#cb143-3" tabindex="-1"></a>   <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb143-4"><a href="#cb143-4" tabindex="-1"></a>  <span class="fu">row_to_names</span>(<span class="at">row_number =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb143-5"><a href="#cb143-5" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb143-6"><a href="#cb143-6" tabindex="-1"></a>  <span class="fu">mutate_if</span>(is.character, as.numeric)  <span class="sc">%&gt;%</span></span>
<span id="cb143-7"><a href="#cb143-7" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(., <span class="st">&quot;Code_function&quot;</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb143-8"><a href="#cb143-8" tabindex="-1"></a>  <span class="fu">left_join</span>(.,unique_funct_db[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)],<span class="at">by =</span> <span class="fu">join_by</span>(Code_function <span class="sc">==</span> Code_function))<span class="sc">%&gt;%</span></span>
<span id="cb143-9"><a href="#cb143-9" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Code_function)<span class="sc">%&gt;%</span></span>
<span id="cb143-10"><a href="#cb143-10" tabindex="-1"></a>  <span class="fu">select</span>(Function, <span class="fu">everything</span>())<span class="sc">%&gt;%</span></span>
<span id="cb143-11"><a href="#cb143-11" tabindex="-1"></a>   <span class="fu">t</span>()<span class="sc">%&gt;%</span></span>
<span id="cb143-12"><a href="#cb143-12" tabindex="-1"></a>  <span class="fu">row_to_names</span>(<span class="at">row_number =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb143-13"><a href="#cb143-13" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb143-14"><a href="#cb143-14" tabindex="-1"></a>  <span class="fu">mutate_if</span>(is.character, as.numeric)  <span class="sc">%&gt;%</span></span>
<span id="cb143-15"><a href="#cb143-15" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(., <span class="st">&quot;sample&quot;</span>)<span class="sc">%&gt;%</span> </span>
<span id="cb143-16"><a href="#cb143-16" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,sex,broad_environment), <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample))</span>
<span id="cb143-17"><a href="#cb143-17" tabindex="-1"></a></span>
<span id="cb143-18"><a href="#cb143-18" tabindex="-1"></a></span>
<span id="cb143-19"><a href="#cb143-19" tabindex="-1"></a>colNames <span class="ot">&lt;-</span> <span class="fu">names</span>(function_gift_names)[<span class="dv">2</span>]</span>
<span id="cb143-20"><a href="#cb143-20" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> colNames){</span>
<span id="cb143-21"><a href="#cb143-21" tabindex="-1"></a>  plt <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(function_gift_names, <span class="fu">aes</span>(<span class="at">x=</span>sex, <span class="at">y=</span>.data[[i]], <span class="at">color =</span> sex, <span class="at">fill=</span>sex)) <span class="sc">+</span></span>
<span id="cb143-22"><a href="#cb143-22" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.2</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">width =</span> <span class="fl">0.3</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb143-23"><a href="#cb143-23" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">show.legend =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb143-24"><a href="#cb143-24" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values=</span>treatment_colors)<span class="sc">+</span></span>
<span id="cb143-25"><a href="#cb143-25" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>treatment_colors)<span class="sc">+</span></span>
<span id="cb143-26"><a href="#cb143-26" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb143-27"><a href="#cb143-27" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb143-28"><a href="#cb143-28" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fl">0.5</span>, <span class="at">linetype =</span> <span class="st">&quot;solid&quot;</span>, <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb143-29"><a href="#cb143-29" tabindex="-1"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb143-30"><a href="#cb143-30" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb143-31"><a href="#cb143-31" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>())</span>
<span id="cb143-32"><a href="#cb143-32" tabindex="-1"></a><span class="fu">print</span>(plt)</span>
<span id="cb143-33"><a href="#cb143-33" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="community-domains-differences" class="section level3 hasAnchor" number="10.2.3">
<h3 class="hasAnchor"><span class="header-section-number">10.2.3</span> Community domains differences<a href="#community-domains-differences" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" tabindex="-1"></a>domain_gift <span class="ot">&lt;-</span> GIFTs_domains_community <span class="sc">%&gt;%</span> </span>
<span id="cb144-2"><a href="#cb144-2" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb144-3"><a href="#cb144-3" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(., <span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb144-4"><a href="#cb144-4" tabindex="-1"></a>  <span class="fu">merge</span>(sample_metadata <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,sex,broad_environment), <span class="at">by=</span><span class="st">&quot;sample&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" tabindex="-1"></a>unique_domain_db<span class="ot">&lt;-</span> GIFT_db[<span class="fu">c</span>(<span class="dv">4</span>)] <span class="sc">%&gt;%</span> </span>
<span id="cb145-2"><a href="#cb145-2" tabindex="-1"></a>  <span class="fu">distinct</span>(Domain, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb145-3"><a href="#cb145-3" tabindex="-1"></a></span>
<span id="cb145-4"><a href="#cb145-4" tabindex="-1"></a>significant_domain <span class="ot">&lt;-</span> domain_gift <span class="sc">%&gt;%</span></span>
<span id="cb145-5"><a href="#cb145-5" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(sample,sex,broad_environment), <span class="at">names_to =</span> <span class="st">&quot;trait&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb145-6"><a href="#cb145-6" tabindex="-1"></a>    <span class="fu">group_by</span>(trait) <span class="sc">%&gt;%</span></span>
<span id="cb145-7"><a href="#cb145-7" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">p_value =</span> <span class="fu">wilcox.test</span>(value <span class="sc">~</span> sex)<span class="sc">$</span>p.value) <span class="sc">%&gt;%</span></span>
<span id="cb145-8"><a href="#cb145-8" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">p_adjust=</span><span class="fu">p.adjust</span>(p_value, <span class="at">method=</span><span class="st">&quot;BH&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb145-9"><a href="#cb145-9" tabindex="-1"></a>    <span class="fu">filter</span>(p_adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code></pre></div>
<!--chapter:end:08_microbiota_functional_analysis.Rmd-->
</div>
</div>
</div>
<!--bookdown:body:end-->
            </section>

          </div>
        </div>
      </div>
<!--bookdown:link_prev-->
<!--bookdown:link_next-->
    </div>
  </div>
<!--bookdown:config-->

</body>

</html>
